---
title: "5. Alpha Diversity Estimates"
description: |
  Reproducible workflow to assess alpha diversity across temperature treatments using Hill numbers.
author:
#  - name: Jarrod J Scott
#    url: https://example.com/norajones
#    affiliation: Spacely Sprockets
#    affiliation_nrl: https://example.com/spacelysprokets
bibliography: assets/cite.bib
output:
    distill::distill_article:
      css: assets/css/styles.css
      toc: true
      toc_depth: 3
---

<details markdown="1">
<summary><strong>Click here</strong> for setup information.</summary>

```{r setup, message=FALSE, results = 'hide'}
knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
set.seed(119)
#library(conflicted)
library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
pacman::p_load(tidyverse, DT, ampvis2, hilldiv, 
               microbiome, phytools, phangorn, 
               pairwiseAdonis, codefolder, naniar, 
               labdsv, patchwork, agricolae, ggpubr,
               reactable, downloadthis, captioner,
               install = FALSE, update = FALSE)
source("hack_code/div_test_plot_jjs.R")
source("hack_code/div_test_jjs.R")

options(scipen=999)
knitr::opts_current$get(c(
  "cache",
  "cache.path",
  "cache.rebuild",
  "dependson",
  "autodep"
))
```

```{r, echo=FALSE, eval=TRUE}
xaringanExtra::use_panelset()
```

```{r, message=FALSE, results = 'hide', eval=TRUE}
## Load to build page only #2
remove(list = ls())
load("page_build/alpha_wf_ssu.rdata")
```

```{r, message=FALSE, results = 'hide', eval=TRUE}
### COmmon formatting scripts
### NOTE: captioner.R must be read BEFORE captions_XYZ.R
source("assets/captioner/captioner.R")
source("assets/captioner/captions/captions_alpha.R")
source("assets/reactable/download_this_fun.R")
source("assets/reactable/styles.R")
source("assets/reactable/table_functions/alpha.R")
```

</details>


# Synopsis

This workflow contains diversity assessments for the 2018 high temperature data sets. In order to run the workflow, you either need to first run the  [DADA2 Workflow for 2018 High Temp samples](dada2.html) **and** the [Data Preparation workflow](data-prep.html) **or** begin with the output files from the Data Preparation workflow. See the [Data Availability](data-availability.html) page for complete details.

# 16S rRNA

```{r, include=FALSE}
## Initial Load for  ANALYSIS #1
remove(list = ls())
set.seed(119)
ssu_ps_work <- readRDS("files/filtering/working_ps/ssu_ps_work.rds")
ssu_ps_filt <- readRDS("files/filtering/working_ps/ssu_ps_filt.rds")
ssu_ps_perfect <- readRDS("files/filtering/working_ps/ssu_ps_perfect.rds")
ssu_ps_pime <- readRDS("files/filtering/working_ps/ssu_ps_pime.rds")
```

To account for presence of rare sequence variants caused by sequencing errors or other technical artifacts, we use Hill numbers [@alberdi2019guide]. Hill numbers allow the weight put on rare versus abundant sequence variants to be scaled while providing intuitive comparisons of diversity levels using “effective number of ASVs” as a measuring unit. This approach allows for balancing the over representation of rare ASVs that might be inflated due to sequencing errors.

We will then use Shapiro-Wilk tests to test for normalcy and then, depending on the results, either use  parametric ANOVA or non-parametric Kruskal-Wallis to compare alpha diversity among treatments.

```{r, echo=FALSE}
objects()
ssu_ps_work@phy_tree <- NULL
ssu_ps_filt@phy_tree <- NULL
ssu_ps_pime@phy_tree <- NULL
ssu_ps_perfect@phy_tree <- NULL
```

Before we begin the analysis, let's subset each data set by Year and take a look at some summary data.

```{r, code_folding=TRUE}
ssu_alpha_ds <- c("ssu_ps_work", "ssu_ps_filt", "ssu_ps_perfect", "ssu_ps_pime")
for (i in ssu_alpha_ds) {
  tmp_get <- get(i)
  tmp_samples <- row.names(data.frame(otu_table(tmp_get)))
  tmp_y0_samps <- stringr::str_subset(tmp_samples, "_Y0_")
  tmp_y1_samps <- stringr::str_subset(tmp_samples, "_Y1_")
  tmp_y4_samps <- stringr::str_subset(tmp_samples, "_Y4_")

  tmp_y0 <- prune_samples(tmp_y0_samps, tmp_get) 
  tmp_y1 <- prune_samples(tmp_y1_samps, tmp_get)
  tmp_y4 <- prune_samples(tmp_y4_samps, tmp_get)
  
  tmp_y0 <- prune_taxa(taxa_sums(tmp_y0) > 0, tmp_y0)
  tmp_y1 <- prune_taxa(taxa_sums(tmp_y1) > 0, tmp_y1)
  tmp_y4 <- prune_taxa(taxa_sums(tmp_y4) > 0, tmp_y4)
  
  tmp_ps_name_y0 <- purrr::map_chr(i, ~ paste0(., "_Y0"))
  assign(tmp_ps_name_y0, tmp_y0)
  
  tmp_ps_name_y1 <- purrr::map_chr(i, ~ paste0(., "_Y1"))
  assign(tmp_ps_name_y1, tmp_y1)
  
  tmp_ps_name_y4 <- purrr::map_chr(i, ~ paste0(., "_Y4"))
  assign(tmp_ps_name_y4, tmp_y4)
  
  tmp_path <- file.path("files/alpha/rdata/")
  saveRDS(tmp_y0, paste(tmp_path, tmp_ps_name_y0, ".rds", sep = ""))
  saveRDS(tmp_y1, paste(tmp_path, tmp_ps_name_y1, ".rds", sep = ""))
  saveRDS(tmp_y4, paste(tmp_path, tmp_ps_name_y4, ".rds", sep = ""))
  rm(list = ls(pattern = "tmp_"))
}
```

```{r, echo=FALSE}
tmp_objects <- data.frame(c("FULL data set (Y0)", 
                            "FULL data set (Y1)", 
                            "FULL data set (Y4)", 
                            "Arbitrary filter data (Y0)", 
                            "Arbitrary filter data (Y1)", 
                            "Arbitrary filter data (Y4)", 
                            "PERfect filtered data (Y0)",
                            "PERfect filtered data (Y1)",
                            "PERfect filtered data (Y4)",
                            "PIME (Y0)",
                            "PIME (Y1)", 
                            "PIME (Y4)"))
tmp_samples <- c("ssu_ps_work_Y0", "ssu_ps_work_Y1", "ssu_ps_work_Y4",
                 "ssu_ps_filt_Y0", "ssu_ps_filt_Y1", "ssu_ps_filt_Y4",
                 "ssu_ps_perfect_Y0", "ssu_ps_perfect_Y1", "ssu_ps_perfect_Y4",
                 "ssu_ps_pime_Y0", "ssu_ps_pime_Y1", "ssu_ps_pime_Y4")
tmp_no_samp <- c()
for (i in tmp_samples) {
   tmp_get <- nsamples(get(i))
   tmp_no_samp <- c(append(tmp_no_samp, tmp_get))
}
tmp_no_samp <- data.frame(tmp_no_samp)

tmp_rc <- c()
for (i in tmp_samples) {
   tmp_get <- sum(readcount(get(i)))
   tmp_rc <- c(append(tmp_rc, tmp_get))
}
tmp_rc <- data.frame(tmp_rc)
tmp_asv <- c()
for (i in tmp_samples) {
   tmp_get <- ntaxa(get(i))
   tmp_asv <- c(append(tmp_asv, tmp_get))
}
tmp_asv <- data.frame(tmp_asv)

ssu_split_summary <- dplyr::bind_cols(tmp_objects, tmp_samples) %>%
                      dplyr::bind_cols(., tmp_no_samp) %>%
                      dplyr::bind_cols(., tmp_rc) %>%
                      dplyr::bind_cols(., tmp_asv) %>%
  dplyr::rename("Description" = 1, "object name" = 2, "no. samples" = 3,
                "total reads" = 4, "total asvs" = 5)
rm(list = ls(pattern = "tmp_"))

write.table(ssu_split_summary, "files/alpha/tables/ssu_split_by_year.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r, echo=FALSE}
ssu_alpha_ds <- c("ssu_ps_work", "ssu_ps_filt", "ssu_ps_perfect", "ssu_ps_pime")

tmp_objects <- data.frame(c("FULL data set", 
                            "Arbitrary filter", 
                            "PERfect filtered",
                            "PIME filtered"))
tmp_samples <- c("ssu_ps_work",
                 "ssu_ps_filt",
                 "ssu_ps_perfect",
                 "ssu_ps_pime")
tmp_no_samp <- c()
for (i in tmp_samples) {
   tmp_get <- nsamples(get(i))
   tmp_no_samp <- c(append(tmp_no_samp, tmp_get))
}
tmp_no_samp <- data.frame(tmp_no_samp)

tmp_rc <- c()
for (i in tmp_samples) {
   tmp_get <- sum(readcount(get(i)))
   tmp_rc <- c(append(tmp_rc, tmp_get))
}
tmp_rc <- data.frame(tmp_rc)
tmp_asv <- c()
for (i in tmp_samples) {
   tmp_get <- ntaxa(get(i))
   tmp_asv <- c(append(tmp_asv, tmp_get))
}
tmp_asv <- data.frame(tmp_asv)

ssu_filter_summary <- dplyr::bind_cols(tmp_objects, tmp_samples) %>%
                      dplyr::bind_cols(., tmp_no_samp) %>%
                      dplyr::bind_cols(., tmp_rc) %>%
                      dplyr::bind_cols(., tmp_asv) %>%
  dplyr::rename("Description" = 1, "object name" = 2, "no. samples" = 3,
                "total reads" = 4, "total asvs" = 5)
rm(list = ls(pattern = "tmp_"))

write.table(ssu_filter_summary, "files/alpha/tables/ssu_filter_summmary.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r, echo=FALSE}
col_order <- c("Description", "object name", "Year", 
               "no. samples", "total reads", "total asvs")

tmp_all <- ssu_filter_summary
tmp_all$Year <- "All years"
tmp_all <- tmp_all[, col_order]

tmp_subset <- ssu_split_summary
tmp_subset$Year <- tmp_subset$`object name`

tmp_subset$Year <- gsub("^.*_Y", "Year ", tmp_subset$Year)
tmp_subset <- tmp_subset[, col_order]

ssu_dataset_summary <- rbind(tmp_all, tmp_subset)
ssu_dataset_summary <- ssu_dataset_summary %>% 
                       dplyr::slice(1,5:7,2,8:10,3,11:13,4,14:16)
rm(list = ls(pattern = "tmp_"))
```

<small>`r caption_tab_ssu("ssu_dataset_summary")`</small>

```{r eval=TRUE, echo=FALSE, layout="l-body-outset"}
download_this_fun(ssu_dataset_summary)
```

```{r, echo=FALSE, layout="l-body-outset", eval=TRUE}
ds_summ_table(seq_table)
rm(seq_table)
```


## Calculate Hill Numbers

To calculate Hill numbers, we use the R package `hilldiv` [@alberdi2019hilldiv]. We calculate three metrics that put more or less weight on common species:

1) Observed richness, where `q-value = 0`.
2) Shannon exponential, which weighs ASVs by their frequency,  where `q-value = 1`.
3) Simpson multiplicative inverse, which over weighs abundant ASVs,  where `q-value = 2`.

We perform each analysis against the Full (unfiltered) data set as well as the Arbitrary, PERFect, and PIME, filtered data sets using the function `hill_div`.

The command is as follows:

```
hill_div(count = x, qvalue = i, tree = ultrametric_tree)
``` 

where `x` is the sample by ASV table, `i` is the q-value corresponding to the metric of interest and `tree` is an ultrametric formatted phylogenetic tree if you choose to look at lineage, rather than ASV, diversity.


Next, transform all the data to relative abundance values, and compute new trees.

```{r, code_folding=TRUE}
ssu_alpha_ds <- c("ssu_ps_work_Y0", "ssu_ps_work_Y1", "ssu_ps_work_Y4", 
                  "ssu_ps_filt_Y0", "ssu_ps_filt_Y1", "ssu_ps_filt_Y4", 
                  "ssu_ps_perfect_Y0", "ssu_ps_perfect_Y1", "ssu_ps_perfect_Y4", 
                  "ssu_ps_pime_Y0", "ssu_ps_pime_Y1", "ssu_ps_pime_Y4")
for (i in ssu_alpha_ds) {
  tmp_ps <- transform_sample_counts(get(i), function(otu) otu/sum(otu))
  tmp_ps@phy_tree <- NULL
  tmp_tree <- rtree(ntaxa(tmp_ps), rooted = TRUE,
                      tip.label = taxa_names(tmp_ps))
  tmp_ps_norm <- merge_phyloseq(tmp_ps, sample_data, tmp_tree)
  tmp_asv <- data.frame(t(otu_table(tmp_ps_norm)))
  tmp_ps_name <- purrr::map_chr(i, ~ paste0(., "_norm"))
  assign(tmp_ps_name, tmp_ps_norm)
  tmp_asv_name <- purrr::map_chr(i, ~ paste0(., "_tu"))
  assign(tmp_asv_name, tmp_asv)
  rm(list = ls(pattern = "tmp_"))
}
```

Next, we run the analysis for all three metrics on each data sets.

```{r, results='hide', code_folding=TRUE}
qvalue <- c(0,1,2)
for (i in qvalue) {
  for (j in ssu_alpha_ds) {
     tmp_asv <- get(purrr::map_chr(j, ~ paste0(., "_tu")))
     tmp_df <- data.frame(hill_div(tmp_asv, qvalue = i))
     tmp_df <- tmp_df %>% dplyr::rename("tmp_name" = 1) %>%
                              tibble::rownames_to_column("SamName")
     tmp_name <- purrr::map_chr(j, ~ paste0(., "_h", i))
     print(tmp_name)
     assign(tmp_name, tmp_df)
     rm(list = ls(pattern = "tmp_"))
  }
}
objects(pattern = "_h")
```

Then we make summary tables to add back into each `ps` object.

```{r, code_folding=TRUE}
for (i in ssu_alpha_ds) {
     tmp_obs <- get(purrr::map_chr(i, ~ paste0(., "_h0")))
     tmp_sha <- get(purrr::map_chr(i, ~ paste0(., "_h1")))
     tmp_sim <- get(purrr::map_chr(i, ~ paste0(., "_h2")))
     tmp_hill <- dplyr::left_join(tmp_obs, tmp_sha, by = "SamName") %>%
       dplyr::left_join(., tmp_sim, by = "SamName") %>%
       dplyr::rename("Observed" = 2, "Shannon_exp" = 3, "InvSimpson" = 4)
     tmp_name <- purrr::map_chr(i, ~ paste0(., "_hill"))
     assign(tmp_name, tmp_hill)
     rm(list = ls(pattern = "tmp_"))
}
```

And then create the new objects with the diversity data.

```{r, code_folding=TRUE}
for (i in ssu_alpha_ds) {
     tmp_ps <- get(i)
     tmp_tree <- rtree(ntaxa(tmp_ps), rooted = TRUE,
                      tip.label = taxa_names(tmp_ps))
     tmp_ps <- merge_phyloseq(tmp_ps, sample_data, tmp_tree)
     tmp_hill <- get(purrr::map_chr(i, ~ paste0(., "_hill")))
     tmp_hill_samp <- dplyr::left_join(data.frame(sample_data(tmp_ps)),
                                         tmp_hill, by = "SamName")
     tmp_hill_samp$ID <- tmp_hill_samp$SamName
     tmp_hill_samp <- tmp_hill_samp %>%  tibble::column_to_rownames("ID")
     tmp_ps2 <- merge_phyloseq(otu_table(tmp_ps),
                                sample_data(tmp_hill_samp),
                                tax_table(tmp_ps),
                                phy_tree(tmp_ps))
     assign(i, tmp_ps2)
     tmp_path <- file.path("files/alpha/rdata/")
     saveRDS(tmp_ps2, paste(tmp_path, i, ".rds", sep = ""))
     rm(list = ls(pattern = "tmp_"))
}
objects(pattern = "_hill")
```

## Hill Numbers 

Now we summarize the data for each sample against all three metrics. The table contains the results of ASV diversity estimates from the full data set and the three filtered data sets.

The suffix `_fi` indicates metrics for the Arbitrary data set.  The suffix `_pe` indicates metrics for the PERFect data set and the suffix `_pi` indicates metrics for the PIME data set.

```{r, echo=FALSE}
######################### FULL ASV
for (i in objects(pattern = "ssu_ps_work_Y[0-9]_hill")) {
  tmp_get <- get(i)
  tmp_samp <- tmp_get %>% dplyr::rename("Observed" = 2,"Shannon_exp" = 3, "InvSimpson" = 4)
  tmp_get_ps <- get(gsub("_hill", "", i, fixed = TRUE))
  tmp_samp <- dplyr::left_join(data.frame(sample_data(tmp_get_ps)), tmp_samp)
  tmp_samp <- tmp_samp[order(tmp_samp$YEAR, tmp_samp$TREAT),]
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_samp")) 
  assign(tmp_name, tmp_samp)  
  rm(list = ls(pattern = "tmp_"))
} 

for (i in objects(pattern = "ssu_ps_filt_Y[0-9]_hill")) {
  tmp_get <- get(i)
  tmp_samp <- tmp_get %>% dplyr::rename("Observed_fi" = 2,"Shannon_exp_fi" = 3, "InvSimpson_fi" = 4)
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_samp")) 
  assign(tmp_name, tmp_samp)  
  rm(list = ls(pattern = "tmp_"))
} 

for (i in objects(pattern = "ssu_ps_perfect_Y[0-9]_hill")) {
  tmp_get <- get(i)
  tmp_samp <- tmp_get %>% dplyr::rename("Observed_pe" = 2,"Shannon_exp_pe" = 3, "InvSimpson_pe" = 4)
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_samp")) 
  assign(tmp_name, tmp_samp)  
  rm(list = ls(pattern = "tmp_"))
} 

for (i in objects(pattern = "ssu_ps_pime_Y[0-9]_hill")) {
  tmp_get <- get(i)
  tmp_samp <- tmp_get %>% dplyr::rename("Observed_pi" = 2,"Shannon_exp_pi" = 3, "InvSimpson_pi" = 4)
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_samp")) 
  assign(tmp_name, tmp_samp)  
  rm(list = ls(pattern = "tmp_"))
} 
objects(pattern = "_samp")
```

```{r, echo=FALSE}
######################### ASV TABLE
year_select <- c("Y0", "Y1", "Y4")
for (i in year_select) {
  tmp_work <- get(paste0("ssu_ps_work_", i, "_hill_samp"))
  tmp_filt <- get(paste0("ssu_ps_filt_", i, "_hill_samp"))
  tmp_perfect <- get(paste0("ssu_ps_perfect_", i, "_hill_samp"))
  tmp_pime <- get(paste0("ssu_ps_pime_", i, "_hill_samp"))
  tmp_alpha_div<- dplyr::left_join(tmp_work, tmp_filt) %>%
                  dplyr::left_join(., tmp_perfect) %>%
                  dplyr::left_join(., tmp_pime) %>% 
         dplyr::relocate(c("Observed", "Observed_fi", "Observed_pe", "Observed_pi"), .after = "YR_TREAT") %>% 
         dplyr::relocate(c("Shannon_exp", "Shannon_exp_fi", "Shannon_exp_pe", "Shannon_exp_pi"), .after = "Observed_pi") %>%
         dplyr::rename("Sample_ID" = "SamName") 
  tmp_alpha_div <- tmp_alpha_div%>% dplyr::mutate(across(.cols = c(13:ncol(tmp_alpha_div)), 
                                                               round, digits = 1))
  tmp_name <- paste0("ssu_tab_alpha_div_", i, sep = "" ) 
  assign(tmp_name, tmp_alpha_div)  
  
  rm(list = ls(pattern = "tmp_"))
}
```


::: l-page
::: {.panelset}
::: {.panel}

#### Year 0

<small>`r caption_tab_ssu("ssu_tab_alpha_div_Y0")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_tab_alpha_div_Y0)
seq_table[3:4] <- NULL
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
hill_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### Year 1

<small>`r caption_tab_ssu("ssu_tab_alpha_div_Y1")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_tab_alpha_div_Y1)
seq_table[3:4] <- NULL
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
hill_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### Year 4

<small>`r caption_tab_ssu("ssu_tab_alpha_div_Y4")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_tab_alpha_div_Y4)
seq_table[3:4] <- NULL
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
hill_table(seq_table)
rm(seq_table)
```
:::
:::
:::

## Normality Tests

Before running significance tests, we need to know if data is normally distributed, which will tell use whether to use a parametric or non-parametric test. To test if the data are normally distributed, we use the Shapiro-Wilk Normality test and the Bartlett Test of Homogeneity of Variances.

If the p-values are both **not significant** (p > 0.05) from the tests, we accept the null hypothesis (that the results are normally distributed) and test for significance between samples using an ANOVA. If the p-values are both **significant** (p < 0.05), we reject the null hypothesis (that the results are normally distributed) and test for significance between samples using Kruskal-Wallis (non-parametric equivalent of ANOVA).

The commands are as follows:

`shapiro.test(x)`, where `x` is a numeric vector of alpha diversity  values from the sample data table.

`bartlett.test(Value ~ Group, data = df)` Where `Value` is the metric of interest, `Group` in the treatment to compare, and `df` is the data frame.  

First the Shapiro-Wilk Normality test.

```{r, code_folding=TRUE}
#rm(list = ls(pattern = "shap"))
shap_test <-  function(alpha_div_tab) {
  shap_deparse <- deparse(substitute(alpha_div_tab))
  shap_year <- gsub(".*_", "", shap_deparse)
  shap_test_list <- c()
  for (i in colnames(alpha_div_tab[,9:ncol(alpha_div_tab)])) {
      tmp_name <- purrr::map_chr(i, ~ paste0("ssu_shap_", shap_year, "_", .))
      shap_test_list <- append(shap_test_list, tmp_name)
      tmp_test <- eval(shapiro.test(alpha_div_tab[[i]]))
      tmp_test$data.name <- tmp_name
      assign(tmp_name, tmp_test, envir = parent.frame() )
      rm(list = ls(pattern = "tmp_"))
  }
}
```

```{r}
shap_test(ssu_tab_alpha_div_Y0)
shap_test(ssu_tab_alpha_div_Y1)
shap_test(ssu_tab_alpha_div_Y4)
objects(pattern = "ssu_shap_")
```

And then the Bartlett Test of Homogeneity of Variances. For the Bartlett Test, we need to define `g`, a vector or factor object giving the group for the corresponding elements of x. 

```{r, code_folding=TRUE}
bart_test <-  function(alpha_div_tab) {
  bart_g <- "YR_TREAT"
  bart_deparse <- deparse(substitute(alpha_div_tab))
  bart_year <- gsub(".*_", "", bart_deparse)
  bart_test_list <- c()
  for (i in colnames(alpha_div_tab[,9:ncol(alpha_div_tab)])) {
      tmp_name <- purrr::map_chr(i, ~ paste0("ssu_bart_", bart_year, "_", .))
      bart_test_list <- append(bart_test_list, tmp_name)
      tmp_test <- eval(bartlett.test(alpha_div_tab[[i]] ~ get(bart_g), data = alpha_div_tab))
      tmp_test$data.name <- tmp_name
      assign(tmp_name, tmp_test, envir = parent.frame() )
      rm(list = ls(pattern = "tmp_"))
  }
}
```

```{r}
bart_test(ssu_tab_alpha_div_Y0)
bart_test(ssu_tab_alpha_div_Y1)
bart_test(ssu_tab_alpha_div_Y4)
```

Here we see which Shapiro-Wilk Normality **and** Bartlett tests were significant and which were not. So wherever the value of both p-values in `> 0.05` we can use an ANOVA, otherwise we use Kruskal-Wallis.

```{r, code_folding=TRUE}
combo_tests <-  function(alpha_div_tab) {
  combo_deparse <- deparse(substitute(alpha_div_tab))
  combo_year <- gsub(".*_", "", combo_deparse)
  combo_results <- c()
  for (i in colnames(alpha_div_tab[,9:ncol(alpha_div_tab)])) {
     tmp_get_shap <- get(purrr::map_chr(i, ~ paste0("ssu_shap_", combo_year, "_", .)))
     tmp_shap_p <- round(tmp_get_shap[[2]], 4)
     tmp_get_bart <- get(purrr::map_chr(i, ~ paste0("ssu_bart_", combo_year, "_", .)))
     tmp_bart_p <- round(tmp_get_bart[[3]], 4)
     tmp_test <- if(tmp_shap_p > 0.05 & tmp_bart_p > 0.05 ){
          tmp_test <- "ANOVA"
        } else {
          tmp_test <- "Kruskal-Wallis"
        }
     tmp_df <- data.frame(i, tmp_shap_p, tmp_bart_p, tmp_test)
     combo_results <- dplyr::bind_rows(combo_results, tmp_df)
     
     rm(list = ls(pattern = "tmp_"))
  }
  combo_results[,1] <- gsub('_exp', '', combo_results[,1])
  combo_results <- combo_results %>% separate(col = 1, 
                                                into = c("metric", "dataset"), 
                                                sep = "_", remove = TRUE)
  combo_results <- combo_results %>% tidyr::replace_na(list(dataset = "FULL"))
  combo_results$dataset <- stringr::str_replace(combo_results$dataset, "^fi$", "FILT") %>%
                           stringr::str_replace(., "^pi$", "PIME") %>%
                           stringr::str_replace(., "^pe$", "PERFect")

  combo_results <- combo_results %>% 
    dplyr::rename(c("p-value (normality)" = 3, "p-value (homogeneity)" = 4, "method" = 5))
  
  norm_res_name <- paste0("ssu_norm_res_", combo_year)
  cat("object saved as ", norm_res_name)
  assign(norm_res_name, combo_results, envir = parent.frame() )
}
```

```{r}
combo_tests(ssu_tab_alpha_div_Y0)
combo_tests(ssu_tab_alpha_div_Y1)
combo_tests(ssu_tab_alpha_div_Y4)
```

::: l-page
::: {.panelset}
::: {.panel}

#### Year 0

<small>`r caption_tab_ssu("ssu_norm_res_Y0")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_norm_res_Y0)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
norm_res_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### Year 1

<small>`r caption_tab_ssu("ssu_norm_res_Y1")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_norm_res_Y1)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
norm_res_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### Year 4

<small>`r caption_tab_ssu("ssu_norm_res_Y4")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_norm_res_Y4)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
norm_res_table(seq_table)
rm(seq_table)
```
:::
:::
:::


## Significance Tests

To begin, we need to create a hierarchy variable; a two-column matrix specifying the relationship between samples (first column) and groups (second column).

```{r, code_folding=TRUE}
sig_group <- "YR_TREAT"
for (i in objects(pattern = "ssu_tab_alpha_div")) {
    tmp_hill_hier <- get(i)
    tmp_hill_hier <- tmp_hill_hier %>% dplyr::select("Sample_ID", sig_group) %>%
      tibble::remove_rownames()
    tmp_hill_hier <- tmp_hill_hier[order(tmp_hill_hier[[2]]), ]
    #tmp_hill_hier$YR_TREAT = paste(tmp_hill_hier$YR_TREAT, 'C', sep='')
    tmp_hill_hier <- tmp_hill_hier %>% tibble::remove_rownames()
    tmp_yr <- gsub(".*_", "", i)
    tmp_name <- paste("ssu_hill_hier", tmp_yr, sep = "_")
    assign(tmp_name, tmp_hill_hier)
    tmp_path <- file.path("files/alpha/rdata/")
    saveRDS(tmp_hill_hier, paste(tmp_path, tmp_name, ".rds", sep = ""))
    rm(list = ls(pattern = "tmp_"))
}

objects(pattern = "ssu_hill_hier")
```

Again, we start by testing significance of ASV diversity for the data sets against each of the three metrics using the `div_test` function.

The command is as follows:

```
div_test(countable = x, qvalue = i, hierarchy = hier, 
         tree = ultrametric_tree, posthoc = TRUE)
```

Where `x` is ASV by sample table, `i` is the q-value corresponding to the metric of interest, `hier` is the hierarchy matrix, `tree` is an ultrametric formatted phylogenetic tree (if you choose to explore lineage diversity), and `posthoc` indicates whether to run post hoc pairwise analyses.

```{r, results='hide', code_folding=TRUE}
qvalue <- c(0,1,2)
for (i in ssu_alpha_ds) {
     for (j in qvalue) {
         tmp_year <- gsub(".*_", "", i)
         tmp_get <- get(purrr::map_chr(i, ~ paste0(., "_tu")))
         tmp_heir <- get(paste("ssu_hill_hier", tmp_year, sep = "_"))
         tmp_test <- div_test(tmp_get, qvalue = j,
                      hierarchy = tmp_heir,
                      posthoc = TRUE)
         tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
         print(tmp_name)
         assign(tmp_name, tmp_test)
         rm(list = ls(pattern = "tmp_"))
 }
}
```


```{r, echo=FALSE}
tmp_metric <- data.frame(c("Observed", "Shannon exponential", "Inverse Simpson"))
tmp_qvalue <- data.frame(c("0", "1", "2"))

qvalue <- c(0,1,2)
for (i in ssu_alpha_ds) {

  tmp_h_pvalue <- c()
     for (j in qvalue) {
          tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
          tmp_get <- get(tmp_name)$homogeneity.pvalue
          tmp_h_pvalue <- c(append(tmp_h_pvalue, tmp_get))
     }
  tmp_h_pvalue <- data.frame(tmp_h_pvalue)

  tmp_n_pvalue <- c()
      for (j in qvalue) {
          tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
          tmp_get <- get(tmp_name)$normality.pvalue
          tmp_n_pvalue <- c(append(tmp_n_pvalue, tmp_get))
     }
  tmp_n_pvalue <- data.frame(tmp_n_pvalue)

  tmp_method <- c()
      for (j in qvalue) {
          tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
          tmp_get <- get(tmp_name)$method
          tmp_method <- c(append(tmp_method, tmp_get))
     }
  tmp_method <- data.frame(tmp_method)

  tmp_phoc_method <- c()
      for (j in qvalue) {
          tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
          tmp_get <- get(tmp_name)$posthoc.method
          tmp_phoc_method <- c(append(tmp_phoc_method, tmp_get))
     }
  tmp_phoc_method <- data.frame(tmp_phoc_method)

  tmp_df <- dplyr::bind_cols(tmp_metric, tmp_qvalue) %>%
                     dplyr::bind_cols(., tmp_n_pvalue) %>%
                     dplyr::bind_cols(., tmp_h_pvalue) %>%
                     dplyr::bind_cols(., tmp_method) %>%
                     dplyr::bind_cols(., tmp_phoc_method) %>%
  dplyr::rename("metric" = 1, "q-value" = 2, "normality p-value" = 3, 
                "homogeneity p-value" = 4, "method" = 5, "posthoc method" = 6)
  tmp_name <- purrr::map_chr(i, ~ paste0(i, "_sig_tab"))
  assign(tmp_name, tmp_df)
}
```


```{r, echo=FALSE}
## Format as follows
## FALSE Kruskal-Wallis Test = its_pime_q1_adt$test[[3]]
## TRUE Tukey post-hoc  = its_ps_work_q0_adt$test[[1]][[1, 5]]
get_pvalue <-  function(value) {
        if (value$method == "ANOVA") {
          tmp_pvalue <- "[[1]][[1, 5]]" 
        } else if (value$method == "Kruskal-Wallis Test") {
          tmp_pvalue <- "[[3]]"
        } 
  tmp_test <- paste(deparse(substitute(value)), "$test",  noquote(tmp_pvalue), sep = "")
  return(eval(parse(text = tmp_test)))
}
objects(pattern = "_adt")
```

```{r, echo=FALSE}
qvalue <- c(0,1,2)
gen_summ <-  function(ps_object, year) {
  for (i in qvalue) {
  tmp_qname <-  paste("tmp_q", i, sep = "")  
  tmp_get <- get(paste(ps_object, "_", year, "_q", i,  "_adt", sep = ""))
  assign(tmp_qname, tmp_get, envir = parent.frame())
  }
  tmp_pvalue <- data.frame(c(get_pvalue(tmp_q0),
                             get_pvalue(tmp_q1),
                             get_pvalue(tmp_q2)
                             )
                           )
  tmp_sig_tab_df <- get(paste(ps_object, "_", year, "_sig_tab", sep = ""))
  tmp_sig_tab <- dplyr::bind_cols(tmp_sig_tab_df, tmp_pvalue) %>%
                         dplyr::rename("posthoc p-value" = 7)
  tmp_name <- paste(ps_object, "_", year, "_sig_tab_post", sep = "")
  assign(tmp_name, tmp_sig_tab, envir = parent.frame())
  rm(list = ls(pattern = "tmp_"))
} 
objects()
```

```{r, echo=FALSE}
gen_summ("ssu_ps_work", "Y0")
gen_summ("ssu_ps_filt", "Y0")
gen_summ("ssu_ps_perfect", "Y0")
gen_summ("ssu_ps_pime", "Y0")

gen_summ("ssu_ps_work", "Y1")
gen_summ("ssu_ps_filt", "Y1")
gen_summ("ssu_ps_perfect", "Y1")
gen_summ("ssu_ps_pime", "Y1")

gen_summ("ssu_ps_work", "Y4")
gen_summ("ssu_ps_filt", "Y4")
gen_summ("ssu_ps_perfect", "Y4")
gen_summ("ssu_ps_pime", "Y4")
rm(list = ls(pattern = "tmp_"))
```

```{r, echo=FALSE}
for (i in objects(pattern = "_sig_tab_post")) {
  tmp_get <- get(i)
  tmp_ds_type <- stringr::str_remove(i, "ssu_ps_") %>%
                 stringr::str_remove(., "_.*")
      if (tmp_ds_type == "work") {
          tmp_get$dataset <- "FULL" 
        } else if (tmp_ds_type == "filt") {
          tmp_get$dataset <- "FILT"
        } else if (tmp_ds_type == "perfect") {
          tmp_get$dataset <- "PERFect"
        } else if (tmp_ds_type == "pime") {
          tmp_get$dataset <- "PIME"
        } 
  tmp_ds_year <- stringr::str_extract(i, "Y[0-9]") 
      if (tmp_ds_year == "Y0") {
          tmp_get$year <- "Year 0" 
        } else if (tmp_ds_year == "Y1") {
          tmp_get$year <- "Year 1"
        } else if (tmp_ds_year == "Y4") {
          tmp_get$year <- "Year 4"
        } 

  tmp_get <- tmp_get[,c(8:9,1:7)]
  assign(i, tmp_get)
  rm(list = ls(pattern = "tmp_"))
}
objects(pattern = "_sig_tab_post")
```

```{r, echo=FALSE}
ssu_sig_tab_all_Y0 <- rbind(ssu_ps_work_Y0_sig_tab_post, 
                           ssu_ps_filt_Y0_sig_tab_post, 
                           ssu_ps_perfect_Y0_sig_tab_post,
                           ssu_ps_pime_Y0_sig_tab_post
                           )
ssu_sig_tab_all_Y1 <- rbind(ssu_ps_work_Y1_sig_tab_post, 
                           ssu_ps_filt_Y1_sig_tab_post, 
                           ssu_ps_perfect_Y1_sig_tab_post,
                           ssu_ps_pime_Y1_sig_tab_post
                           )
ssu_sig_tab_all_Y4 <- rbind(ssu_ps_work_Y4_sig_tab_post, 
                           ssu_ps_filt_Y4_sig_tab_post, 
                           ssu_ps_perfect_Y4_sig_tab_post,
                           ssu_ps_pime_Y4_sig_tab_post
                           )
```

::: l-page
::: {.panelset}
::: {.panel}

#### Year 0

<small>`r caption_tab_ssu("ssu_sign_tests_Y0")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_sig_tab_all_Y0)
seq_table <- seq_table %>% dplyr::mutate(across(.cols = c(5:6,9), round, digits = 4)) 
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
sig_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}

#### Year 1

<small>`r caption_tab_ssu("ssu_sign_tests_Y1")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_sig_tab_all_Y1)
seq_table <- seq_table %>% dplyr::mutate(across(.cols = c(5:6,9), round, digits = 4)) 
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
sig_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}

#### Year 4

<small>`r caption_tab_ssu("ssu_sign_tests_Y4")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_sig_tab_all_Y4)
seq_table <- seq_table %>% dplyr::mutate(across(.cols = c(5:6,9), round, digits = 4)) 
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
sig_table(seq_table)
rm(seq_table)
```
:::
:::
:::

## Results of PostHoc Analyses 

Here we can inspect the results of each posthoc analysis for the unfiltered and filtered data sets.

::: l-body
::: {.panelset}
::: {.panel}

#### Year 0

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Full</strong> data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_f0_lab <- "Y0 FULL (Observed)"
ssu_Y0_f0_lab
ssu_ps_work_Y0_q0_adt$posthoc.method
data.frame(ssu_ps_work_Y0_q0_adt$posthoc)
objects()
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_f1_lab <- "Y0 FULL (Shannon exponential)"
ssu_Y0_f1_lab
ssu_ps_work_Y0_q1_adt$posthoc.method
data.frame(ssu_ps_work_Y0_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_f2_lab <- "Y0 FULL (Inverse Simpson)"
ssu_Y0_f2_lab
ssu_ps_work_Y0_q2_adt$posthoc.method
data.frame(ssu_ps_work_Y0_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Arbitrary</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_l0_lab <- "Y0 FILT (Observed)"
ssu_Y0_l0_lab
ssu_ps_filt_Y0_q0_adt$posthoc.method
data.frame(ssu_ps_filt_Y0_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_l1_lab <- "Y0 FILT (Shannon exponential)"
ssu_Y0_l1_lab
ssu_ps_filt_Y0_q1_adt$posthoc.method
data.frame(ssu_ps_filt_Y0_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_l2_lab <- "Y0 FILT (Inverse Simpson)"
ssu_Y0_l2_lab
ssu_ps_filt_Y0_q2_adt$posthoc.method
data.frame(ssu_ps_filt_Y0_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PERFect</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_r0_lab <- "Y0 PERFect (Observed)"
ssu_Y0_r0_lab
ssu_ps_perfect_Y0_q0_adt$posthoc.method
data.frame(ssu_ps_perfect_Y0_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_r1_lab <- "Y0 PERFect (Shannon exponential)"
ssu_Y0_r1_lab
ssu_ps_perfect_Y0_q1_adt$posthoc.method
data.frame(ssu_ps_perfect_Y0_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_r2_lab <- "Y0 PERFect (Inverse Simpson)"
ssu_Y0_r2_lab
ssu_ps_perfect_Y0_q2_adt$posthoc.method
data.frame(ssu_ps_perfect_Y0_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PIME</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_p0_lab <- "Y0 PIME (Observed)"
ssu_Y0_p0_lab
ssu_ps_pime_Y0_q0_adt$posthoc.method
data.frame(ssu_ps_pime_Y0_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_p1_lab <- "Y0 PIME (Shannon exponential)"
ssu_Y0_p1_lab
ssu_ps_pime_Y0_q1_adt$posthoc.method
data.frame(ssu_ps_pime_Y0_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y0_p2_lab <- "Y0 PIME (Inverse Simpson)"
ssu_Y0_p2_lab
ssu_ps_pime_Y0_q2_adt$posthoc.method
data.frame(ssu_ps_pime_Y0_q2_adt$posthoc)
```
</details>
:::

::: {.panel}

#### Year 1

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Full</strong> data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_f0_lab <- "Y1 FULL (Observed)"
ssu_Y1_f0_lab
ssu_ps_work_Y1_q0_adt$posthoc.method
data.frame(ssu_ps_work_Y1_q0_adt$posthoc)
objects()
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_f1_lab <- "Y1 FULL (Shannon exponential)"
ssu_Y1_f1_lab
ssu_ps_work_Y1_q1_adt$posthoc.method
data.frame(ssu_ps_work_Y1_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_f2_lab <- "Y1 FULL (Inverse Simpson)"
ssu_Y1_f2_lab
ssu_ps_work_Y1_q2_adt$posthoc.method
data.frame(ssu_ps_work_Y1_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Arbitrary</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_l0_lab <- "Y1 FILT (Observed)"
ssu_Y1_l0_lab
ssu_ps_filt_Y1_q0_adt$posthoc.method
data.frame(ssu_ps_filt_Y1_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_l1_lab <- "Y1 FILT (Shannon exponential)"
ssu_Y1_l1_lab
ssu_ps_filt_Y1_q1_adt$posthoc.method
data.frame(ssu_ps_filt_Y1_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_l2_lab <- "Y1 FILT (Inverse Simpson)"
ssu_Y1_l2_lab
ssu_ps_filt_Y1_q2_adt$posthoc.method
data.frame(ssu_ps_filt_Y1_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PERFect</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_r0_lab <- "Y1 PERFect (Observed)"
ssu_Y1_r0_lab
ssu_ps_perfect_Y1_q0_adt$posthoc.method
data.frame(ssu_ps_perfect_Y1_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_r1_lab <- "Y1 PERFect (Shannon exponential)"
ssu_Y1_r1_lab
ssu_ps_perfect_Y1_q1_adt$posthoc.method
data.frame(ssu_ps_perfect_Y1_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_r2_lab <- "Y1 PERFect (Inverse Simpson)"
ssu_Y1_r2_lab
ssu_ps_perfect_Y1_q2_adt$posthoc.method
data.frame(ssu_ps_perfect_Y1_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PIME</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_p0_lab <- "Y1 PIME (Observed)"
ssu_Y1_p0_lab
ssu_ps_pime_Y1_q0_adt$posthoc.method
data.frame(ssu_ps_pime_Y1_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_p1_lab <- "Y1 PIME (Shannon exponential)"
ssu_Y1_p1_lab
ssu_ps_pime_Y1_q1_adt$posthoc.method
data.frame(ssu_ps_pime_Y1_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y1_p2_lab <- "Y1 PIME (Inverse Simpson)"
ssu_Y1_p2_lab
ssu_ps_pime_Y1_q2_adt$posthoc.method
data.frame(ssu_ps_pime_Y1_q2_adt$posthoc)
```
</details>
:::

::: {.panel}

#### Year 4

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Full</strong> data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_f0_lab <- "Y4 FULL (Observed)"
ssu_Y4_f0_lab
ssu_ps_work_Y4_q0_adt$posthoc.method
data.frame(ssu_ps_work_Y4_q0_adt$posthoc)
objects()
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_f1_lab <- "Y4 FULL (Shannon exponential)"
ssu_Y4_f1_lab
ssu_ps_work_Y4_q1_adt$posthoc.method
data.frame(ssu_ps_work_Y4_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_f2_lab <- "Y4 FULL (Inverse Simpson)"
ssu_Y4_f2_lab
ssu_ps_work_Y4_q2_adt$posthoc.method
data.frame(ssu_ps_work_Y4_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Arbitrary</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_l0_lab <- "Y4 FILT (Observed)"
ssu_Y4_l0_lab
ssu_ps_filt_Y4_q0_adt$posthoc.method
data.frame(ssu_ps_filt_Y4_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_l1_lab <- "Y4 FILT (Shannon exponential)"
ssu_Y4_l1_lab
ssu_ps_filt_Y4_q1_adt$posthoc.method
data.frame(ssu_ps_filt_Y4_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_l2_lab <- "Y4 FILT (Inverse Simpson)"
ssu_Y4_l2_lab
ssu_ps_filt_Y4_q2_adt$posthoc.method
data.frame(ssu_ps_filt_Y4_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PERFect</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_r0_lab <- "Y4 PERFect (Observed)"
ssu_Y4_r0_lab
ssu_ps_perfect_Y4_q0_adt$posthoc.method
data.frame(ssu_ps_perfect_Y4_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_r1_lab <- "Y4 PERFect (Shannon exponential)"
ssu_Y4_r1_lab
ssu_ps_perfect_Y4_q1_adt$posthoc.method
data.frame(ssu_ps_perfect_Y4_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_r2_lab <- "Y4 PERFect (Inverse Simpson)"
ssu_Y4_r2_lab
ssu_ps_perfect_Y4_q2_adt$posthoc.method
data.frame(ssu_ps_perfect_Y4_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PIME</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_p0_lab <- "Y4 PIME (Observed)"
ssu_Y4_p0_lab
ssu_ps_pime_Y4_q0_adt$posthoc.method
data.frame(ssu_ps_pime_Y4_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_p1_lab <- "Y4 PIME (Shannon exponential)"
ssu_Y4_p1_lab
ssu_ps_pime_Y4_q1_adt$posthoc.method
data.frame(ssu_ps_pime_Y4_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
ssu_Y4_p2_lab <- "Y4 PIME (Inverse Simpson)"
ssu_Y4_p2_lab
ssu_ps_pime_Y4_q2_adt$posthoc.method
data.frame(ssu_ps_pime_Y4_q2_adt$posthoc)
```
</details>
:::
:::
:::


## Alpha Diversity Plots

Now we can plot the results from the posthoc analyses for each metric and data set using the function `div_test_plot_jjs`. I modified the original function (`div_test_plot`) to control a little of the formatting.

The command is as follows:

```
div_test_plot(divtest = x, chart = "type", colour = col.pal, 
              posthoc = TRUE, threshold = value))
```

Where `x` is the results from the `div_test` function, `"type"` is chart type (box, jitter, or violin), `colour` is is a color palette, `posthoc` indicates whether to run posthoc pairwise analyses, and `value` is the maximum p-value to show in pairwise posthoc results. **WARNING** if none of the posthoc results are below the specified threshold, the function will throw an error. Therefore, until this is fixed, all posthoc values are shown.


```{r, code_folding=TRUE}
agua_col <- c("#B26322", "#29B222", "#2271B2", "#AB22B2")

source("hack_code/div_test_plot_jjs.R")
tmp_order <- c('virginica', 'versicolor', 'setosa') 
rm(list = ls(pattern="_adt_plot"))
for (i in objects(pattern="_adt")) {
     tmp_name <- purrr::map_chr(i, ~ paste0(., "_plot"))
     tmp_get <- get(i)
     tmp_ds_year <- stringr::str_extract(i, "Y[0-9]") 
     tmp_order <- c(
       paste(tmp_ds_year, "CTL", sep = "_"), 
       paste(tmp_ds_year, "N", sep = "_"),
       paste(tmp_ds_year, "P", sep = "_"),
       paste(tmp_ds_year, "NP", sep = "_")
       ) 
     tmp_get$data <- tmp_get$data %>% 
                     mutate(Group = factor(Group, levels = tmp_order)) %>% 
                     arrange(Group)  
     tmp_df <- div_test_plot_jjs(tmp_get, chart = "box", 
                             posthoc = TRUE, 
                             #threshold = 0.05,
                             colour = agua_col)
     tmp_df <- ggpar(tmp_df, legend = "none")
     #print(tmp_name)
     assign(tmp_name, tmp_df)
     rm(list = ls(pattern = "tmp_"))
}
```

```{r, echo=FALSE}
div_plot_theme <- list(theme(
  plot.title = element_text(size = 10, face = "bold"), 
  plot.title.position = "plot",
  axis.text.x = element_text(size = 8, angle = 0, hjust = 0.5), 
  axis.text.y = element_text(size = 8),
  )
)
new_labs <- c("Y4_CTL" = "CTL", "Y4_N" = "N", "Y4_P" = "P", "Y4_NP" = "NP")
```

```{r, echo=FALSE}
ssu_ps_work_Y0_q0_adt_plot <- ssu_ps_work_Y0_q0_adt_plot +
                              ggtitle(ssu_Y0_f0_lab) + 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
ssu_ps_work_Y0_q1_adt_plot <- ssu_ps_work_Y0_q1_adt_plot +
                              ggtitle(ssu_Y0_f1_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
ssu_ps_work_Y0_q2_adt_plot <- ssu_ps_work_Y0_q2_adt_plot +
                              ggtitle(ssu_Y0_f2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

ssu_ps_work_Y1_q0_adt_plot <- ssu_ps_work_Y1_q0_adt_plot +
                              ggtitle(ssu_Y1_f0_lab)+ 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
ssu_ps_work_Y1_q1_adt_plot <- ssu_ps_work_Y1_q1_adt_plot +
                              ggtitle(ssu_Y1_f1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
ssu_ps_work_Y1_q2_adt_plot <- ssu_ps_work_Y1_q2_adt_plot +
                              ggtitle(ssu_Y1_f2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

ssu_ps_work_Y4_q0_adt_plot <- ssu_ps_work_Y4_q0_adt_plot +
                              ggtitle(ssu_Y4_f0_lab)+ 
                              div_plot_theme + 
                              scale_x_discrete(labels = new_labs)
ssu_ps_work_Y4_q1_adt_plot <- ssu_ps_work_Y4_q1_adt_plot +
                              ggtitle(ssu_Y4_f1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)

ssu_ps_work_Y4_q2_adt_plot <- ssu_ps_work_Y4_q2_adt_plot +
                              ggtitle(ssu_Y4_f2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)
```

```{r, echo=FALSE}
ssu_ps_filt_Y0_q0_adt_plot <- ssu_ps_filt_Y0_q0_adt_plot +
                              ggtitle(ssu_Y0_l0_lab) + 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
ssu_ps_filt_Y0_q1_adt_plot <- ssu_ps_filt_Y0_q1_adt_plot +
                              ggtitle(ssu_Y0_l1_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
ssu_ps_filt_Y0_q2_adt_plot <- ssu_ps_filt_Y0_q2_adt_plot +
                              ggtitle(ssu_Y0_l2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

ssu_ps_filt_Y1_q0_adt_plot <- ssu_ps_filt_Y1_q0_adt_plot +
                              ggtitle(ssu_Y1_l0_lab)+ 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
ssu_ps_filt_Y1_q1_adt_plot <- ssu_ps_filt_Y1_q1_adt_plot +
                              ggtitle(ssu_Y1_l1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
ssu_ps_filt_Y1_q2_adt_plot <- ssu_ps_filt_Y1_q2_adt_plot +
                              ggtitle(ssu_Y1_l2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

ssu_ps_filt_Y4_q0_adt_plot <- ssu_ps_filt_Y4_q0_adt_plot +
                              ggtitle(ssu_Y4_l0_lab)+ 
                              div_plot_theme + 
                              scale_x_discrete(labels = new_labs)
ssu_ps_filt_Y4_q1_adt_plot <- ssu_ps_filt_Y4_q1_adt_plot +
                              ggtitle(ssu_Y4_l1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)

ssu_ps_filt_Y4_q2_adt_plot <- ssu_ps_filt_Y4_q2_adt_plot +
                              ggtitle(ssu_Y4_l2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)
```

```{r, echo=FALSE}
ssu_ps_perfect_Y0_q0_adt_plot <- ssu_ps_perfect_Y0_q0_adt_plot +
                              ggtitle(ssu_Y0_r0_lab) + 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
ssu_ps_perfect_Y0_q1_adt_plot <- ssu_ps_perfect_Y0_q1_adt_plot +
                              ggtitle(ssu_Y0_r1_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
ssu_ps_perfect_Y0_q2_adt_plot <- ssu_ps_perfect_Y0_q2_adt_plot +
                              ggtitle(ssu_Y0_r2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

ssu_ps_perfect_Y1_q0_adt_plot <- ssu_ps_perfect_Y1_q0_adt_plot +
                              ggtitle(ssu_Y1_r0_lab)+ 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
ssu_ps_perfect_Y1_q1_adt_plot <- ssu_ps_perfect_Y1_q1_adt_plot +
                              ggtitle(ssu_Y1_r1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
ssu_ps_perfect_Y1_q2_adt_plot <- ssu_ps_perfect_Y1_q2_adt_plot +
                              ggtitle(ssu_Y1_r2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

ssu_ps_perfect_Y4_q0_adt_plot <- ssu_ps_perfect_Y4_q0_adt_plot +
                              ggtitle(ssu_Y4_r0_lab)+ 
                              div_plot_theme + 
                              scale_x_discrete(labels = new_labs)
ssu_ps_perfect_Y4_q1_adt_plot <- ssu_ps_perfect_Y4_q1_adt_plot +
                              ggtitle(ssu_Y4_r1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)

ssu_ps_perfect_Y4_q2_adt_plot <- ssu_ps_perfect_Y4_q2_adt_plot +
                              ggtitle(ssu_Y4_r2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)
```

```{r, echo=FALSE}
ssu_ps_pime_Y0_q0_adt_plot <- ssu_ps_pime_Y0_q0_adt_plot +
                              ggtitle(ssu_Y0_p0_lab) + 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
ssu_ps_pime_Y0_q1_adt_plot <- ssu_ps_pime_Y0_q1_adt_plot +
                              ggtitle(ssu_Y0_p1_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
ssu_ps_pime_Y0_q2_adt_plot <- ssu_ps_pime_Y0_q2_adt_plot +
                              ggtitle(ssu_Y0_p2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

ssu_ps_pime_Y1_q0_adt_plot <- ssu_ps_pime_Y1_q0_adt_plot +
                              ggtitle(ssu_Y1_p0_lab)+ 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
ssu_ps_pime_Y1_q1_adt_plot <- ssu_ps_pime_Y1_q1_adt_plot +
                              ggtitle(ssu_Y1_p1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
ssu_ps_pime_Y1_q2_adt_plot <- ssu_ps_pime_Y1_q2_adt_plot +
                              ggtitle(ssu_Y1_p2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

ssu_ps_pime_Y4_q0_adt_plot <- ssu_ps_pime_Y4_q0_adt_plot +
                              ggtitle(ssu_Y4_p0_lab)+ 
                              div_plot_theme + 
                              scale_x_discrete(labels = new_labs)
ssu_ps_pime_Y4_q1_adt_plot <- ssu_ps_pime_Y4_q1_adt_plot +
                              ggtitle(ssu_Y4_p1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)

ssu_ps_pime_Y4_q2_adt_plot <- ssu_ps_pime_Y4_q2_adt_plot +
                              ggtitle(ssu_Y4_p2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)
```


```{r, echo=FALSE}
ssu_alph_div_plots_work <- ggarrange(
           ssu_ps_work_Y0_q0_adt_plot, 
           ssu_ps_work_Y0_q1_adt_plot, 
           ssu_ps_work_Y0_q2_adt_plot, 
           ssu_ps_work_Y1_q0_adt_plot, 
           ssu_ps_work_Y1_q1_adt_plot, 
           ssu_ps_work_Y1_q2_adt_plot, 
           ssu_ps_work_Y4_q0_adt_plot, 
           ssu_ps_work_Y4_q1_adt_plot, 
           ssu_ps_work_Y4_q2_adt_plot, 
           ncol = 3, nrow = 3
           )

ssu_alph_div_plots_work
```


```{r, echo=FALSE}
ssu_alph_div_plots_filt <- ggarrange(
           ssu_ps_filt_Y0_q0_adt_plot, 
           ssu_ps_filt_Y0_q1_adt_plot, 
           ssu_ps_filt_Y0_q2_adt_plot, 
           ssu_ps_filt_Y1_q0_adt_plot, 
           ssu_ps_filt_Y1_q1_adt_plot, 
           ssu_ps_filt_Y1_q2_adt_plot, 
           ssu_ps_filt_Y4_q0_adt_plot, 
           ssu_ps_filt_Y4_q1_adt_plot, 
           ssu_ps_filt_Y4_q2_adt_plot, 
           ncol = 3, nrow = 3)
ssu_alph_div_plots_filt
```


```{r, echo=FALSE}
ssu_alph_div_plots_perfect <- ggarrange(
           ssu_ps_perfect_Y0_q0_adt_plot, 
           ssu_ps_perfect_Y0_q1_adt_plot, 
           ssu_ps_perfect_Y0_q2_adt_plot, 
           ssu_ps_perfect_Y1_q0_adt_plot, 
           ssu_ps_perfect_Y1_q1_adt_plot, 
           ssu_ps_perfect_Y1_q2_adt_plot, 
           ssu_ps_perfect_Y4_q0_adt_plot, 
           ssu_ps_perfect_Y4_q1_adt_plot, 
           ssu_ps_perfect_Y4_q2_adt_plot, 
           ncol = 3, nrow = 3)
ssu_alph_div_plots_perfect

```

```{r, echo=FALSE}
ssu_alph_div_plots_pime <- ggarrange(
           ssu_ps_pime_Y0_q0_adt_plot, 
           ssu_ps_pime_Y0_q1_adt_plot, 
           ssu_ps_pime_Y0_q2_adt_plot, 
           ssu_ps_pime_Y1_q0_adt_plot, 
           ssu_ps_pime_Y1_q1_adt_plot, 
           ssu_ps_pime_Y1_q2_adt_plot, 
           ssu_ps_pime_Y4_q0_adt_plot, 
           ssu_ps_pime_Y4_q1_adt_plot, 
           ssu_ps_pime_Y4_q2_adt_plot, 
           ncol = 3, nrow = 3
           )
ssu_alph_div_plots_pime
```


```{r, echo=FALSE}
plots_to_save <- c("ssu_alph_div_plots_work", "ssu_alph_div_plots_filt", 
                   "ssu_alph_div_plots_perfect", "ssu_alph_div_plots_pime")

for (i in plots_to_save) {
  tmp_get <- get(i)
  ggplot2::ggsave(tmp_get, file = paste0("files/alpha/figures/", i, ".png", sep = ""), 
                   height = 20, width = 28, units = 'cm', bg = "white")
  ggplot2::ggsave(tmp_get, file = paste0("files/alpha/figures/", i, ".pdf", sep = ""), 
                   height = 20, width = 28, units = 'cm', bg = "white")
  rm(list = ls(pattern = "tmp_"))
}

system("cp files/alpha/figures/ssu_alph_div_plots_work.png include/alpha/ssu_alph_div_plots_work.png")
system("cp files/alpha/figures/ssu_alph_div_plots_filt.png include/alpha/ssu_alph_div_plots_filt.png")
system("cp files/alpha/figures/ssu_alph_div_plots_perfect.png include/alpha/ssu_alph_div_plots_perfect.png")
system("cp files/alpha/figures/ssu_alph_div_plots_pime.png include/alpha/ssu_alph_div_plots_pime.png")
```


> Posthoc adjusted p-values given for each significant pairwise comparison.

::: l-page
::: {.panelset}
::: {.panel}

### FULL data set

```{r, echo=FALSE, warning=FALSE, fig.height=5, layout='l-page', eval=TRUE}
knitr::include_graphics("include/alpha/ssu_alph_div_plots_work.png")
```
<small>`r caption_fig_ssu("ssu_alpha_div_plots_full")`</small>
:::

::: {.panel}

### Arbitrary filtered

```{r, echo=FALSE, warning=FALSE, fig.height=5, layout='l-page', eval=TRUE}
knitr::include_graphics("include/alpha/ssu_alph_div_plots_filt.png")
```
<small>`r caption_fig_ssu("ssu_alpha_div_plots_filt")`</small>
:::

::: {.panel}

### PERFect filtered

```{r, echo=FALSE, warning=FALSE, fig.height=5, layout='l-page', eval=TRUE}
knitr::include_graphics("include/alpha/ssu_alph_div_plots_perfect.png")
```
<small>`r caption_fig_ssu("ssu_alpha_div_plots_perfect")`</small>

:::

::: {.panel}

### PIME filtered

```{r, echo=FALSE, warning=FALSE, fig.height=5, layout='l-page', eval=TRUE}
knitr::include_graphics("include/alpha/ssu_alph_div_plots_pime.png")
```
<small>`r caption_fig_ssu("ssu_alpha_div_plots_pime")`</small>
:::
:::
:::


## Posthoc sample comparisons

```{r, echo=FALSE}
rm(list = ls(pattern = "_reform"))
for (i in objects(pattern = "_adt$")) {
  tmp_get <- get(i)$posthoc
  tmp_form <- if(is.data.frame(tmp_get)){
            tmp_form <- tmp_get[ -c(1:2) ]
            tmp_form <- tmp_form %>% tibble::rownames_to_column("Comparison") %>%
              dplyr::rename("p_value" = 2) #%>%
              #tidyr::separate(col = Comparison, into = c("S1", "S2"), sep = "-")
          } else {
            tmp_form <- data.frame(tmp_get)
            tmp_form <- tmp_form[ -c(1:3) ]
            tmp_form <- tmp_form %>% tibble::rownames_to_column("Comparison") %>%
              dplyr::rename("p_value" = 2) %>%
              tidyr::separate(col = Comparison, into = c("S2", "S1"), sep = "-")
            tmp_form <- tmp_form[, c(2, 1, 3)]
            tmp_form <- tmp_form %>% tidyr::unite("Comparison", 1:2, sep = "-", 
                                                  remove = TRUE, na.rm = FALSE)
          }
#  tmp_form <- tmp_form %>% tibble::rownames_to_column("Comparison") %>%
#    dplyr::rename("p_value" = 2)
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_reform"))
  assign(tmp_name, tmp_form)
  rm(list = ls(pattern = "tmp_"))
}
objects(pattern = "_reform")
```


```{r, echo=FALSE}
ssu_ps_work_Y0_hill_sum <- dplyr::left_join(ssu_ps_work_Y0_q0_adt_reform, ssu_ps_work_Y0_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_work_Y0_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y0" = 3, "Shannon_exp_Y0" = 4, "InvSimpson_Y0" = 5)

ssu_ps_filt_Y0_hill_sum <- dplyr::left_join(ssu_ps_filt_Y0_q0_adt_reform, ssu_ps_filt_Y0_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_filt_Y0_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y0" = 3, "Shannon_exp_Y0" = 4, "InvSimpson_Y0" = 5)
ssu_ps_perfect_Y0_hill_sum <- dplyr::left_join(ssu_ps_perfect_Y0_q0_adt_reform, ssu_ps_perfect_Y0_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_perfect_Y0_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y0" = 3, "Shannon_exp_Y0" = 4, "InvSimpson_Y0" = 5)

ssu_ps_pime_Y0_hill_sum <- dplyr::left_join(ssu_ps_pime_Y0_q0_adt_reform, ssu_ps_pime_Y0_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_pime_Y0_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y0" = 3, "Shannon_exp_Y0" = 4, "InvSimpson_Y0" = 5)

```

```{r, echo=FALSE}
ssu_ps_work_Y1_hill_sum <- dplyr::left_join(ssu_ps_work_Y1_q0_adt_reform, ssu_ps_work_Y1_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_work_Y1_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y1" = 3, "Shannon_exp_Y1" = 4, "InvSimpson_Y1" = 5)

ssu_ps_filt_Y1_hill_sum <- dplyr::left_join(ssu_ps_filt_Y1_q0_adt_reform, ssu_ps_filt_Y1_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_filt_Y1_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y1" = 3, "Shannon_exp_Y1" = 4, "InvSimpson_Y1" = 5)
ssu_ps_perfect_Y1_hill_sum <- dplyr::left_join(ssu_ps_perfect_Y1_q0_adt_reform, ssu_ps_perfect_Y1_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_perfect_Y1_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y1" = 3, "Shannon_exp_Y1" = 4, "InvSimpson_Y1" = 5)

ssu_ps_pime_Y1_hill_sum <- dplyr::left_join(ssu_ps_pime_Y1_q0_adt_reform, ssu_ps_pime_Y1_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_pime_Y1_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y1" = 3, "Shannon_exp_Y1" = 4, "InvSimpson_Y1" = 5)
```

```{r, echo=FALSE}
ssu_ps_work_Y4_hill_sum <- dplyr::left_join(ssu_ps_work_Y4_q0_adt_reform, ssu_ps_work_Y4_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_work_Y4_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y4" = 3, "Shannon_exp_Y4" = 4, "InvSimpson_Y4" = 5)

ssu_ps_filt_Y4_hill_sum <- dplyr::left_join(ssu_ps_filt_Y4_q0_adt_reform, ssu_ps_filt_Y4_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_filt_Y4_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y4" = 3, "Shannon_exp_Y4" = 4, "InvSimpson_Y4" = 5)
ssu_ps_perfect_Y4_hill_sum <- dplyr::left_join(ssu_ps_perfect_Y4_q0_adt_reform, ssu_ps_perfect_Y4_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_perfect_Y4_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y4" = 3, "Shannon_exp_Y4" = 4, "InvSimpson_Y4" = 5)

ssu_ps_pime_Y4_hill_sum <- dplyr::left_join(ssu_ps_pime_Y4_q0_adt_reform, ssu_ps_pime_Y4_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., ssu_ps_pime_Y4_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y4" = 3, "Shannon_exp_Y4" = 4, "InvSimpson_Y4" = 5)

```


```{r, echo=FALSE}
## REMOVE YEAR
for (i in objects(pattern = "_hill_sum")) {
  tmp_get <- get(i)
  tmp_get$Group_1 <- tmp_get$Group_1 %>% str_remove_all(., "Y[0-9]_")
  tmp_get$Group_2 <- tmp_get$Group_2 %>% str_remove_all(., "Y[0-9]_")
  assign(i, tmp_get)
  rm(list = ls(pattern = "tmp_"))
}
```


```{r, echo=FALSE}
col_order <- c("Group_1", "Group_2", 
               "Observed_Y0", "Observed_Y1", "Observed_Y4", 
               "Shannon_exp_Y0", "Shannon_exp_Y1", "Shannon_exp_Y4", 
               "InvSimpson_Y0", "InvSimpson_Y1", "InvSimpson_Y4")
ssu_alpha_hill_work_posthoc <- dplyr::left_join(ssu_ps_work_Y0_hill_sum, ssu_ps_work_Y1_hill_sum, by = c("Group_1", "Group_2")) %>%
                             dplyr::left_join(., ssu_ps_work_Y4_hill_sum, by = c("Group_1", "Group_2"))
ssu_alpha_hill_work_posthoc <- ssu_alpha_hill_work_posthoc[, col_order]

ssu_alpha_hill_filt_posthoc <- dplyr::left_join(ssu_ps_filt_Y0_hill_sum, ssu_ps_filt_Y1_hill_sum, by = c("Group_1", "Group_2")) %>%
                             dplyr::left_join(., ssu_ps_filt_Y4_hill_sum, by = c("Group_1", "Group_2"))
ssu_alpha_hill_filt_posthoc <- ssu_alpha_hill_filt_posthoc[, col_order]

ssu_alpha_hill_perfect_posthoc <- dplyr::left_join(ssu_ps_perfect_Y0_hill_sum, ssu_ps_perfect_Y1_hill_sum, by = c("Group_1", "Group_2")) %>%
                             dplyr::left_join(., ssu_ps_perfect_Y4_hill_sum, by = c("Group_1", "Group_2"))
ssu_alpha_hill_perfect_posthoc <- ssu_alpha_hill_perfect_posthoc[, col_order]

ssu_alpha_hill_pime_posthoc <- dplyr::left_join(ssu_ps_pime_Y0_hill_sum, ssu_ps_pime_Y1_hill_sum, by = c("Group_1", "Group_2")) %>%
                             dplyr::left_join(., ssu_ps_pime_Y4_hill_sum, by = c("Group_1", "Group_2"))
ssu_alpha_hill_pime_posthoc <- ssu_alpha_hill_pime_posthoc[, col_order]

```

::: l-page
::: {.panelset}
::: {.panel}

#### FULL data set

<small>`r caption_tab_ssu("ssu_alpha_hill_work_posthoc")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_alpha_hill_work_posthoc)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
poshoc_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}

#### Arbitrary filtered

<small>`r caption_tab_ssu("ssu_alpha_hill_filt_posthoc")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_alpha_hill_filt_posthoc)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
poshoc_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}

#### PERFect filtered

<small>`r caption_tab_ssu("ssu_alpha_hill_perfect_posthoc")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_alpha_hill_perfect_posthoc)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
poshoc_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### PIME filtered

<small>`r caption_tab_ssu("ssu_alpha_hill_pime_posthoc")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(ssu_alpha_hill_pime_posthoc)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
poshoc_table(seq_table)
rm(seq_table)
```
:::
:::
:::

# ITS

```{r, include=FALSE, eval=TRUE}
## Load to build page only #2
#gdata::keep(caption_tab_ssu, caption_fig_ssu, caption_tab_its, caption_fig_its, ref, download_this_fun, sure = TRUE)
remove(list = ls())
load("page_build/alpha_wf_its.rdata")
```

```{r, echo = FALSE, eval = TRUE}
### COmmon formatting scripts
### NOTE: captioner.R must be read BEFORE captions_XYZ.R
source("assets/captioner/captioner.R")
source("assets/captioner/captions/captions_alpha.R")
source("assets/reactable/download_this_fun.R")
source("assets/reactable/styles.R")
source("assets/reactable/table_functions/alpha.R")
```


```{r, include=FALSE}
## Initial Load for  ANALYSIS #1
remove(list = ls())
set.seed(119)
its_ps_work <- readRDS("files/filtering/working_ps/its_ps_work.rds")
its_ps_filt <- readRDS("files/filtering/working_ps/its_ps_filt.rds")
its_ps_perfect <- readRDS("files/filtering/working_ps/its_ps_perfect.rds")
its_ps_pime <- readRDS("files/filtering/working_ps/its_ps_pime.rds")
```

To account for presence of rare sequence variants caused by sequencing errors or other technical artifacts, we use Hill numbers [@alberdi2019guide]. Hill numbers allow the weight put on rare versus abundant sequence variants to be scaled while providing intuitive comparisons of diversity levels using “effective number of ASVs” as a measuring unit. This approach allows for balancing the over representation of rare ASVs that might be inflated due to sequencing errors.

We will then use Shapiro-Wilk tests to test for normalcy and then, depending on the results, either use  parametric ANOVA or non-parametric Kruskal-Wallis to compare alpha diversity among treatments.

```{r, echo=FALSE}
its_ps_work@phy_tree <- NULL
its_ps_filt@phy_tree <- NULL
its_ps_pime@phy_tree <- NULL
its_ps_perfect@phy_tree <- NULL
```

Before we begin the analysis, let's subset each data set by Year and take a look at some summary data.

```{r, code_folding=TRUE}
its_alpha_ds <- c("its_ps_work", "its_ps_filt", "its_ps_perfect", "its_ps_pime")
for (i in its_alpha_ds) {
  tmp_get <- get(i)
  tmp_samples <- row.names(data.frame(otu_table(tmp_get)))
  tmp_y0_samps <- stringr::str_subset(tmp_samples, "_Y0_")
  tmp_y1_samps <- stringr::str_subset(tmp_samples, "_Y1_")
  tmp_y4_samps <- stringr::str_subset(tmp_samples, "_Y4_")

  tmp_y0 <- prune_samples(tmp_y0_samps, tmp_get) 
  tmp_y1 <- prune_samples(tmp_y1_samps, tmp_get)
  tmp_y4 <- prune_samples(tmp_y4_samps, tmp_get)
  
  tmp_y0 <- prune_taxa(taxa_sums(tmp_y0) > 0, tmp_y0)
  tmp_y1 <- prune_taxa(taxa_sums(tmp_y1) > 0, tmp_y1)
  tmp_y4 <- prune_taxa(taxa_sums(tmp_y4) > 0, tmp_y4)
  
  tmp_ps_name_y0 <- purrr::map_chr(i, ~ paste0(., "_Y0"))
  assign(tmp_ps_name_y0, tmp_y0)
  
  tmp_ps_name_y1 <- purrr::map_chr(i, ~ paste0(., "_Y1"))
  assign(tmp_ps_name_y1, tmp_y1)
  
  tmp_ps_name_y4 <- purrr::map_chr(i, ~ paste0(., "_Y4"))
  assign(tmp_ps_name_y4, tmp_y4)
  
  tmp_path <- file.path("files/alpha/rdata/")
  saveRDS(tmp_y0, paste(tmp_path, tmp_ps_name_y0, ".rds", sep = ""))
  saveRDS(tmp_y1, paste(tmp_path, tmp_ps_name_y1, ".rds", sep = ""))
  saveRDS(tmp_y4, paste(tmp_path, tmp_ps_name_y4, ".rds", sep = ""))
  rm(list = ls(pattern = "tmp_"))
}
```

```{r, echo=FALSE}
tmp_objects <- data.frame(c("FULL data set (Y0)", 
                            "FULL data set (Y1)", 
                            "FULL data set (Y4)", 
                            "Arbitrary filter data (Y0)", 
                            "Arbitrary filter data (Y1)", 
                            "Arbitrary filter data (Y4)", 
                            "PERfect filtered data (Y0)",
                            "PERfect filtered data (Y1)",
                            "PERfect filtered data (Y4)",
                            "PIME (Y0)",
                            "PIME (Y1)", 
                            "PIME (Y4)"))
tmp_samples <- c("its_ps_work_Y0", "its_ps_work_Y1", "its_ps_work_Y4",
                 "its_ps_filt_Y0", "its_ps_filt_Y1", "its_ps_filt_Y4",
                 "its_ps_perfect_Y0", "its_ps_perfect_Y1", "its_ps_perfect_Y4",
                 "its_ps_pime_Y0", "its_ps_pime_Y1", "its_ps_pime_Y4")
tmp_no_samp <- c()
for (i in tmp_samples) {
   tmp_get <- nsamples(get(i))
   tmp_no_samp <- c(append(tmp_no_samp, tmp_get))
}
tmp_no_samp <- data.frame(tmp_no_samp)

tmp_rc <- c()
for (i in tmp_samples) {
   tmp_get <- sum(readcount(get(i)))
   tmp_rc <- c(append(tmp_rc, tmp_get))
}
tmp_rc <- data.frame(tmp_rc)
tmp_asv <- c()
for (i in tmp_samples) {
   tmp_get <- ntaxa(get(i))
   tmp_asv <- c(append(tmp_asv, tmp_get))
}
tmp_asv <- data.frame(tmp_asv)

its_split_summary <- dplyr::bind_cols(tmp_objects, tmp_samples) %>%
                      dplyr::bind_cols(., tmp_no_samp) %>%
                      dplyr::bind_cols(., tmp_rc) %>%
                      dplyr::bind_cols(., tmp_asv) %>%
  dplyr::rename("Description" = 1, "object name" = 2, "no. samples" = 3,
                "total reads" = 4, "total asvs" = 5)
rm(list = ls(pattern = "tmp_"))

write.table(its_split_summary, "files/alpha/tables/its_split_by_year.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r, echo=FALSE}
its_alpha_ds <- c("its_ps_work", "its_ps_filt", "its_ps_perfect", "its_ps_pime")

tmp_objects <- data.frame(c("FULL data set", 
                            "Arbitrary filter", 
                            "PERfect filtered",
                            "PIME filtered"))
tmp_samples <- c("its_ps_work",
                 "its_ps_filt",
                 "its_ps_perfect",
                 "its_ps_pime")
tmp_no_samp <- c()
for (i in tmp_samples) {
   tmp_get <- nsamples(get(i))
   tmp_no_samp <- c(append(tmp_no_samp, tmp_get))
}
tmp_no_samp <- data.frame(tmp_no_samp)

tmp_rc <- c()
for (i in tmp_samples) {
   tmp_get <- sum(readcount(get(i)))
   tmp_rc <- c(append(tmp_rc, tmp_get))
}
tmp_rc <- data.frame(tmp_rc)
tmp_asv <- c()
for (i in tmp_samples) {
   tmp_get <- ntaxa(get(i))
   tmp_asv <- c(append(tmp_asv, tmp_get))
}
tmp_asv <- data.frame(tmp_asv)

its_filter_summary <- dplyr::bind_cols(tmp_objects, tmp_samples) %>%
                      dplyr::bind_cols(., tmp_no_samp) %>%
                      dplyr::bind_cols(., tmp_rc) %>%
                      dplyr::bind_cols(., tmp_asv) %>%
  dplyr::rename("Description" = 1, "object name" = 2, "no. samples" = 3,
                "total reads" = 4, "total asvs" = 5)
rm(list = ls(pattern = "tmp_"))

write.table(its_filter_summary, "files/alpha/tables/its_filter_summmary.txt", 
            sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r, echo=FALSE}
col_order <- c("Description", "object name", "Year", 
               "no. samples", "total reads", "total asvs")

tmp_all <- its_filter_summary
tmp_all$Year <- "All years"
tmp_all <- tmp_all[, col_order]

tmp_subset <- its_split_summary
tmp_subset$Year <- tmp_subset$`object name`

tmp_subset$Year <- gsub("^.*_Y", "Year ", tmp_subset$Year)
tmp_subset <- tmp_subset[, col_order]

its_dataset_summary <- rbind(tmp_all, tmp_subset)
its_dataset_summary <- its_dataset_summary %>% 
                       dplyr::slice(1,5:7,2,8:10,3,11:13,4,14:16)
rm(list = ls(pattern = "tmp_"))
```

<small>`r caption_tab_its("its_dataset_summary")`</small>

```{r eval=TRUE, echo=FALSE, layout="l-body-outset"}
download_this_fun(its_dataset_summary)
```

```{r, echo=FALSE, layout="l-body-outset", eval=TRUE}
ds_summ_table(seq_table)
rm(seq_table)
```


## Calculate Hill Numbers

To calculate Hill numbers, we use the R package `hilldiv` [@alberdi2019hilldiv]. We calculate three metrics that put more or less weight on common species:

1) Observed richness, where `q-value = 0`.
2) Shannon exponential, which weighs ASVs by their frequency,  where `q-value = 1`.
3) Simpson multiplicative inverse, which over weighs abundant ASVs,  where `q-value = 2`.

We perform each analysis against the Full (unfiltered) data set as well as the Arbitrary, PERFect, and PIME, filtered data sets using the function `hill_div`.

The command is as follows:

```
hill_div(count = x, qvalue = i, tree = ultrametric_tree)
``` 

where `x` is the sample by ASV table, `i` is the q-value corresponding to the metric of interest and `tree` is an ultrametric formatted phylogenetic tree if you choose to look at lineage, rather than ASV, diversity.


Next, transform all the data to relative abundance values, and compute new trees.

```{r, code_folding=TRUE}
its_alpha_ds <- c("its_ps_work_Y0", "its_ps_work_Y1", "its_ps_work_Y4", 
                  "its_ps_filt_Y0", "its_ps_filt_Y1", "its_ps_filt_Y4", 
                  "its_ps_perfect_Y0", "its_ps_perfect_Y1", "its_ps_perfect_Y4", 
                  "its_ps_pime_Y0", "its_ps_pime_Y1", "its_ps_pime_Y4")
for (i in its_alpha_ds) {
  tmp_ps <- transform_sample_counts(get(i), function(otu) otu/sum(otu))
  tmp_ps@phy_tree <- NULL
  tmp_ps_norm <- merge_phyloseq(tmp_ps, sample_data)
  tmp_asv <- data.frame(t(otu_table(tmp_ps_norm)))
  tmp_ps_name <- purrr::map_chr(i, ~ paste0(., "_norm"))
  assign(tmp_ps_name, tmp_ps_norm)
  tmp_asv_name <- purrr::map_chr(i, ~ paste0(., "_tu"))
  assign(tmp_asv_name, tmp_asv)
  rm(list = ls(pattern = "tmp_"))
}
```

Next, we run the analysis for all three metrics on each data sets.

```{r, results='hide', code_folding=TRUE}
qvalue <- c(0,1,2)
for (i in qvalue) {
  for (j in its_alpha_ds) {
     tmp_asv <- get(purrr::map_chr(j, ~ paste0(., "_tu")))
     tmp_df <- data.frame(hill_div(tmp_asv, qvalue = i))
     tmp_df <- tmp_df %>% dplyr::rename("tmp_name" = 1) %>%
                              tibble::rownames_to_column("SamName")
     tmp_name <- purrr::map_chr(j, ~ paste0(., "_h", i))
     print(tmp_name)
     assign(tmp_name, tmp_df)
     rm(list = ls(pattern = "tmp_"))
  }
}
```

Then we make summary tables to add back into each `ps` object.

```{r, code_folding=TRUE}
for (i in its_alpha_ds) {
     tmp_obs <- get(purrr::map_chr(i, ~ paste0(., "_h0")))
     tmp_sha <- get(purrr::map_chr(i, ~ paste0(., "_h1")))
     tmp_sim <- get(purrr::map_chr(i, ~ paste0(., "_h2")))
     tmp_hill <- dplyr::left_join(tmp_obs, tmp_sha, by = "SamName") %>%
       dplyr::left_join(., tmp_sim, by = "SamName") %>%
       dplyr::rename("Observed" = 2, "Shannon_exp" = 3, "InvSimpson" = 4)
     tmp_name <- purrr::map_chr(i, ~ paste0(., "_hill"))
     assign(tmp_name, tmp_hill)
     rm(list = ls(pattern = "tmp_"))
}
```

And then create the new objects with the diversity data.

```{r, code_folding=TRUE}
for (i in its_alpha_ds) {
     tmp_ps <- get(i)
     tmp_ps <- merge_phyloseq(tmp_ps, sample_data)
     tmp_hill <- get(purrr::map_chr(i, ~ paste0(., "_hill")))
     tmp_hill_samp <- dplyr::left_join(data.frame(sample_data(tmp_ps)),
                                         tmp_hill, by = "SamName")
     tmp_hill_samp$ID <- tmp_hill_samp$SamName
     tmp_hill_samp <- tmp_hill_samp %>%  tibble::column_to_rownames("ID")
     tmp_ps2 <- merge_phyloseq(otu_table(tmp_ps),
                                sample_data(tmp_hill_samp),
                                tax_table(tmp_ps)
                               )
     assign(i, tmp_ps2)
     tmp_path <- file.path("files/alpha/rdata/")
     saveRDS(tmp_ps2, paste(tmp_path, i, ".rds", sep = ""))
     rm(list = ls(pattern = "tmp_"))
}
```

## Hill Numbers 

Now we summarize the data for each sample against all three metrics. The table contains the results of ASV diversity estimates from the full data set and the three filtered data sets.

The suffix `_fi` indicates metrics for the Arbitrary data set.  The suffix `_pe` indicates metrics for the PERFect data set and the suffix `_pi` indicates metrics for the PIME data set.

```{r, echo=FALSE}
######################### FULL ASV
for (i in objects(pattern = "its_ps_work_Y[0-9]_hill")) {
  tmp_get <- get(i)
  tmp_samp <- tmp_get %>% dplyr::rename("Observed" = 2,"Shannon_exp" = 3, "InvSimpson" = 4)
  tmp_get_ps <- get(gsub("_hill", "", i, fixed = TRUE))
  tmp_samp <- dplyr::left_join(data.frame(sample_data(tmp_get_ps)), tmp_samp)
  tmp_samp <- tmp_samp[order(tmp_samp$YEAR, tmp_samp$TREAT),]
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_samp")) 
  assign(tmp_name, tmp_samp)  
  rm(list = ls(pattern = "tmp_"))
} 

for (i in objects(pattern = "its_ps_filt_Y[0-9]_hill")) {
  tmp_get <- get(i)
  tmp_samp <- tmp_get %>% dplyr::rename("Observed_fi" = 2,"Shannon_exp_fi" = 3, "InvSimpson_fi" = 4)
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_samp")) 
  assign(tmp_name, tmp_samp)  
  rm(list = ls(pattern = "tmp_"))
} 

for (i in objects(pattern = "its_ps_perfect_Y[0-9]_hill")) {
  tmp_get <- get(i)
  tmp_samp <- tmp_get %>% dplyr::rename("Observed_pe" = 2,"Shannon_exp_pe" = 3, "InvSimpson_pe" = 4)
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_samp")) 
  assign(tmp_name, tmp_samp)  
  rm(list = ls(pattern = "tmp_"))
} 

for (i in objects(pattern = "its_ps_pime_Y[0-9]_hill")) {
  tmp_get <- get(i)
  tmp_samp <- tmp_get %>% dplyr::rename("Observed_pi" = 2,"Shannon_exp_pi" = 3, "InvSimpson_pi" = 4)
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_samp")) 
  assign(tmp_name, tmp_samp)  
  rm(list = ls(pattern = "tmp_"))
} 
objects(pattern = "_samp")
```

```{r, echo=FALSE}
######################### ASV TABLE
year_select <- c("Y0", "Y1", "Y4")
for (i in year_select) {
  tmp_work <- get(paste0("its_ps_work_", i, "_hill_samp"))
  tmp_filt <- get(paste0("its_ps_filt_", i, "_hill_samp"))
  tmp_perfect <- get(paste0("its_ps_perfect_", i, "_hill_samp"))
  tmp_pime <- get(paste0("its_ps_pime_", i, "_hill_samp"))
  tmp_alpha_div<- dplyr::left_join(tmp_work, tmp_filt) %>%
                  dplyr::left_join(., tmp_perfect) %>%
                  dplyr::left_join(., tmp_pime) %>% 
         dplyr::relocate(c("Observed", "Observed_fi", "Observed_pe", "Observed_pi"), .after = "YR_TREAT") %>% 
         dplyr::relocate(c("Shannon_exp", "Shannon_exp_fi", "Shannon_exp_pe", "Shannon_exp_pi"), .after = "Observed_pi") %>%
         dplyr::rename("Sample_ID" = "SamName") 
  tmp_alpha_div <- tmp_alpha_div%>% dplyr::mutate(across(.cols = c(13:ncol(tmp_alpha_div)), 
                                                               round, digits = 1))
  tmp_name <- paste0("its_tab_alpha_div_", i, sep = "" ) 
  assign(tmp_name, tmp_alpha_div)  
  
  rm(list = ls(pattern = "tmp_"))
}
```

::: l-page
::: {.panelset}
::: {.panel}

#### Year 0

<small>`r caption_tab_its("its_tab_alpha_div_Y0")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_tab_alpha_div_Y0)
seq_table[3:4] <- NULL
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
hill_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### Year 1

<small>`r caption_tab_its("its_tab_alpha_div_Y1")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_tab_alpha_div_Y1)
seq_table[3:4] <- NULL
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
hill_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### Year 4

<small>`r caption_tab_its("its_tab_alpha_div_Y4")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_tab_alpha_div_Y4)
seq_table[3:4] <- NULL
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
hill_table(seq_table)
rm(seq_table)
```
:::
:::
:::

## Normality Tests

Before running significance tests, we need to know if data is normally distributed, which will tell use whether to use a parametric or non-parametric test. To test if the data are normally distributed, we use the Shapiro-Wilk Normality test and the Bartlett Test of Homogeneity of Variances.

If the p-values are both **not significant** (p > 0.05) from the tests, we accept the null hypothesis (that the results are normally distributed) and test for significance between samples using an ANOVA. If the p-values are both **significant** (p < 0.05), we reject the null hypothesis (that the results are normally distributed) and test for significance between samples using Kruskal-Wallis (non-parametric equivalent of ANOVA).

The commands are as follows:

`shapiro.test(x)`, where `x` is a numeric vector of alpha diversity  values from the sample data table.

`bartlett.test(Value ~ Group, data = df)` Where `Value` is the metric of interest, `Group` in the treatment to compare, and `df` is the data frame.  

First the Shapiro-Wilk Normality test.

```{r, code_folding=TRUE}
#rm(list = ls(pattern = "shap"))
shap_test <-  function(alpha_div_tab) {
  shap_deparse <- deparse(substitute(alpha_div_tab))
  shap_year <- gsub(".*_", "", shap_deparse)
  shap_test_list <- c()
  for (i in colnames(alpha_div_tab[,9:ncol(alpha_div_tab)])) {
      tmp_name <- purrr::map_chr(i, ~ paste0("its_shap_", shap_year, "_", .))
      shap_test_list <- append(shap_test_list, tmp_name)
      tmp_test <- eval(shapiro.test(alpha_div_tab[[i]]))
      tmp_test$data.name <- tmp_name
      assign(tmp_name, tmp_test, envir = parent.frame() )
      rm(list = ls(pattern = "tmp_"))
  }
}
```

```{r}
shap_test(its_tab_alpha_div_Y0)
shap_test(its_tab_alpha_div_Y1)
shap_test(its_tab_alpha_div_Y4)
objects(pattern = "its_shap_")
```

And then the Bartlett Test of Homogeneity of Variances. For the Bartlett Test, we need to define `g`, a vector or factor object giving the group for the corresponding elements of x. 

```{r, code_folding=TRUE}
bart_test <-  function(alpha_div_tab) {
  bart_g <- "YR_TREAT"
  bart_deparse <- deparse(substitute(alpha_div_tab))
  bart_year <- gsub(".*_", "", bart_deparse)
  bart_test_list <- c()
  for (i in colnames(alpha_div_tab[,9:ncol(alpha_div_tab)])) {
      tmp_name <- purrr::map_chr(i, ~ paste0("its_bart_", bart_year, "_", .))
      bart_test_list <- append(bart_test_list, tmp_name)
      tmp_test <- eval(bartlett.test(alpha_div_tab[[i]] ~ get(bart_g), data = alpha_div_tab))
      tmp_test$data.name <- tmp_name
      assign(tmp_name, tmp_test, envir = parent.frame() )
      rm(list = ls(pattern = "tmp_"))
  }
}
```

```{r}
bart_test(its_tab_alpha_div_Y0)
bart_test(its_tab_alpha_div_Y1)
bart_test(its_tab_alpha_div_Y4)
```

Here we see which Shapiro-Wilk Normality **and** Bartlett tests were significant and which were not. So wherever the value of both p-values in `> 0.05` we can use an ANOVA, otherwise we use Kruskal-Wallis.

```{r, code_folding=TRUE}
combo_tests <-  function(alpha_div_tab) {
  combo_deparse <- deparse(substitute(alpha_div_tab))
  combo_year <- gsub(".*_", "", combo_deparse)
  combo_results <- c()
  for (i in colnames(alpha_div_tab[,9:ncol(alpha_div_tab)])) {
     tmp_get_shap <- get(purrr::map_chr(i, ~ paste0("its_shap_", combo_year, "_", .)))
     tmp_shap_p <- round(tmp_get_shap[[2]], 4)
     tmp_get_bart <- get(purrr::map_chr(i, ~ paste0("its_bart_", combo_year, "_", .)))
     tmp_bart_p <- round(tmp_get_bart[[3]], 4)
     tmp_test <- if(tmp_shap_p > 0.05 & tmp_bart_p > 0.05 ){
          tmp_test <- "ANOVA"
        } else {
          tmp_test <- "Kruskal-Wallis"
        }
     tmp_df <- data.frame(i, tmp_shap_p, tmp_bart_p, tmp_test)
     combo_results <- dplyr::bind_rows(combo_results, tmp_df)
     
     rm(list = ls(pattern = "tmp_"))
  }
  combo_results[,1] <- gsub('_exp', '', combo_results[,1])
  combo_results <- combo_results %>% separate(col = 1, 
                                                into = c("metric", "dataset"), 
                                                sep = "_", remove = TRUE)
  combo_results <- combo_results %>% tidyr::replace_na(list(dataset = "FULL"))
  combo_results$dataset <- stringr::str_replace(combo_results$dataset, "^fi$", "FILT") %>%
                           stringr::str_replace(., "^pi$", "PIME") %>%
                           stringr::str_replace(., "^pe$", "PERFect")

  combo_results <- combo_results %>% 
    dplyr::rename(c("p-value (normality)" = 3, "p-value (homogeneity)" = 4, "method" = 5))
  
  norm_res_name <- paste0("its_norm_res_", combo_year)
  cat("object saved as ", norm_res_name)
  assign(norm_res_name, combo_results, envir = parent.frame() )
}
```

```{r}
combo_tests(its_tab_alpha_div_Y0)
combo_tests(its_tab_alpha_div_Y1)
combo_tests(its_tab_alpha_div_Y4)
```

::: l-page
::: {.panelset}
::: {.panel}

#### Year 0

<small>`r caption_tab_its("its_norm_res_Y0")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_norm_res_Y0)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
norm_res_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### Year 1

<small>`r caption_tab_its("its_norm_res_Y1")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_norm_res_Y1)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
norm_res_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### Year 4

<small>`r caption_tab_its("its_norm_res_Y4")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_norm_res_Y4)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
norm_res_table(seq_table)
rm(seq_table)
```
:::
:::
:::


## Significance Tests

To begin, we need to create a hierarchy variable; a two-column matrix specifying the relationship between samples (first column) and groups (second column).

```{r, code_folding=TRUE}
sig_group <- "YR_TREAT"
for (i in objects(pattern = "its_tab_alpha_div")) {
    tmp_hill_hier <- get(i)
    tmp_hill_hier <- tmp_hill_hier %>% dplyr::select("Sample_ID", sig_group) %>%
      tibble::remove_rownames()
    tmp_hill_hier <- tmp_hill_hier[order(tmp_hill_hier[[2]]), ]
    #tmp_hill_hier$YR_TREAT = paste(tmp_hill_hier$YR_TREAT, 'C', sep='')
    tmp_hill_hier <- tmp_hill_hier %>% tibble::remove_rownames()
    tmp_yr <- gsub(".*_", "", i)
    tmp_name <- paste("its_hill_hier", tmp_yr, sep = "_")
    assign(tmp_name, tmp_hill_hier)
    tmp_path <- file.path("files/alpha/rdata/")
    saveRDS(tmp_hill_hier, paste(tmp_path, tmp_name, ".rds", sep = ""))
    rm(list = ls(pattern = "tmp_"))
}

objects(pattern = "its_hill_hier")
```

Again, we start by testing significance of ASV diversity for the data sets against each of the three metrics using the `div_test` function.

The command is as follows:

```
div_test(countable = x, qvalue = i, hierarchy = hier, 
         tree = ultrametric_tree, posthoc = TRUE)
```

Where `x` is ASV by sample table, `i` is the q-value corresponding to the metric of interest, `hier` is the hierarchy matrix, `tree` is an ultrametric formatted phylogenetic tree (if you choose to explore lineage diversity), and `posthoc` indicates whether to run post hoc pairwise analyses.

```{r, results='hide', code_folding=TRUE}
qvalue <- c(0,1,2)
for (i in its_alpha_ds) {
     for (j in qvalue) {
         tmp_year <- gsub(".*_", "", i)
         tmp_get <- get(purrr::map_chr(i, ~ paste0(., "_tu")))
         tmp_heir <- get(paste("its_hill_hier", tmp_year, sep = "_"))
         tmp_test <- div_test(tmp_get, qvalue = j,
                      hierarchy = tmp_heir,
                      posthoc = TRUE)
         tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
         print(tmp_name)
         assign(tmp_name, tmp_test)
         rm(list = ls(pattern = "tmp_"))
 }
}
```


```{r, echo=FALSE}
tmp_metric <- data.frame(c("Observed", "Shannon exponential", "Inverse Simpson"))
tmp_qvalue <- data.frame(c("0", "1", "2"))

qvalue <- c(0,1,2)
for (i in its_alpha_ds) {

  tmp_h_pvalue <- c()
     for (j in qvalue) {
          tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
          tmp_get <- get(tmp_name)$homogeneity.pvalue
          tmp_h_pvalue <- c(append(tmp_h_pvalue, tmp_get))
     }
  tmp_h_pvalue <- data.frame(tmp_h_pvalue)

  tmp_n_pvalue <- c()
      for (j in qvalue) {
          tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
          tmp_get <- get(tmp_name)$normality.pvalue
          tmp_n_pvalue <- c(append(tmp_n_pvalue, tmp_get))
     }
  tmp_n_pvalue <- data.frame(tmp_n_pvalue)

  tmp_method <- c()
      for (j in qvalue) {
          tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
          tmp_get <- get(tmp_name)$method
          tmp_method <- c(append(tmp_method, tmp_get))
     }
  tmp_method <- data.frame(tmp_method)

  tmp_phoc_method <- c()
      for (j in qvalue) {
          tmp_name <- purrr::map_chr(i, ~ paste0(., "_q", j, "_adt"))
          tmp_get <- get(tmp_name)$posthoc.method
          tmp_phoc_method <- c(append(tmp_phoc_method, tmp_get))
     }
  tmp_phoc_method <- data.frame(tmp_phoc_method)

  tmp_df <- dplyr::bind_cols(tmp_metric, tmp_qvalue) %>%
                     dplyr::bind_cols(., tmp_n_pvalue) %>%
                     dplyr::bind_cols(., tmp_h_pvalue) %>%
                     dplyr::bind_cols(., tmp_method) %>%
                     dplyr::bind_cols(., tmp_phoc_method) %>%
  dplyr::rename("metric" = 1, "q-value" = 2, "normality p-value" = 3, 
                "homogeneity p-value" = 4, "method" = 5, "posthoc method" = 6)
  tmp_name <- purrr::map_chr(i, ~ paste0(i, "_sig_tab"))
  assign(tmp_name, tmp_df)
}
```


```{r, echo=FALSE}
## Format as follows
## FALSE Kruskal-Wallis Test = its_pime_q1_adt$test[[3]]
## TRUE Tukey post-hoc  = its_ps_work_q0_adt$test[[1]][[1, 5]]
get_pvalue <-  function(value) {
        if (value$method == "ANOVA") {
          tmp_pvalue <- "[[1]][[1, 5]]" 
        } else if (value$method == "Kruskal-Wallis Test") {
          tmp_pvalue <- "[[3]]"
        } 
  tmp_test <- paste(deparse(substitute(value)), "$test",  noquote(tmp_pvalue), sep = "")
  return(eval(parse(text = tmp_test)))
}
objects(pattern = "_adt")
```

```{r, echo=FALSE}
qvalue <- c(0,1,2)
gen_summ <-  function(ps_object, year) {
  for (i in qvalue) {
  tmp_qname <-  paste("tmp_q", i, sep = "")  
  tmp_get <- get(paste(ps_object, "_", year, "_q", i,  "_adt", sep = ""))
  assign(tmp_qname, tmp_get, envir = parent.frame())
  }
  tmp_pvalue <- data.frame(c(get_pvalue(tmp_q0),
                             get_pvalue(tmp_q1),
                             get_pvalue(tmp_q2)
                             )
                           )
  tmp_sig_tab_df <- get(paste(ps_object, "_", year, "_sig_tab", sep = ""))
  tmp_sig_tab <- dplyr::bind_cols(tmp_sig_tab_df, tmp_pvalue) %>%
                         dplyr::rename("posthoc p-value" = 7)
  tmp_name <- paste(ps_object, "_", year, "_sig_tab_post", sep = "")
  assign(tmp_name, tmp_sig_tab, envir = parent.frame())
  rm(list = ls(pattern = "tmp_"))
} 
objects()
```

```{r, echo=FALSE}
gen_summ("its_ps_work", "Y0")
gen_summ("its_ps_filt", "Y0")
gen_summ("its_ps_perfect", "Y0")
gen_summ("its_ps_pime", "Y0")

gen_summ("its_ps_work", "Y1")
gen_summ("its_ps_filt", "Y1")
gen_summ("its_ps_perfect", "Y1")
gen_summ("its_ps_pime", "Y1")

gen_summ("its_ps_work", "Y4")
gen_summ("its_ps_filt", "Y4")
gen_summ("its_ps_perfect", "Y4")
gen_summ("its_ps_pime", "Y4")
rm(list = ls(pattern = "tmp_"))
```

```{r, echo=FALSE}
for (i in objects(pattern = "_sig_tab_post")) {
  tmp_get <- get(i)
  tmp_ds_type <- stringr::str_remove(i, "its_ps_") %>%
                 stringr::str_remove(., "_.*")
      if (tmp_ds_type == "work") {
          tmp_get$dataset <- "FULL" 
        } else if (tmp_ds_type == "filt") {
          tmp_get$dataset <- "FILT"
        } else if (tmp_ds_type == "perfect") {
          tmp_get$dataset <- "PERFect"
        } else if (tmp_ds_type == "pime") {
          tmp_get$dataset <- "PIME"
        } 
  tmp_ds_year <- stringr::str_extract(i, "Y[0-9]") 
      if (tmp_ds_year == "Y0") {
          tmp_get$year <- "Year 0" 
        } else if (tmp_ds_year == "Y1") {
          tmp_get$year <- "Year 1"
        } else if (tmp_ds_year == "Y4") {
          tmp_get$year <- "Year 4"
        } 

  tmp_get <- tmp_get[,c(8:9,1:7)]
  assign(i, tmp_get)
  rm(list = ls(pattern = "tmp_"))
}
objects(pattern = "_sig_tab_post")
```

```{r, echo=FALSE}
its_sig_tab_all_Y0 <- rbind(its_ps_work_Y0_sig_tab_post, 
                           its_ps_filt_Y0_sig_tab_post, 
                           its_ps_perfect_Y0_sig_tab_post,
                           its_ps_pime_Y0_sig_tab_post
                           )
its_sig_tab_all_Y1 <- rbind(its_ps_work_Y1_sig_tab_post, 
                           its_ps_filt_Y1_sig_tab_post, 
                           its_ps_perfect_Y1_sig_tab_post,
                           its_ps_pime_Y1_sig_tab_post
                           )
its_sig_tab_all_Y4 <- rbind(its_ps_work_Y4_sig_tab_post, 
                           its_ps_filt_Y4_sig_tab_post, 
                           its_ps_perfect_Y4_sig_tab_post,
                           its_ps_pime_Y4_sig_tab_post
                           )
```

::: l-page
::: {.panelset}
::: {.panel}

#### Year 0

<small>`r caption_tab_its("its_sign_tests_Y0")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_sig_tab_all_Y0)
seq_table <- seq_table %>% dplyr::mutate(across(.cols = c(5:6,9), round, digits = 4)) 
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
sig_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}

#### Year 1

<small>`r caption_tab_its("its_sign_tests_Y1")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_sig_tab_all_Y1)
seq_table <- seq_table %>% dplyr::mutate(across(.cols = c(5:6,9), round, digits = 4)) 
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
sig_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}

#### Year 4

<small>`r caption_tab_its("its_sign_tests_Y4")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_sig_tab_all_Y4)
seq_table <- seq_table %>% dplyr::mutate(across(.cols = c(5:6,9), round, digits = 4)) 
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
sig_table(seq_table)
rm(seq_table)
```
:::
:::
:::

## Results of PostHoc Analyses 

Here we can inspect the results of each posthoc analysis for the unfiltered and filtered data sets.

::: l-body
::: {.panelset}
::: {.panel}

#### Year 0

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Full</strong> data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_f0_lab <- "Y0 FULL (Observed)"
its_Y0_f0_lab
its_ps_work_Y0_q0_adt$posthoc.method
data.frame(its_ps_work_Y0_q0_adt$posthoc)
objects()
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_f1_lab <- "Y0 FULL (Shannon exponential)"
its_Y0_f1_lab
its_ps_work_Y0_q1_adt$posthoc.method
data.frame(its_ps_work_Y0_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_f2_lab <- "Y0 FULL (Inverse Simpson)"
its_Y0_f2_lab
its_ps_work_Y0_q2_adt$posthoc.method
data.frame(its_ps_work_Y0_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Arbitrary</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_l0_lab <- "Y0 FILT (Observed)"
its_Y0_l0_lab
its_ps_filt_Y0_q0_adt$posthoc.method
data.frame(its_ps_filt_Y0_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_l1_lab <- "Y0 FILT (Shannon exponential)"
its_Y0_l1_lab
its_ps_filt_Y0_q1_adt$posthoc.method
data.frame(its_ps_filt_Y0_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_l2_lab <- "Y0 FILT (Inverse Simpson)"
its_Y0_l2_lab
its_ps_filt_Y0_q2_adt$posthoc.method
data.frame(its_ps_filt_Y0_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PERFect</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_r0_lab <- "Y0 PERFect (Observed)"
its_Y0_r0_lab
its_ps_perfect_Y0_q0_adt$posthoc.method
data.frame(its_ps_perfect_Y0_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_r1_lab <- "Y0 PERFect (Shannon exponential)"
its_Y0_r1_lab
its_ps_perfect_Y0_q1_adt$posthoc.method
data.frame(its_ps_perfect_Y0_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_r2_lab <- "Y0 PERFect (Inverse Simpson)"
its_Y0_r2_lab
its_ps_perfect_Y0_q2_adt$posthoc.method
data.frame(its_ps_perfect_Y0_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PIME</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_p0_lab <- "Y0 PIME (Observed)"
its_Y0_p0_lab
its_ps_pime_Y0_q0_adt$posthoc.method
data.frame(its_ps_pime_Y0_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_p1_lab <- "Y0 PIME (Shannon exponential)"
its_Y0_p1_lab
its_ps_pime_Y0_q1_adt$posthoc.method
data.frame(its_ps_pime_Y0_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y0_p2_lab <- "Y0 PIME (Inverse Simpson)"
its_Y0_p2_lab
its_ps_pime_Y0_q2_adt$posthoc.method
data.frame(its_ps_pime_Y0_q2_adt$posthoc)
```
</details>
:::

::: {.panel}

#### Year 1

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Full</strong> data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_f0_lab <- "Y1 FULL (Observed)"
its_Y1_f0_lab
its_ps_work_Y1_q0_adt$posthoc.method
data.frame(its_ps_work_Y1_q0_adt$posthoc)
objects()
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_f1_lab <- "Y1 FULL (Shannon exponential)"
its_Y1_f1_lab
its_ps_work_Y1_q1_adt$posthoc.method
data.frame(its_ps_work_Y1_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_f2_lab <- "Y1 FULL (Inverse Simpson)"
its_Y1_f2_lab
its_ps_work_Y1_q2_adt$posthoc.method
data.frame(its_ps_work_Y1_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Arbitrary</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_l0_lab <- "Y1 FILT (Observed)"
its_Y1_l0_lab
its_ps_filt_Y1_q0_adt$posthoc.method
data.frame(its_ps_filt_Y1_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_l1_lab <- "Y1 FILT (Shannon exponential)"
its_Y1_l1_lab
its_ps_filt_Y1_q1_adt$posthoc.method
data.frame(its_ps_filt_Y1_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_l2_lab <- "Y1 FILT (Inverse Simpson)"
its_Y1_l2_lab
its_ps_filt_Y1_q2_adt$posthoc.method
data.frame(its_ps_filt_Y1_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PERFect</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_r0_lab <- "Y1 PERFect (Observed)"
its_Y1_r0_lab
its_ps_perfect_Y1_q0_adt$posthoc.method
data.frame(its_ps_perfect_Y1_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_r1_lab <- "Y1 PERFect (Shannon exponential)"
its_Y1_r1_lab
its_ps_perfect_Y1_q1_adt$posthoc.method
data.frame(its_ps_perfect_Y1_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_r2_lab <- "Y1 PERFect (Inverse Simpson)"
its_Y1_r2_lab
its_ps_perfect_Y1_q2_adt$posthoc.method
data.frame(its_ps_perfect_Y1_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PIME</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_p0_lab <- "Y1 PIME (Observed)"
its_Y1_p0_lab
its_ps_pime_Y1_q0_adt$posthoc.method
data.frame(its_ps_pime_Y1_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_p1_lab <- "Y1 PIME (Shannon exponential)"
its_Y1_p1_lab
its_ps_pime_Y1_q1_adt$posthoc.method
data.frame(its_ps_pime_Y1_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y1_p2_lab <- "Y1 PIME (Inverse Simpson)"
its_Y1_p2_lab
its_ps_pime_Y1_q2_adt$posthoc.method
data.frame(its_ps_pime_Y1_q2_adt$posthoc)
```
</details>
:::

::: {.panel}

#### Year 4

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Full</strong> data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_f0_lab <- "Y4 FULL (Observed)"
its_Y4_f0_lab
its_ps_work_Y4_q0_adt$posthoc.method
data.frame(its_ps_work_Y4_q0_adt$posthoc)
objects()
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_f1_lab <- "Y4 FULL (Shannon exponential)"
its_Y4_f1_lab
its_ps_work_Y4_q1_adt$posthoc.method
data.frame(its_ps_work_Y4_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_f2_lab <- "Y4 FULL (Inverse Simpson)"
its_Y4_f2_lab
its_ps_work_Y4_q2_adt$posthoc.method
data.frame(its_ps_work_Y4_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>Arbitrary</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_l0_lab <- "Y4 FILT (Observed)"
its_Y4_l0_lab
its_ps_filt_Y4_q0_adt$posthoc.method
data.frame(its_ps_filt_Y4_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_l1_lab <- "Y4 FILT (Shannon exponential)"
its_Y4_l1_lab
its_ps_filt_Y4_q1_adt$posthoc.method
data.frame(its_ps_filt_Y4_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_l2_lab <- "Y4 FILT (Inverse Simpson)"
its_Y4_l2_lab
its_ps_filt_Y4_q2_adt$posthoc.method
data.frame(its_ps_filt_Y4_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PERFect</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_r0_lab <- "Y4 PERFect (Observed)"
its_Y4_r0_lab
its_ps_perfect_Y4_q0_adt$posthoc.method
data.frame(its_ps_perfect_Y4_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_r1_lab <- "Y4 PERFect (Shannon exponential)"
its_Y4_r1_lab
its_ps_perfect_Y4_q1_adt$posthoc.method
data.frame(its_ps_perfect_Y4_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_r2_lab <- "Y4 PERFect (Inverse Simpson)"
its_Y4_r2_lab
its_ps_perfect_Y4_q2_adt$posthoc.method
data.frame(its_ps_perfect_Y4_q2_adt$posthoc)
```
</details>

<details markdown="1">
<summary><strong>Click here</strong> for detailed results from the <strong>PIME</strong> filtered data set.</summary>

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_p0_lab <- "Y4 PIME (Observed)"
its_Y4_p0_lab
its_ps_pime_Y4_q0_adt$posthoc.method
data.frame(its_ps_pime_Y4_q0_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_p1_lab <- "Y4 PIME (Shannon exponential)"
its_Y4_p1_lab
its_ps_pime_Y4_q1_adt$posthoc.method
data.frame(its_ps_pime_Y4_q1_adt$posthoc)
```

```{r, echo=FALSE, eval=TRUE, results='hold', comment=''}
its_Y4_p2_lab <- "Y4 PIME (Inverse Simpson)"
its_Y4_p2_lab
its_ps_pime_Y4_q2_adt$posthoc.method
data.frame(its_ps_pime_Y4_q2_adt$posthoc)
```
</details>
:::
:::
:::


## Alpha Diversity Plots

Now we can plot the results from the posthoc analyses for each metric and data set using the function `div_test_plot_jjs`. I modified the original function (`div_test_plot`) to control a little of the formatting.

The command is as follows:

```
div_test_plot(divtest = x, chart = "type", colour = col.pal, 
              posthoc = TRUE, threshold = value))
```

Where `x` is the results from the `div_test` function, `"type"` is chart type (box, jitter, or violin), `colour` is is a color palette, `posthoc` indicates whether to run posthoc pairwise analyses, and `value` is the maximum p-value to show in pairwise posthoc results. **WARNING** if none of the posthoc results are below the specified threshold, the function will throw an error. Therefore, until this is fixed, all posthoc values are shown.


```{r, code_folding=TRUE}
agua_col <- c("#B26322", "#29B222", "#2271B2", "#AB22B2")

source("hack_code/div_test_plot_jjs.R")
tmp_order <- c('virginica', 'versicolor', 'setosa') 
rm(list = ls(pattern="_adt_plot"))
for (i in objects(pattern="_adt")) {
     tmp_name <- purrr::map_chr(i, ~ paste0(., "_plot"))
     tmp_get <- get(i)
     tmp_ds_year <- stringr::str_extract(i, "Y[0-9]") 
     tmp_order <- c(
       paste(tmp_ds_year, "CTL", sep = "_"), 
       paste(tmp_ds_year, "N", sep = "_"),
       paste(tmp_ds_year, "P", sep = "_"),
       paste(tmp_ds_year, "NP", sep = "_")
       ) 
     tmp_get$data <- tmp_get$data %>% 
                     mutate(Group = factor(Group, levels = tmp_order)) %>% 
                     arrange(Group)  
     tmp_df <- div_test_plot_jjs(tmp_get, chart = "box", 
                             posthoc = TRUE, 
                             #threshold = 0.05,
                             colour = agua_col)
     tmp_df <- ggpar(tmp_df, legend = "none")
     #print(tmp_name)
     assign(tmp_name, tmp_df)
     rm(list = ls(pattern = "tmp_"))
}
```

```{r, echo=FALSE}
div_plot_theme <- list(theme(
  plot.title = element_text(size = 10, face = "bold"), 
  plot.title.position = "plot",
  axis.text.x = element_text(size = 8, angle = 0, hjust = 0.5), 
  axis.text.y = element_text(size = 8),
  )
)
new_labs <- c("Y4_CTL" = "CTL", "Y4_N" = "N", "Y4_P" = "P", "Y4_NP" = "NP")
```

```{r, echo=FALSE}
its_ps_work_Y0_q0_adt_plot <- its_ps_work_Y0_q0_adt_plot +
                              ggtitle(its_Y0_f0_lab) + 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
its_ps_work_Y0_q1_adt_plot <- its_ps_work_Y0_q1_adt_plot +
                              ggtitle(its_Y0_f1_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
its_ps_work_Y0_q2_adt_plot <- its_ps_work_Y0_q2_adt_plot +
                              ggtitle(its_Y0_f2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

its_ps_work_Y1_q0_adt_plot <- its_ps_work_Y1_q0_adt_plot +
                              ggtitle(its_Y1_f0_lab)+ 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
its_ps_work_Y1_q1_adt_plot <- its_ps_work_Y1_q1_adt_plot +
                              ggtitle(its_Y1_f1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
its_ps_work_Y1_q2_adt_plot <- its_ps_work_Y1_q2_adt_plot +
                              ggtitle(its_Y1_f2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

its_ps_work_Y4_q0_adt_plot <- its_ps_work_Y4_q0_adt_plot +
                              ggtitle(its_Y4_f0_lab)+ 
                              div_plot_theme + 
                              scale_x_discrete(labels = new_labs)
its_ps_work_Y4_q1_adt_plot <- its_ps_work_Y4_q1_adt_plot +
                              ggtitle(its_Y4_f1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)

its_ps_work_Y4_q2_adt_plot <- its_ps_work_Y4_q2_adt_plot +
                              ggtitle(its_Y4_f2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)
```

```{r, echo=FALSE}
its_ps_filt_Y0_q0_adt_plot <- its_ps_filt_Y0_q0_adt_plot +
                              ggtitle(its_Y0_l0_lab) + 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
its_ps_filt_Y0_q1_adt_plot <- its_ps_filt_Y0_q1_adt_plot +
                              ggtitle(its_Y0_l1_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
its_ps_filt_Y0_q2_adt_plot <- its_ps_filt_Y0_q2_adt_plot +
                              ggtitle(its_Y0_l2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

its_ps_filt_Y1_q0_adt_plot <- its_ps_filt_Y1_q0_adt_plot +
                              ggtitle(its_Y1_l0_lab)+ 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
its_ps_filt_Y1_q1_adt_plot <- its_ps_filt_Y1_q1_adt_plot +
                              ggtitle(its_Y1_l1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
its_ps_filt_Y1_q2_adt_plot <- its_ps_filt_Y1_q2_adt_plot +
                              ggtitle(its_Y1_l2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

its_ps_filt_Y4_q0_adt_plot <- its_ps_filt_Y4_q0_adt_plot +
                              ggtitle(its_Y4_l0_lab)+ 
                              div_plot_theme + 
                              scale_x_discrete(labels = new_labs)
its_ps_filt_Y4_q1_adt_plot <- its_ps_filt_Y4_q1_adt_plot +
                              ggtitle(its_Y4_l1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)

its_ps_filt_Y4_q2_adt_plot <- its_ps_filt_Y4_q2_adt_plot +
                              ggtitle(its_Y4_l2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)
```

```{r, echo=FALSE}
its_ps_perfect_Y0_q0_adt_plot <- its_ps_perfect_Y0_q0_adt_plot +
                              ggtitle(its_Y0_r0_lab) + 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
its_ps_perfect_Y0_q1_adt_plot <- its_ps_perfect_Y0_q1_adt_plot +
                              ggtitle(its_Y0_r1_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
its_ps_perfect_Y0_q2_adt_plot <- its_ps_perfect_Y0_q2_adt_plot +
                              ggtitle(its_Y0_r2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

its_ps_perfect_Y1_q0_adt_plot <- its_ps_perfect_Y1_q0_adt_plot +
                              ggtitle(its_Y1_r0_lab)+ 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
its_ps_perfect_Y1_q1_adt_plot <- its_ps_perfect_Y1_q1_adt_plot +
                              ggtitle(its_Y1_r1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
its_ps_perfect_Y1_q2_adt_plot <- its_ps_perfect_Y1_q2_adt_plot +
                              ggtitle(its_Y1_r2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

its_ps_perfect_Y4_q0_adt_plot <- its_ps_perfect_Y4_q0_adt_plot +
                              ggtitle(its_Y4_r0_lab)+ 
                              div_plot_theme + 
                              scale_x_discrete(labels = new_labs)
its_ps_perfect_Y4_q1_adt_plot <- its_ps_perfect_Y4_q1_adt_plot +
                              ggtitle(its_Y4_r1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)

its_ps_perfect_Y4_q2_adt_plot <- its_ps_perfect_Y4_q2_adt_plot +
                              ggtitle(its_Y4_r2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)
```

```{r, echo=FALSE}
its_ps_pime_Y0_q0_adt_plot <- its_ps_pime_Y0_q0_adt_plot +
                              ggtitle(its_Y0_p0_lab) + 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
its_ps_pime_Y0_q1_adt_plot <- its_ps_pime_Y0_q1_adt_plot +
                              ggtitle(its_Y0_p1_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
its_ps_pime_Y0_q2_adt_plot <- its_ps_pime_Y0_q2_adt_plot +
                              ggtitle(its_Y0_p2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

its_ps_pime_Y1_q0_adt_plot <- its_ps_pime_Y1_q0_adt_plot +
                              ggtitle(its_Y1_p0_lab)+ 
                              div_plot_theme + 
                              theme(axis.title.x = element_blank(), 
                                    axis.text.x = element_blank())
its_ps_pime_Y1_q1_adt_plot <- its_ps_pime_Y1_q1_adt_plot +
                              ggtitle(its_Y1_p1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())
its_ps_pime_Y1_q2_adt_plot <- its_ps_pime_Y1_q2_adt_plot +
                              ggtitle(its_Y1_p2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank(), 
                                    axis.title.x = element_blank(),
                                    axis.text.x = element_blank())

its_ps_pime_Y4_q0_adt_plot <- its_ps_pime_Y4_q0_adt_plot +
                              ggtitle(its_Y4_p0_lab)+ 
                              div_plot_theme + 
                              scale_x_discrete(labels = new_labs)
its_ps_pime_Y4_q1_adt_plot <- its_ps_pime_Y4_q1_adt_plot +
                              ggtitle(its_Y4_p1_lab) + 
                              div_plot_theme +
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)

its_ps_pime_Y4_q2_adt_plot <- its_ps_pime_Y4_q2_adt_plot +
                              ggtitle(its_Y4_p2_lab) + 
                              div_plot_theme + 
                              theme(axis.title.y = element_blank()) + 
                              scale_x_discrete(labels = new_labs)
```


```{r, echo=FALSE}
its_alph_div_plots_work <- ggarrange(
           its_ps_work_Y0_q0_adt_plot, 
           its_ps_work_Y0_q1_adt_plot, 
           its_ps_work_Y0_q2_adt_plot, 
           its_ps_work_Y1_q0_adt_plot, 
           its_ps_work_Y1_q1_adt_plot, 
           its_ps_work_Y1_q2_adt_plot, 
           its_ps_work_Y4_q0_adt_plot, 
           its_ps_work_Y4_q1_adt_plot, 
           its_ps_work_Y4_q2_adt_plot, 
           ncol = 3, nrow = 3
           )

its_alph_div_plots_work
```


```{r, echo=FALSE}
its_alph_div_plots_filt <- ggarrange(
           its_ps_filt_Y0_q0_adt_plot, 
           its_ps_filt_Y0_q1_adt_plot, 
           its_ps_filt_Y0_q2_adt_plot, 
           its_ps_filt_Y1_q0_adt_plot, 
           its_ps_filt_Y1_q1_adt_plot, 
           its_ps_filt_Y1_q2_adt_plot, 
           its_ps_filt_Y4_q0_adt_plot, 
           its_ps_filt_Y4_q1_adt_plot, 
           its_ps_filt_Y4_q2_adt_plot, 
           ncol = 3, nrow = 3)
its_alph_div_plots_filt
```


```{r, echo=FALSE}
its_alph_div_plots_perfect <- ggarrange(
           its_ps_perfect_Y0_q0_adt_plot, 
           its_ps_perfect_Y0_q1_adt_plot, 
           its_ps_perfect_Y0_q2_adt_plot, 
           its_ps_perfect_Y1_q0_adt_plot, 
           its_ps_perfect_Y1_q1_adt_plot, 
           its_ps_perfect_Y1_q2_adt_plot, 
           its_ps_perfect_Y4_q0_adt_plot, 
           its_ps_perfect_Y4_q1_adt_plot, 
           its_ps_perfect_Y4_q2_adt_plot, 
           ncol = 3, nrow = 3)
its_alph_div_plots_perfect

```

```{r, echo=FALSE}
its_alph_div_plots_pime <- ggarrange(
           its_ps_pime_Y0_q0_adt_plot, 
           its_ps_pime_Y0_q1_adt_plot, 
           its_ps_pime_Y0_q2_adt_plot, 
           its_ps_pime_Y1_q0_adt_plot, 
           its_ps_pime_Y1_q1_adt_plot, 
           its_ps_pime_Y1_q2_adt_plot, 
           its_ps_pime_Y4_q0_adt_plot, 
           its_ps_pime_Y4_q1_adt_plot, 
           its_ps_pime_Y4_q2_adt_plot, 
           ncol = 3, nrow = 3
           )
its_alph_div_plots_pime
```


```{r, echo=FALSE}
plots_to_save <- c("its_alph_div_plots_work", "its_alph_div_plots_filt", 
                   "its_alph_div_plots_perfect", "its_alph_div_plots_pime")

for (i in plots_to_save) {
  tmp_get <- get(i)
  ggplot2::ggsave(tmp_get, file = paste0("files/alpha/figures/", i, ".png", sep = ""), 
                   height = 20, width = 28, units = 'cm', bg = "white")
  ggplot2::ggsave(tmp_get, file = paste0("files/alpha/figures/", i, ".pdf", sep = ""), 
                   height = 20, width = 28, units = 'cm', bg = "white")
  rm(list = ls(pattern = "tmp_"))
}

system("cp files/alpha/figures/its_alph_div_plots_work.png include/alpha/its_alph_div_plots_work.png")
system("cp files/alpha/figures/its_alph_div_plots_filt.png include/alpha/its_alph_div_plots_filt.png")
system("cp files/alpha/figures/its_alph_div_plots_perfect.png include/alpha/its_alph_div_plots_perfect.png")
system("cp files/alpha/figures/its_alph_div_plots_pime.png include/alpha/its_alph_div_plots_pime.png")
```


> Posthoc adjusted p-values given for each significant pairwise comparison.

::: l-page
::: {.panelset}
::: {.panel}

### FULL data set

```{r, echo=FALSE, warning=FALSE, fig.height=5, layout='l-page', eval=TRUE}
knitr::include_graphics("include/alpha/its_alph_div_plots_work.png")
```
<small>`r caption_fig_its("its_alpha_div_plots_full")`</small>
:::

::: {.panel}

### Arbitrary filtered

```{r, echo=FALSE, warning=FALSE, fig.height=5, layout='l-page', eval=TRUE}
knitr::include_graphics("include/alpha/its_alph_div_plots_filt.png")
```
<small>`r caption_fig_its("its_alpha_div_plots_filt")`</small>
:::

::: {.panel}

### PERFect filtered

```{r, echo=FALSE, warning=FALSE, fig.height=5, layout='l-page', eval=TRUE}
knitr::include_graphics("include/alpha/its_alph_div_plots_perfect.png")
```
<small>`r caption_fig_its("its_alpha_div_plots_perfect")`</small>

:::

::: {.panel}

### PIME filtered

```{r, echo=FALSE, warning=FALSE, fig.height=5, layout='l-page', eval=TRUE}
knitr::include_graphics("include/alpha/its_alph_div_plots_pime.png")
```
<small>`r caption_fig_its("its_alpha_div_plots_pime")`</small>
:::
:::
:::

## Posthoc sample comparisons

```{r, echo=FALSE}
rm(list = ls(pattern = "_reform"))
for (i in objects(pattern = "_adt$")) {
  tmp_get <- get(i)$posthoc
  tmp_form <- if(is.data.frame(tmp_get)){
            tmp_form <- tmp_get[ -c(1:2) ]
            tmp_form <- tmp_form %>% tibble::rownames_to_column("Comparison") %>%
              dplyr::rename("p_value" = 2) #%>%
              #tidyr::separate(col = Comparison, into = c("S1", "S2"), sep = "-")
          } else {
            tmp_form <- data.frame(tmp_get)
            tmp_form <- tmp_form[ -c(1:3) ]
            tmp_form <- tmp_form %>% tibble::rownames_to_column("Comparison") %>%
              dplyr::rename("p_value" = 2) %>%
              tidyr::separate(col = Comparison, into = c("S2", "S1"), sep = "-")
            tmp_form <- tmp_form[, c(2, 1, 3)]
            tmp_form <- tmp_form %>% tidyr::unite("Comparison", 1:2, sep = "-", 
                                                  remove = TRUE, na.rm = FALSE)
          }
#  tmp_form <- tmp_form %>% tibble::rownames_to_column("Comparison") %>%
#    dplyr::rename("p_value" = 2)
  tmp_name <- purrr::map_chr(i, ~ paste0(., "_reform"))
  assign(tmp_name, tmp_form)
  rm(list = ls(pattern = "tmp_"))
}
objects(pattern = "_reform")
```


```{r, echo=FALSE}
its_ps_work_Y0_hill_sum <- dplyr::left_join(its_ps_work_Y0_q0_adt_reform, its_ps_work_Y0_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_work_Y0_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y0" = 3, "Shannon_exp_Y0" = 4, "InvSimpson_Y0" = 5)

its_ps_filt_Y0_hill_sum <- dplyr::left_join(its_ps_filt_Y0_q0_adt_reform, its_ps_filt_Y0_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_filt_Y0_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y0" = 3, "Shannon_exp_Y0" = 4, "InvSimpson_Y0" = 5)
its_ps_perfect_Y0_hill_sum <- dplyr::left_join(its_ps_perfect_Y0_q0_adt_reform, its_ps_perfect_Y0_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_perfect_Y0_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y0" = 3, "Shannon_exp_Y0" = 4, "InvSimpson_Y0" = 5)

its_ps_pime_Y0_hill_sum <- dplyr::left_join(its_ps_pime_Y0_q0_adt_reform, its_ps_pime_Y0_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_pime_Y0_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y0" = 3, "Shannon_exp_Y0" = 4, "InvSimpson_Y0" = 5)

```

```{r, echo=FALSE}
its_ps_work_Y1_hill_sum <- dplyr::left_join(its_ps_work_Y1_q0_adt_reform, its_ps_work_Y1_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_work_Y1_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y1" = 3, "Shannon_exp_Y1" = 4, "InvSimpson_Y1" = 5)

its_ps_filt_Y1_hill_sum <- dplyr::left_join(its_ps_filt_Y1_q0_adt_reform, its_ps_filt_Y1_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_filt_Y1_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y1" = 3, "Shannon_exp_Y1" = 4, "InvSimpson_Y1" = 5)
its_ps_perfect_Y1_hill_sum <- dplyr::left_join(its_ps_perfect_Y1_q0_adt_reform, its_ps_perfect_Y1_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_perfect_Y1_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y1" = 3, "Shannon_exp_Y1" = 4, "InvSimpson_Y1" = 5)

its_ps_pime_Y1_hill_sum <- dplyr::left_join(its_ps_pime_Y1_q0_adt_reform, its_ps_pime_Y1_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_pime_Y1_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y1" = 3, "Shannon_exp_Y1" = 4, "InvSimpson_Y1" = 5)
```

```{r, echo=FALSE}
its_ps_work_Y4_hill_sum <- dplyr::left_join(its_ps_work_Y4_q0_adt_reform, its_ps_work_Y4_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_work_Y4_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y4" = 3, "Shannon_exp_Y4" = 4, "InvSimpson_Y4" = 5)

its_ps_filt_Y4_hill_sum <- dplyr::left_join(its_ps_filt_Y4_q0_adt_reform, its_ps_filt_Y4_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_filt_Y4_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y4" = 3, "Shannon_exp_Y4" = 4, "InvSimpson_Y4" = 5)
its_ps_perfect_Y4_hill_sum <- dplyr::left_join(its_ps_perfect_Y4_q0_adt_reform, its_ps_perfect_Y4_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_perfect_Y4_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y4" = 3, "Shannon_exp_Y4" = 4, "InvSimpson_Y4" = 5)

its_ps_pime_Y4_hill_sum <- dplyr::left_join(its_ps_pime_Y4_q0_adt_reform, its_ps_pime_Y4_q1_adt_reform, by = "Comparison") %>%
                           dplyr::left_join(., its_ps_pime_Y4_q2_adt_reform, by = "Comparison") %>%
  tidyr::separate(col = Comparison, into = c("Group_1", "Group_2"), sep = "-") %>% 
  dplyr::rename("Observed_Y4" = 3, "Shannon_exp_Y4" = 4, "InvSimpson_Y4" = 5)

```


```{r, echo=FALSE}
## REMOVE YEAR
for (i in objects(pattern = "_hill_sum")) {
  tmp_get <- get(i)
  tmp_get$Group_1 <- tmp_get$Group_1 %>% str_remove_all(., "Y[0-9]_")
  tmp_get$Group_2 <- tmp_get$Group_2 %>% str_remove_all(., "Y[0-9]_")
  assign(i, tmp_get)
  rm(list = ls(pattern = "tmp_"))
}
```


```{r, echo=FALSE}
col_order <- c("Group_1", "Group_2", 
               "Observed_Y0", "Observed_Y1", "Observed_Y4", 
               "Shannon_exp_Y0", "Shannon_exp_Y1", "Shannon_exp_Y4", 
               "InvSimpson_Y0", "InvSimpson_Y1", "InvSimpson_Y4")
its_alpha_hill_work_posthoc <- dplyr::left_join(its_ps_work_Y0_hill_sum, its_ps_work_Y1_hill_sum, by = c("Group_1", "Group_2")) %>%
                             dplyr::left_join(., its_ps_work_Y4_hill_sum, by = c("Group_1", "Group_2"))
its_alpha_hill_work_posthoc <- its_alpha_hill_work_posthoc[, col_order]

its_alpha_hill_filt_posthoc <- dplyr::left_join(its_ps_filt_Y0_hill_sum, its_ps_filt_Y1_hill_sum, by = c("Group_1", "Group_2")) %>%
                             dplyr::left_join(., its_ps_filt_Y4_hill_sum, by = c("Group_1", "Group_2"))
its_alpha_hill_filt_posthoc <- its_alpha_hill_filt_posthoc[, col_order]

its_alpha_hill_perfect_posthoc <- dplyr::left_join(its_ps_perfect_Y0_hill_sum, its_ps_perfect_Y1_hill_sum, by = c("Group_1", "Group_2")) %>%
                             dplyr::left_join(., its_ps_perfect_Y4_hill_sum, by = c("Group_1", "Group_2"))
its_alpha_hill_perfect_posthoc <- its_alpha_hill_perfect_posthoc[, col_order]

its_alpha_hill_pime_posthoc <- dplyr::left_join(its_ps_pime_Y0_hill_sum, its_ps_pime_Y1_hill_sum, by = c("Group_1", "Group_2")) %>%
                             dplyr::left_join(., its_ps_pime_Y4_hill_sum, by = c("Group_1", "Group_2"))
its_alpha_hill_pime_posthoc <- its_alpha_hill_pime_posthoc[, col_order]

```

::: l-page
::: {.panelset}
::: {.panel}

#### FULL data set

<small>`r caption_tab_its("its_alpha_hill_work_posthoc")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_alpha_hill_work_posthoc)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
poshoc_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}

#### Arbitrary filtered

<small>`r caption_tab_its("its_alpha_hill_filt_posthoc")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_alpha_hill_filt_posthoc)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
poshoc_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}

#### PERFect filtered

<small>`r caption_tab_its("its_alpha_hill_perfect_posthoc")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_alpha_hill_perfect_posthoc)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
poshoc_table(seq_table)
rm(seq_table)
```
:::

::: {.panel}
#### PIME filtered

<small>`r caption_tab_its("its_alpha_hill_pime_posthoc")`</small>

```{r, echo=FALSE, eval=TRUE}
download_this_fun(its_alpha_hill_pime_posthoc)
```

```{r, echo=FALSE, layout="l-page", eval=TRUE}
poshoc_table(seq_table)
rm(seq_table)
```
:::
:::
:::



```{r, echo=FALSE, eval=FALSE}
## Load to build page only #2
rm(its_ps_work, its_ps_filt, its_ps_perfect, its_ps_pime)
objects()
save.image("page_build/alpha_wf_its.rdata")
```

```{r, echo=FALSE, eval=FALSE}
remove(list = ls())
pacman::p_unload(all)
```


##  Source Code {.appendix}

The source code for this page can be accessed on GitHub by [clicking this link](https://github.com/agua-salud/nutrients/blob/main/alpha.Rmd).

## Data Availability {.appendix}

Data generated in this workflow and the Rdata need to run the workflow can be accessed on figshare at [10.25573/data.16826779](https://doi.org/10.25573/data.16826779).