<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>

<style>
  div.csl-bib-body { }
  div.csl-entry {
    clear: both;
    }
  .hanging div.csl-entry {
    margin-left:2em;
    text-indent:-2em;
  }
  div.csl-left-margin {
    min-width:2em;
    float:left;
  }
  div.csl-right-inline {
    margin-left:2em;
    padding-left:1em;
  }
  div.csl-indent {
    margin-left: 2em;
  }
</style>

  <!--radix_placeholder_meta_tags-->
  <title>agua_salud: 6. Beta Diversity &amp; Dispersion Estimates</title>

  <meta property="description" itemprop="description" content="Reproducible workflow for assessing community dissimilarity across temperature treatments."/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="agua_salud: 6. Beta Diversity &amp; Dispersion Estimates"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Reproducible workflow for assessing community dissimilarity across temperature treatments."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="agua_salud"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="agua_salud: 6. Beta Diversity &amp; Dispersion Estimates"/>
  <meta property="twitter:description" content="Reproducible workflow for assessing community dissimilarity across temperature treatments."/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Vegan: Community ecology package;citation_publication_date=2012;citation_volume=2;citation_author=Jari Oksanen;citation_author=F. Guillaume Blanchet;citation_author=Roeland Kindt;citation_author=Pierre Legendre;citation_author=Peter R. Minchin;citation_author=R. B. O’Hara;citation_author=Gavin L. Simpson;citation_author=Peter Sólymos;citation_author=M. Henry H. Stevens;citation_author=Helene Wagner"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["6. Beta Diversity & Dispersion Estimates"]},{"type":"character","attributes":{},"value":["Reproducible workflow for assessing community dissimilarity across temperature treatments.\n"]},{"type":"character","attributes":{},"value":["assets/cite.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["css","toc","toc_depth"]}},"value":[{"type":"character","attributes":{},"value":["assets/css/styles.css"]},{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[2]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/font-awesome-5.13.0/js/script.js"></script>
  <script src="site_libs/core-js-2.5.3/shim.min.js"></script>
  <script src="site_libs/react-17.0.0/react.min.js"></script>
  <script src="site_libs/react-17.0.0/react-dom.min.js"></script>
  <script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
  <script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
  <script src="site_libs/reactable-binding-0.2.3.9000/reactable.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500|Karla:400,400i,700,700i');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.1em;
  	font-family: 'Work Sans';
    font-style: normal;
    color: #323232;
    margin: 0.5em 0 0.5em;
  }

  d-article blockquote {
    border-left: 4px solid;
    border-left-color: #323232;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #b54a4d;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #4A4DB5;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;
    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  li code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
    color: #686868;
  }
  .nav-dropdown-content a {
    font-family: 'Work Sans'
  }
  .nav-dropdown .nav-dropbtn {
  /*  color: #FEFEFE; */
    font-weight: 400;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }

  .distill-site-nav {
      background-color: #323232;
      font-size: 16px;
      font-weight: 300;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }

  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;
  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;
  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;
  }

  summary:focus {
    outline: none;
  }

  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  /********************************/
  /* controls downloadthis  */
  /********************************/

  button.btn.btn-default {
    font-size: 0.8em;
    font-family: "PT Sans";
    border-color: #4DB54A;
    border-width: medium;
    margin-bottom: 15px;
  }

  /********************************/
  /* controls reactable  */
  /********************************/

  div.rt-text-content {
    font-size: 1.2em;
  }

  /********************************/
  /* controls horizontal rule, ie ***  */
  /********************************/

  d-article hr {
    margin: 30px 0px;
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/css/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"6. Beta Diversity & Dispersion Estimates","description":"Reproducible workflow for assessing community dissimilarity across temperature treatments.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title"></a>
<a href="index.html">
<i class="fa fa-home" aria-hidden="true"></i>
</a>
</div>
<div class="nav-right">
<div class="nav-dropdown">
<button class="nav-dropbtn">
A. Processing
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Workflows</span>
<hr/>
<a href="dada2.html">1. DADA2</a>
<a href="data-prep.html">2. Data Set Prep</a>
<a href="filtering.html">3. Filtering</a>
</div>
</div>
<div class="nav-dropdown">
<button class="nav-dropbtn">
B. Diversity
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<span class="nav-dropdown-header">Workflows</span>
<hr/>
<a href="taxa.html">4. Taxonomic</a>
<a href="alpha.html">5. Alpha Diversity</a>
<a href="beta.html">6. Beta Diversity</a>
<a href="da.html">7. DA ASVs &amp; Taxa</a>
<a href="metadata.html">8. Metadata</a>
</div>
</div>
<a href="data-availability.html">Data Products</a>
<a href="som.html">SOM</a>
<a href="https://github.com/agua-salud/nutrients/">
<i class="fa fa-github fa-lg" aria-hidden="true"></i>
</a>
<a href="about.html">
<i class="fa fa-question" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>6. Beta Diversity &amp; Dispersion Estimates</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Reproducible workflow for assessing community dissimilarity across temperature treatments.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#synopsis">Synopsis</a></li>
<li><a href="#s-rrna">16s rRNA</a>
<ul>
<li><a href="#beta-dispersion">Beta Dispersion</a></li>
<li><a href="#significance-tests">Significance Tests</a></li>
<li><a href="#summaries">Summaries</a></li>
</ul></li>
<li><a href="#its">ITS</a>
<ul>
<li><a href="#beta-dispersion-1">Beta Dispersion</a></li>
<li><a href="#significance-tests-1">Significance Tests</a></li>
<li><a href="#summaries-1">Summaries</a></li>
</ul></li>
<li><a href="#ordinations">Ordinations</a>
<ul>
<li><a href="#plot-code">Plot Code</a></li>
<li><a href="#s-rrna-plots">16S rRNA plots</a></li>
<li><a href="#its-plots">ITS Plots</a></li>
</ul></li>
<li><a href="#within-between-group-distance-plots">Within &amp; Between Group Distance Plots</a>
<ul>
<li><a href="#s-rrna-plots-1">16S rRNA Plots</a></li>
<li><a href="#its-plots-1">ITS Plots</a></li>
<li><a href="#source-code">Source Code</a></li>
<li><a href="#data-availability">Data Availability</a></li>
</ul></li>
</ul>
</nav>
</div>
<details>
<summary>
<strong>Click here</strong> for setup information.
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">eval =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">119</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq); <span class="fu">packageVersion</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings); <span class="fu">packageVersion</span>(<span class="st">&quot;Biostrings&quot;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, microeco, file2meco, magrittr,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>               microbiome, phytools, phangorn, reactable, Matrix, </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>               pairwiseAdonis, codefolder, naniar, downloadthis,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>               labdsv, patchwork, agricolae, captioner, </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends(agricolae, local = TRUE)  </span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co">#pacman::p_depends_reverse(agricolae, local = TRUE)  </span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_current<span class="sc">$</span><span class="fu">get</span>(<span class="fu">c</span>(</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache&quot;</span>,</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.path&quot;</span>,</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;cache.rebuild&quot;</span>,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;dependson&quot;</span>,</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;autodep&quot;</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>))</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><style>.panelset{--panel-tab-active-foreground: #2271B2;--panel-tab-hover-foreground: #AB22B2;}</style></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">### COmmon formatting scripts</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="do">### </span><span class="al">NOTE</span><span class="do">: captioner.R must be read BEFORE captions_XYZ.R</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;assets/captioner/captioner.R&quot;</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;assets/captioner/captions/captions_beta.R&quot;</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;assets/reactable/download_this_fun.R&quot;</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;assets/reactable/styles.R&quot;</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;assets/reactable/table_functions/beta.R&quot;</span>)</span></code></pre></div>
</div>
</details>
<h1 id="synopsis">Synopsis</h1>
<p>This workflow contains diversity assessments for the high temperature data sets. In order to run the workflow, you either need to first run the <a href="dada2.html">DADA2 Workflow for 2018 High Temp samples</a> <strong>and</strong> the <a href="data-prep.html">Data Preparation workflow</a> <strong>or</strong> begin with the output files from the Data Preparation workflow. See the <a href="data-availability.html">Data Availability</a> page for complete details.</p>
<p>In this workflow…</p>
<h1 id="s-rrna">16s rRNA</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>In order to test for significance between sample groups, we must perform the following steps:</p>
<ol type="1">
<li>Transform sample counts to relative abundance.</li>
<li>Create distance matrices based on some metrics, for example <code>wunifrac</code>, <code>unifrac</code>, and <code>jsd</code>. For this we use the function <code>phyloseq::distance</code>.</li>
<li>Next, calculate beta dispersion using the <code>betadisper</code> function from the <code>vegan</code> package <span class="citation" data-cites="oksanen2013community">(<a href="#ref-oksanen2013community" role="doc-biblioref">Oksanen et al. 2012</a>)</span>.</li>
<li>Then, use the function <code>permutest</code> to run a Permutation test for homogeneity of multivariate dispersions.</li>
<li>If the beta dispersion tests are not significant we will run a PERMANOVA using <code>adonis</code> (since PERMANOVA assumes equal dispersion), otherwise we will use Analysis of Similarity (ANOSIM), both available in the <code>vegan</code> package <span class="citation" data-cites="oksanen2013community">(<a href="#ref-oksanen2013community" role="doc-biblioref">Oksanen et al. 2012</a>)</span>.</li>
</ol>
<p>Steps 2–4 are combined in a <code>for</code> loop that tests all three distance metrics.</p>
<p>First, transform sample counts to relative abundance and generate some new trees.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ssu_samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;ssu_.*_Y[0-9]$&quot;</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                                          <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>     tmp_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(tmp_ps), <span class="at">rooted =</span> <span class="cn">TRUE</span>, </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(tmp_ps))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data, tmp_tree)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>choose_group <span class="ot">&lt;-</span> <span class="st">&quot;YR_TREAT&quot;</span></span></code></pre></div>
</div>
<h2 id="beta-dispersion">Beta Dispersion</h2>
<p>Here we create distance matrices for each metric, calculate the beta dispersion, and run a permutation test for homogeneity of multivariate dispersions.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>ssu_dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;jsd&quot;</span>, <span class="st">&quot;unifrac&quot;</span>, <span class="st">&quot;wunifrac&quot;</span>, <span class="st">&quot;jaccard&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (d <span class="cf">in</span> ssu_dist){</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>       tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>       tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span>(d <span class="sc">==</span> <span class="st">&quot;jaccard&quot;</span>){</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>         tmp_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_get, <span class="at">method =</span> d, <span class="at">binary =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>       } <span class="cf">else</span> {</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>         tmp_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_get, <span class="at">method =</span> d)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>         }</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>       <span class="co">#tmp_df &lt;- phyloseq::distance(tmp_get, method = d)</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>       tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dist_&quot;</span>, .))</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name, tmp_df)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>       tmp_df2 <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(tmp_df, tmp_samp[[choose_group]], <span class="at">bias.adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>       tmp_df_name2 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dispersion_&quot;</span>, .))</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name2, tmp_df2)</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>       tmp_df3 <span class="ot">&lt;-</span> <span class="fu">permutest</span>(tmp_df2, <span class="at">pairwise =</span> <span class="cn">TRUE</span>,</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>                            <span class="at">permutations =</span> <span class="dv">1000</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>       tmp_df_name3 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_permutest_&quot;</span>, .))</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name3, tmp_df3)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<summary>
Results of Beta Dispersion &amp; Permutation tests <strong>(FULL data set)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.005201 0.0017338 0.3864   1000 0.7762
Residuals 55 0.246802 0.0044873                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.92408 0.46354 0.7363
Y0_N   0.91777         0.44955 0.8282
Y0_NP  0.46578 0.43535         0.3267
Y0_P   0.72692 0.81318 0.32546       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.002500 0.00083345 0.5799   1000 0.6204
Residuals 55 0.079055 0.00143736                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.98601 0.41758 0.5135
Y0_N   0.98899         0.50050 0.6134
Y0_NP  0.43028 0.48804         0.2078
Y0_P   0.53754 0.58847 0.21812       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001545 0.00051511 0.3563   1000 0.7692
Residuals 55 0.079513 0.00144569                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.91908 0.42058 0.7772
Y0_N   0.92688         0.53846 0.7243
Y0_NP  0.43456 0.53763         0.3666
Y0_P   0.76105 0.72350 0.35655       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.003245 0.0010818 0.4434   1000 0.7363
Residuals 55 0.134185 0.0024397                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.73526 0.57343 0.5694
Y0_N   0.74504         0.42957 0.7952
Y0_NP  0.57853 0.42122         0.2777
Y0_P   0.54959 0.79009 0.29774       </code></pre>
</div>
</section>
<section id="year-1" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.010681 0.0035603 0.8112   1000 0.4875
Residuals 41 0.179949 0.0043890                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.86214 0.83217 0.1558
Y1_N   0.86746         0.98002 0.2168
Y1_NP  0.82766 0.98105         0.1219
Y1_P   0.15498 0.21037 0.14153       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.0035194 0.00117312 1.5584   1000 0.1908
Residuals 41 0.0308637 0.00075277                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.753247 0.573427 0.0210
Y1_N   0.750591          0.857143 0.1019
Y1_NP  0.578617 0.850890          0.0759
Y1_P   0.035108 0.101561 0.082945       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.004833 0.0016111 1.1797   1000 0.3506
Residuals 41 0.055992 0.0013656                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.791209 0.889111 0.1269
Y1_N   0.793654          0.686314 0.1009
Y1_NP  0.864149 0.670396          0.2308
Y1_P   0.132720 0.081015 0.218505       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.002956 0.00098522 0.8277   1000 0.4575
Residuals 41 0.048804 0.00119034                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.871129 0.900100 0.0829
Y1_N   0.870619          0.951049 0.2797
Y1_NP  0.911412 0.938894          0.1269
Y1_P   0.097781 0.267965 0.136669       </code></pre>
</div>
</section>
<section id="year-4" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.007562 0.0025207 1.0355   1000 0.3846
Residuals 56 0.136320 0.0024343                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.12288 0.85814 0.7203
Y4_N   0.12548         0.12388 0.2847
Y4_NP  0.88522 0.12368         0.6004
Y4_P   0.70583 0.27336 0.62617       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.002451 0.00081708 1.1544   1000 0.3417
Residuals 56 0.039638 0.00070782                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.084915 0.742258 0.3916
Y4_N   0.088254          0.190809 0.3337
Y4_NP  0.743474 0.181827          0.6553
Y4_P   0.405181 0.317626 0.644142       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.002888 0.00096264 1.2211   1000 0.3277
Residuals 56 0.044146 0.00078831                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.11289 0.90709 0.2158
Y4_N   0.12838         0.18981 0.8172
Y4_NP  0.91438 0.16796         0.2787
Y4_P   0.21642 0.80681 0.26912       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.003484 0.0011612 1.1213   1000 0.3457
Residuals 56 0.057991 0.0010356                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.091908 0.818182 0.4535
Y4_N   0.082806          0.158841 0.3427
Y4_NP  0.819199 0.169883          0.6234
Y4_P   0.443405 0.350764 0.628363       </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Beta Dispersion &amp; Permutation tests <strong>(Arbitrary filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-1" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.005369 0.0017897 0.4269   1000 0.7353
Residuals 55 0.230580 0.0041924                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.87812 0.43457 0.7602
Y0_N   0.89772         0.51449 0.6563
Y0_NP  0.40946 0.51951         0.2647
Y0_P   0.74769 0.67538 0.28995       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.002753 0.00091771 0.4281   1000 0.7463
Residuals 55 0.117899 0.00214361                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.68432 0.24276 0.9710
Y0_N   0.67755         0.52248 0.7752
Y0_NP  0.24987 0.51127         0.3786
Y0_P   0.96776 0.75996 0.37334       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001427 0.00047582 0.3628   1000 0.8172
Residuals 55 0.072141 0.00131165                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.91908 0.47453 0.7143
Y0_N   0.91877         0.63636 0.6643
Y0_NP  0.46518 0.61418         0.3347
Y0_P   0.67632 0.66069 0.34161       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df  Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.00782 0.0026065 0.5921   1000 0.6234
Residuals 55 0.24211 0.0044019                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.68032 0.23077 0.8601
Y0_N   0.69120         0.47852 0.6374
Y0_NP  0.23170 0.48032         0.2547
Y0_P   0.87070 0.62143 0.24385       </code></pre>
</div>
</section>
<section id="year-1-1" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.013593 0.004531 1.0589   1000 0.3826
Residuals 41 0.175440 0.004279                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.82218 0.89710 0.1449
Y1_N   0.82000         0.91908 0.1129
Y1_NP  0.88374 0.92444         0.1159
Y1_P   0.12720 0.12736 0.11270       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.011112 0.0037040 1.5444   1000 0.2078
Residuals 41 0.098335 0.0023984                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.769231 0.759241 0.0290
Y1_N   0.753755          0.973027 0.1229
Y1_NP  0.738787 0.980841          0.0480
Y1_P   0.039336 0.109807 0.052132       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df  Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.00382 0.0012734 1.2678   1000 0.3207
Residuals 41 0.04118 0.0010044                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.998002 0.774226 0.0889
Y1_N   0.997972          0.777223 0.0889
Y1_NP  0.772727 0.770179          0.1439
Y1_P   0.080928 0.087798 0.138533       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.016031 0.0053437 1.448   1000 0.2078
Residuals 41 0.151307 0.0036904                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.852148 0.829171 0.0450
Y1_N   0.850930          0.997003 0.1099
Y1_NP  0.823840 0.998602          0.0529
Y1_P   0.052966 0.118749 0.059438       </code></pre>
</div>
</section>
<section id="year-4-1" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.006059 0.0020197 0.8192   1000 0.4755
Residuals 56 0.138063 0.0024654                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.18482 0.88611 0.5774
Y4_N   0.19735         0.17383 0.4615
Y4_NP  0.89006 0.17362         0.4885
Y4_P   0.56508 0.47497 0.49321       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.004185 0.0013951 1.127   1000 0.3626
Residuals 56 0.069323 0.0012379                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.13487 0.93706 0.2507
Y4_N   0.13581         0.20180 0.7173
Y4_NP  0.94091 0.18151         0.3087
Y4_P   0.24983 0.71551 0.31164       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.002699 0.00089976 1.1671   1000 0.3127
Residuals 56 0.043171 0.00077092                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.11688 0.73726 0.1728
Y4_N   0.12288         0.23477 0.8721
Y4_NP  0.74596 0.23568         0.3087
Y4_P   0.17748 0.86141 0.31697       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.006748 0.0022493 0.9607   1000 0.4326
Residuals 56 0.131111 0.0023413                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.21079 0.95804 0.3886
Y4_N   0.19007         0.18881 0.6923
Y4_NP  0.94937 0.18201         0.3576
Y4_P   0.35230 0.66788 0.33393       </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Beta Dispersion &amp; Permutation tests <strong>(PERFect filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-2" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.005247 0.0017491 0.409   1000 0.7542
Residuals 55 0.235190 0.0042762                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.87213 0.44156 0.7313
Y0_N   0.87878         0.57742 0.6314
Y0_NP  0.43664 0.56006         0.3087
Y0_P   0.73162 0.64439 0.30136       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.004295 0.0014318 0.4917   1000 0.6983
Residuals 55 0.160151 0.0029119                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.76324 0.28272 0.8462
Y0_N   0.75609         0.47952 0.6823
Y0_NP  0.28859 0.48440         0.2937
Y0_P   0.86455 0.67243 0.29495       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001854 0.00061797 0.6695   1000 0.5814
Residuals 55 0.050764 0.00092299                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.54545 0.26673 0.6833
Y0_N   0.56388         0.69630 0.4346
Y0_NP  0.25962 0.69751         0.2428
Y0_P   0.71144 0.44609 0.23450       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.009996 0.0033321 0.5884   1000 0.6194
Residuals 55 0.311455 0.0056628                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.62138 0.25974 0.8521
Y0_N   0.65374         0.51748 0.5495
Y0_NP  0.25318 0.52379         0.2348
Y0_P   0.85212 0.57641 0.25315       </code></pre>
</div>
</section>
<section id="year-1-2" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.012494 0.0041645 0.9717   1000 0.4386
Residuals 41 0.175724 0.0042859                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.84515 0.89810 0.1548
Y1_N   0.83032         0.94106 0.1688
Y1_NP  0.87992 0.93896         0.1309
Y1_P   0.14006 0.14907 0.13010       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.008192 0.0027307 1.463   1000 0.2537
Residuals 41 0.076527 0.0018665                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.840160 0.730270 0.0509
Y1_N   0.822690          0.896104 0.1169
Y1_NP  0.700634 0.896088          0.0729
Y1_P   0.054915 0.100038 0.066617       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.005362 0.0017872 1.2533   1000 0.3167
Residuals 41 0.058464 0.0014260                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.837163 0.687313 0.1229
Y1_N   0.845842          0.539461 0.0739
Y1_NP  0.693412 0.526539          0.2567
Y1_P   0.132596 0.071823 0.232893       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.012468 0.0041559 1.3603   1000 0.2757
Residuals 41 0.125261 0.0030552                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.882118 0.858142 0.0569
Y1_N   0.891497          0.975025 0.1159
Y1_NP  0.851410 0.979297          0.0779
Y1_P   0.066345 0.126274 0.073423       </code></pre>
</div>
</section>
<section id="year-4-2" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.005384 0.0017948 0.6846   1000 0.5654
Residuals 56 0.146816 0.0026217                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.24675 0.91309 0.5305
Y4_N   0.24519         0.21279 0.5914
Y4_NP  0.91756 0.21997         0.4765
Y4_P   0.52745 0.59803 0.47446       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.003907 0.0013023 0.7423   1000 0.5285
Residuals 56 0.098255 0.0017545                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.26074 0.88412 0.4116
Y4_N   0.27636         0.25075 0.7762
Y4_NP  0.86058 0.22954         0.3566
Y4_P   0.42052 0.77310 0.34962       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.002621 0.00087382 1.2431   1000 0.2947
Residuals 56 0.039364 0.00070293                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.14186 0.75724 0.1389
Y4_N   0.12376         0.23876 0.9570
Y4_NP  0.75630 0.22382         0.2657
Y4_P   0.15851 0.95346 0.26958       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.009921 0.0033070 0.903   1000 0.4396
Residuals 56 0.205095 0.0036624                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.21878 0.95804 0.3796
Y4_N   0.19379         0.17582 0.6703
Y4_NP  0.95432 0.19014         0.3546
Y4_P   0.37728 0.66891 0.36357       </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Beta Dispersion &amp; Permutation tests <strong>(PIME filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-3" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.000822 0.00027386 0.1041   1000   0.95
Residuals 55 0.144643 0.00262986                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.96903 0.99600 0.6164
Y0_N   0.97014         0.97403 0.7003
Y0_NP  0.99720 0.97074         0.6763
Y0_P   0.63862 0.69016 0.66682       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.000257 0.00008575 0.0333   1000  0.992
Residuals 55 0.141488 0.00257251                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.96803 0.85914 0.8721
Y0_N   0.96898         0.90909 0.8761
Y0_NP  0.85167 0.89688         0.7872
Y0_P   0.88621 0.87344 0.76329       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.000477 0.00015903 0.1927   1000 0.9001
Residuals 55 0.045382 0.00082513                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.82717 0.52547 0.9351
Y0_N   0.83516         0.67433 0.7992
Y0_NP  0.52903 0.69266         0.5005
Y0_P   0.93577 0.79698 0.52499       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df  Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.00003 0.0000093 0.0012   1000      1
Residuals 55 0.43346 0.0078812                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP  Y0_P
Y0_CTL         0.98402 0.99201 0.977
Y0_N   0.98602         0.99800 0.962
Y0_NP  0.98997 0.99621         0.961
Y0_P   0.96874 0.95894 0.96207      </code></pre>
</div>
</section>
<section id="year-1-3" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.007343 0.0024475 0.7443   1000 0.5225
Residuals 41 0.134830 0.0032885                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.38861 0.69031 0.4825
Y1_N   0.37744         0.67333 0.1548
Y1_NP  0.67664 0.67413         0.3656
Y1_P   0.46710 0.16425 0.35423       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.006654 0.0022179 0.515   1000 0.7463
Residuals 41 0.176570 0.0043066                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.67832 0.47453 0.7113
Y1_N   0.60257         0.89610 0.4835
Y1_NP  0.39659 0.88225         0.1738
Y1_P   0.61558 0.40756 0.19054       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.006473 0.0021577 1.5029   1000 0.2258
Residuals 41 0.058864 0.0014357                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.861139 0.792208 0.0819
Y1_N   0.869649          0.653347 0.0599
Y1_NP  0.774935 0.644776          0.1359
Y1_P   0.083788 0.063945 0.125344       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df  Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.01557 0.0051893 0.5637   1000 0.6933
Residuals 41 0.37741 0.0092050                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.64436 0.47952 0.6334
Y1_N   0.61704         0.86214 0.4296
Y1_NP  0.41514 0.81793         0.2278
Y1_P   0.58657 0.37468 0.19668       </code></pre>
</div>
</section>
<section id="year-4-3" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001641 0.0005469 0.3452   1000 0.7802
Residuals 56 0.088715 0.0015842                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.34266 0.72428 0.4196
Y4_N   0.32852         0.56543 0.8611
Y4_NP  0.74545 0.58140         0.6953
Y4_P   0.43099 0.86058 0.69956       

#################################################### 
BETA DISPERSION significance test unifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.000049 0.00001636 0.0115   1000  0.996
Residuals 56 0.079854 0.00142596                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.92208 0.86314 0.9870
Y4_N   0.93018         0.94605 0.9181
Y4_NP  0.87561 0.95591         0.8731
Y4_P   0.98992 0.92460 0.87312       

#################################################### 
BETA DISPERSION significance test wunifrac distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.002733 0.00091088 1.4436   1000 0.2418
Residuals 56 0.035334 0.00063097                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.070929 0.852148 0.3097
Y4_N   0.059554          0.093906 0.5025
Y4_NP  0.836950 0.095712          0.4156
Y4_P   0.312477 0.505584 0.408411       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df  Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.00001 0.0000032 0.0008   1000      1
Residuals 56 0.22474 0.0040132                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP  Y4_P
Y4_CTL         0.99101 0.99700 0.963
Y4_N   0.99104         0.99201 0.974
Y4_NP  0.99715 0.99385         0.968
Y4_P   0.96540 0.97614 0.96872      </code></pre>
</div>
</section>
</div>
</div>
</details>
<p>Remember, if the beta dispersion p-value is greater than <code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>ssu_perm_pvalues <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (d <span class="cf">in</span> ssu_dist){</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>          tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_permutest_&quot;</span>, d)))</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>          tmp_pv <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>tab<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>          tmp_test <span class="ot">&lt;-</span> <span class="cf">if</span>(tmp_pv <span class="sc">&lt;</span> <span class="fl">0.05</span>){</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>            tmp_test <span class="ot">&lt;-</span> <span class="st">&quot;ANOSIM&quot;</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>            tmp_test <span class="ot">&lt;-</span> <span class="st">&quot;ADONIS&quot;</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>          tmp_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(i, d, tmp_pv, tmp_test)</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>          ssu_perm_pvalues <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(ssu_perm_pvalues, tmp_df)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>ssu_perm_pvalues <span class="ot">&lt;-</span> ssu_perm_pvalues <span class="sc">%&gt;%</span> </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="fu">c</span>(<span class="st">&quot;data_set&quot;</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">&quot;dist&quot;</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">&quot;pval&quot;</span> <span class="ot">=</span> <span class="dv">3</span>, <span class="st">&quot;test&quot;</span> <span class="ot">=</span> <span class="dv">4</span>))</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>ssu_perm_pvalues <span class="ot">&lt;-</span> ssu_perm_pvalues <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(16S rRNA) Table 1 | Summary of Beta Dispersion Tests for unfiltered &amp; filtered data sets against various distance metrics. The <code>test</code> column shows the method chosen (based on the test results) to assess differences in beta diversity. If p-values are significant (red, p-value &lt; 0.05) we used ANOSIM, otherwise we used PERMANOVA (blue, p-value &gt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGF0YXNldCx5ZWFyLGRpc3QscHZhbCx0ZXN0CkZVTEwsWWVhciAwLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjc3NjIsQURPTklTCkZVTEwsWWVhciAwLFVud2VpZ2h0ZWQgVW5pRnJhYywwLjYyMDQsQURPTklTCkZVTEwsWWVhciAwLFdlaWdodGVkIFVuaUZyYWMsMC43NjkyLEFET05JUwpGVUxMLFllYXIgMCxKYWNjYXJkIGluZGV4LDAuNzM2MyxBRE9OSVMKRlVMTCxZZWFyIDEsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuNDg3NSxBRE9OSVMKRlVMTCxZZWFyIDEsVW53ZWlnaHRlZCBVbmlGcmFjLDAuMTkwOCxBRE9OSVMKRlVMTCxZZWFyIDEsV2VpZ2h0ZWQgVW5pRnJhYywwLjM1MDYsQURPTklTCkZVTEwsWWVhciAxLEphY2NhcmQgaW5kZXgsMC40NTc1LEFET05JUwpGVUxMLFllYXIgNCxKZW5zZW7igJNTaGFubm9uIGRpdmVyZ2VuY2UsMC4zODQ2LEFET05JUwpGVUxMLFllYXIgNCxVbndlaWdodGVkIFVuaUZyYWMsMC4zNDE3LEFET05JUwpGVUxMLFllYXIgNCxXZWlnaHRlZCBVbmlGcmFjLDAuMzI3NyxBRE9OSVMKRlVMTCxZZWFyIDQsSmFjY2FyZCBpbmRleCwwLjM0NTcsQURPTklTCkZJTFQsWWVhciAwLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjczNTMsQURPTklTCkZJTFQsWWVhciAwLFVud2VpZ2h0ZWQgVW5pRnJhYywwLjc0NjMsQURPTklTCkZJTFQsWWVhciAwLFdlaWdodGVkIFVuaUZyYWMsMC44MTcyLEFET05JUwpGSUxULFllYXIgMCxKYWNjYXJkIGluZGV4LDAuNjIzNCxBRE9OSVMKRklMVCxZZWFyIDEsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuMzgyNixBRE9OSVMKRklMVCxZZWFyIDEsVW53ZWlnaHRlZCBVbmlGcmFjLDAuMjA3OCxBRE9OSVMKRklMVCxZZWFyIDEsV2VpZ2h0ZWQgVW5pRnJhYywwLjMyMDcsQURPTklTCkZJTFQsWWVhciAxLEphY2NhcmQgaW5kZXgsMC4yMDc4LEFET05JUwpGSUxULFllYXIgNCxKZW5zZW7igJNTaGFubm9uIGRpdmVyZ2VuY2UsMC40NzU1LEFET05JUwpGSUxULFllYXIgNCxVbndlaWdodGVkIFVuaUZyYWMsMC4zNjI2LEFET05JUwpGSUxULFllYXIgNCxXZWlnaHRlZCBVbmlGcmFjLDAuMzEyNyxBRE9OSVMKRklMVCxZZWFyIDQsSmFjY2FyZCBpbmRleCwwLjQzMjYsQURPTklTClBFUkZlY3QsWWVhciAwLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjc1NDIsQURPTklTClBFUkZlY3QsWWVhciAwLFVud2VpZ2h0ZWQgVW5pRnJhYywwLjY5ODMsQURPTklTClBFUkZlY3QsWWVhciAwLFdlaWdodGVkIFVuaUZyYWMsMC41ODE0LEFET05JUwpQRVJGZWN0LFllYXIgMCxKYWNjYXJkIGluZGV4LDAuNjE5NCxBRE9OSVMKUEVSRmVjdCxZZWFyIDEsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuNDM4NixBRE9OSVMKUEVSRmVjdCxZZWFyIDEsVW53ZWlnaHRlZCBVbmlGcmFjLDAuMjUzNyxBRE9OSVMKUEVSRmVjdCxZZWFyIDEsV2VpZ2h0ZWQgVW5pRnJhYywwLjMxNjcsQURPTklTClBFUkZlY3QsWWVhciAxLEphY2NhcmQgaW5kZXgsMC4yNzU3LEFET05JUwpQRVJGZWN0LFllYXIgNCxKZW5zZW7igJNTaGFubm9uIGRpdmVyZ2VuY2UsMC41NjU0LEFET05JUwpQRVJGZWN0LFllYXIgNCxVbndlaWdodGVkIFVuaUZyYWMsMC41Mjg1LEFET05JUwpQRVJGZWN0LFllYXIgNCxXZWlnaHRlZCBVbmlGcmFjLDAuMjk0NyxBRE9OSVMKUEVSRmVjdCxZZWFyIDQsSmFjY2FyZCBpbmRleCwwLjQzOTYsQURPTklTClBJTUUsWWVhciAwLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjk1LEFET05JUwpQSU1FLFllYXIgMCxVbndlaWdodGVkIFVuaUZyYWMsMC45OTIsQURPTklTClBJTUUsWWVhciAwLFdlaWdodGVkIFVuaUZyYWMsMC45MDAxLEFET05JUwpQSU1FLFllYXIgMCxKYWNjYXJkIGluZGV4LDEsQURPTklTClBJTUUsWWVhciAxLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjUyMjUsQURPTklTClBJTUUsWWVhciAxLFVud2VpZ2h0ZWQgVW5pRnJhYywwLjc0NjMsQURPTklTClBJTUUsWWVhciAxLFdlaWdodGVkIFVuaUZyYWMsMC4yMjU4LEFET05JUwpQSU1FLFllYXIgMSxKYWNjYXJkIGluZGV4LDAuNjkzMyxBRE9OSVMKUElNRSxZZWFyIDQsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuNzgwMixBRE9OSVMKUElNRSxZZWFyIDQsVW53ZWlnaHRlZCBVbmlGcmFjLDAuOTk2LEFET05JUwpQSU1FLFllYXIgNCxXZWlnaHRlZCBVbmlGcmFjLDAuMjQxOCxBRE9OSVMKUElNRSxZZWFyIDQsSmFjY2FyZCBpbmRleCwxLEFET05JUwo=" download="ssu_beta_disp_tests.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-6bdbed04d4d1cb372320" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6bdbed04d4d1cb372320">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"dataset":["FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME"],"year":["Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4"],"dist":["Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index"],"pval":[0.7762,0.6204,0.7692,0.7363,0.4875,0.1908,0.3506,0.4575,0.3846,0.3417,0.3277,0.3457,0.7353,0.7463,0.8172,0.6234,0.3826,0.2078,0.3207,0.2078,0.4755,0.3626,0.3127,0.4326,0.7542,0.6983,0.5814,0.6194,0.4386,0.2537,0.3167,0.2757,0.5654,0.5285,0.2947,0.4396,0.95,0.992,0.9001,1,0.5225,0.7463,0.2258,0.6933,0.7802,0.996,0.2418,1],"test":["ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS"]},"columns":[{"accessor":"dataset","name":"Dataset","type":"factor","sortable":true,"resizable":true,"filterable":true,"cell":["FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME"],"header":"Dataset","minWidth":75,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"year","name":"year","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4"],"header":"year","minWidth":75,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"dist","name":"distance","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index","Jensen–Shannon divergence","Unweighted UniFrac","Weighted UniFrac","Jaccard index"],"header":"distance","minWidth":150,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pval","name":"p-value","type":"numeric","sortable":true,"resizable":false,"filterable":false,"cell":["0.7762","0.6204","0.7692","0.7363","0.4875","0.1908","0.3506","0.4575","0.3846","0.3417","0.3277","0.3457","0.7353","0.7463","0.8172","0.6234","0.3826","0.2078","0.3207","0.2078","0.4755","0.3626","0.3127","0.4326","0.7542","0.6983","0.5814","0.6194","0.4386","0.2537","0.3167","0.2757","0.5654","0.5285","0.2947","0.4396","0.9500","0.9920","0.9001","1.0000","0.5225","0.7463","0.2258","0.6933","0.7802","0.9960","0.2418","1.0000"],"header":"p-value","minWidth":75,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"test","name":"test","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS"],"header":"test","minWidth":75,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":12,"showPageSizeOptions":true,"pageSizeOptions":[4,12,24,48],"paginationType":"numbers","showPagination":true,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"ab3344b2a38fca7ba464f2f12cd74cf0"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="significance-tests">Significance Tests</h2>
<p>Depending on the test, we need different data structures. For <code>adonis</code> we need the sample metadata and for <code>anosim</code> we start with a phyloseq object.</p>
<p>We begin by creating separate sample metadata data frames for each phyloseq object (for any <code>adonis</code> analyses).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sampledf&quot;</span>))</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_samp)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_sampledf&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Next we can use a <code>for</code> loop with an <code>if/else</code> to perform significance tests. This loop does the following for every combination of phyloseq object and distance metric:</p>
<ol type="1">
<li>Get the results of each permutation test.</li>
<li>Get the p-value from the test.</li>
<li>If the p-value is &lt; 0.05, <code>anosim</code> is run, otherwise we use <code>adonis</code>.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (d <span class="cf">in</span> ssu_dist){</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>          tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_permutest_&quot;</span>, d)))</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>          tmp_pv <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>tab<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(tmp_pv <span class="sc">&lt;</span> <span class="fl">0.05</span>){</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>            tmp_anosim_data <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>            tmp_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(tmp_anosim_data, choose_group)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>            tmp_prop <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>            tmp_anosim_d <span class="ot">&lt;-</span> <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_prop, d), <span class="at">grouping =</span> tmp_groups)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>            tmp_anosim_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sig_anosim_&quot;</span>, d))</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>            <span class="fu">assign</span>(tmp_anosim_name, tmp_anosim_d)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>            tmp_adonis_data <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_beta_dist_&quot;</span>, d)))</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>            tmp_adonis_sampledf <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sampledf&quot;</span>)))</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>            tmp_adonis <span class="ot">&lt;-</span> <span class="fu">adonis</span>(tmp_adonis_data <span class="sc">~</span> YR_TREAT,</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> tmp_adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>            tmp_adonis2 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(tmp_adonis_data <span class="sc">~</span> YR_TREAT,</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> tmp_adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>            tmp_adonis_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sig_adonis_&quot;</span>, d))</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>            tmp_adonis2_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sig_adonis2_&quot;</span>, d))</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>            <span class="fu">assign</span>(tmp_adonis_name, tmp_adonis)</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>            <span class="fu">assign</span>(tmp_adonis2_name, tmp_adonis2)</span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<summary>
Results of Significance tests <strong>(FULL data set)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-4" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3    0.754 0.04319 0.8275  0.967
Residual 55   16.705 0.95681              
Total    58   17.459 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.08082 0.02853 0.5383 0.9481
Residual 55  2.75247 0.97147              
Total    58  2.83329 1.00000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.3861 0.04238 0.8113  0.975
Residual 55   8.7248 0.95762              
Total    58   9.1109 1.00000              


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2    F Pr(&gt;F)
YR_TREAT  3   0.0586 0.03373 0.64  0.953
Residual 55   1.6786 0.96627            
Total    58   1.7372 1.00000            </code></pre>
</div>
</section>
<section id="year-1-4" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.0141 0.06392 0.9332 0.8511
Residual 41  14.8509 0.93608              
Total    44  15.8650 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.12468 0.04579 0.6558 0.8961
Residual 41  2.59808 0.95421              
Total    44  2.72275 1.00000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.5676 0.06286 0.9168 0.8901
Residual 41   8.4616 0.93714              
Total    44   9.0292 1.00000              


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.09284 0.05604 0.8113 0.7672
Residual 41  1.56392 0.94396              
Total    44  1.65676 1.00000              </code></pre>
</div>
</section>
<section id="year-4-4" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2     F Pr(&gt;F)
YR_TREAT  3   0.9493 0.0509 1.001 0.3726
Residual 56  17.7025 0.9491             
Total    59  18.6518 1.0000             


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F Pr(&gt;F)
YR_TREAT  3  0.13298 0.0522 1.0281 0.3916
Residual 56  2.41451 0.9478              
Total    59  2.54750 1.0000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.5249 0.05124 1.0082 0.3916
Residual 56   9.7192 0.94876              
Total    59  10.2442 1.00000              


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.08091 0.05237 1.0316 0.3586
Residual 56  1.46405 0.94763              
Total    59  1.54496 1.00000              </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Significance tests <strong>(Arbitrary filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-5" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.4024 0.03569 0.6785  0.952
Residual 55  10.8735 0.96431              
Total    58  11.2759 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.04882 0.02448 0.4601 0.9051
Residual 55  1.94550 0.97552              
Total    58  1.99433 1.00000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F Pr(&gt;F)
YR_TREAT  3   0.1491 0.03591 0.683 0.9461
Residual 55   4.0036 0.96409             
Total    58   4.1527 1.00000             


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.03791 0.03497 0.6644 0.8881
Residual 55  1.04612 0.96503              
Total    58  1.08404 1.00000              </code></pre>
</div>
</section>
<section id="year-1-5" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.5265 0.05732 0.8311 0.8052
Residual 41   8.6573 0.94268              
Total    44   9.1838 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.06543 0.03704 0.5257 0.9101
Residual 41  1.70112 0.96296              
Total    44  1.76655 1.00000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F Pr(&gt;F)
YR_TREAT  3   0.2019 0.0574 0.8322 0.7463
Residual 41   3.3156 0.9426              
Total    44   3.5175 1.0000              


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F Pr(&gt;F)
YR_TREAT  3  0.03386 0.05044 0.726 0.8062
Residual 41  0.63746 0.94956             
Total    44  0.67132 1.00000             </code></pre>
</div>
</section>
<section id="year-4-5" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.5481 0.04882 0.9581 0.4665
Residual 56  10.6794 0.95118              
Total    59  11.2275 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.08735 0.05286 1.0418 0.4016
Residual 56  1.56509 0.94714              
Total    59  1.65244 1.00000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.1945 0.04675 0.9154 0.5654
Residual 56   3.9661 0.95325              
Total    59   4.1605 1.00000              


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.04152 0.04493 0.8782 0.5425
Residual 56  0.88241 0.95507              
Total    59  0.92392 1.00000              </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Significance tests <strong>(PERFect filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-6" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3    0.754 0.04319 0.8275  0.967
Residual 55   16.705 0.95681              
Total    58   17.459 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.08082 0.02853 0.5383 0.9481
Residual 55  2.75247 0.97147              
Total    58  2.83329 1.00000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.3861 0.04238 0.8113  0.975
Residual 55   8.7248 0.95762              
Total    58   9.1109 1.00000              


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2    F Pr(&gt;F)
YR_TREAT  3   0.0586 0.03373 0.64  0.953
Residual 55   1.6786 0.96627            
Total    58   1.7372 1.00000            </code></pre>
</div>
</section>
<section id="year-1-6" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.0141 0.06392 0.9332 0.8511
Residual 41  14.8509 0.93608              
Total    44  15.8650 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.12468 0.04579 0.6558 0.8961
Residual 41  2.59808 0.95421              
Total    44  2.72275 1.00000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.5676 0.06286 0.9168 0.8901
Residual 41   8.4616 0.93714              
Total    44   9.0292 1.00000              


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.09284 0.05604 0.8113 0.7672
Residual 41  1.56392 0.94396              
Total    44  1.65676 1.00000              </code></pre>
</div>
</section>
<section id="year-4-6" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2     F Pr(&gt;F)
YR_TREAT  3   0.9493 0.0509 1.001 0.3726
Residual 56  17.7025 0.9491             
Total    59  18.6518 1.0000             


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F Pr(&gt;F)
YR_TREAT  3  0.13298 0.0522 1.0281 0.3916
Residual 56  2.41451 0.9478              
Total    59  2.54750 1.0000              


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.5249 0.05124 1.0082 0.3916
Residual 56   9.7192 0.94876              
Total    59  10.2442 1.00000              


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.08091 0.05237 1.0316 0.3586
Residual 56  1.46405 0.94763              
Total    59  1.54496 1.00000              </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Significance tests <strong>(PIME filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-7" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F   Pr(&gt;F)    
YR_TREAT  3   3.2482 0.82721 87.77 0.000999 ***
Residual 55   0.6785 0.17279                   
Total    58   3.9267 1.00000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3  0.30184 0.40112 12.279 0.000999 ***
Residual 55  0.45067 0.59888                    
Total    58  0.75251 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3  0.99668 0.82404 85.856 0.000999 ***
Residual 55  0.21283 0.17596                    
Total    58  1.20950 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)   
YR_TREAT  3  0.09800 0.14078 3.0038 0.002997 **
Residual 55  0.59813 0.85922                   
Total    58  0.69612 1.00000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</section>
<section id="year-1-7" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   3.7017 0.86907 90.711 0.000999 ***
Residual 41   0.5577 0.13093                    
Total    44   4.2594 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2     F   Pr(&gt;F)    
YR_TREAT  3  0.39798 0.5153 14.53 0.000999 ***
Residual 41  0.37435 0.4847                   
Total    44  0.77233 1.0000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3  1.28767 0.84582 74.975 0.000999 ***
Residual 41  0.23472 0.15418                    
Total    44  1.52239 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3  0.17614 0.22799 4.0361 0.000999 ***
Residual 41  0.59643 0.77201                    
Total    44  0.77257 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</section>
<section id="year-4-7" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   3.3442 0.88527 144.03 0.000999 ***
Residual 56   0.4334 0.11473                    
Total    59   3.7777 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3  0.25748 0.44989 15.266 0.000999 ***
Residual 56  0.31484 0.55011                    
Total    59  0.57231 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Unweighted UniFrac distance ( unifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F   Pr(&gt;F)    
YR_TREAT  3  1.25583 0.8939 157.26 0.000999 ***
Residual 56  0.14906 0.1061                    
Total    59  1.40489 1.0000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Weighted UniFrac distance ( wunifrac ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3  0.08322 0.15209 3.3482 0.000999 ***
Residual 56  0.46396 0.84791                    
Total    59  0.54718 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</section>
</div>
</div>
</details>
<h2 id="summaries">Summaries</h2>
<p>Here is a quick summary of significance tests for the data sets against three distance matrices.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="l-body">
<div class="panelset">
<section id="full-data-sets" class="panel">
<h4>FULL data sets</h4>
<p><small><strong>(16S rRNA) Table 2 | Summary of significant tests by Year for FULL, unfiltered data set. P-values in red indicate significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGlzdGFuY2UsWWVhciAwLFllYXIgMSxZZWFyIDQKSmFjY2FyZCBpbmRleCwwLjk2NywwLjg1MTEsMC4zNzI2CkplbnNlbi1TaGFubm9uIERpdmVyZ2VuY2UsMC45NDgxLDAuODk2MSwwLjM5MTYKVW53ZWlnaHRlZCBVbmlmcmFjLDAuOTc1LDAuODkwMSwwLjM5MTYKV2VpZ2h0ZWQgVW5pZnJhYywwLjk1MywwLjc2NzIsMC4zNTg2Cg==" download="ssu_tab_sig_test_full.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-6ae34aaabe18c8b6d848" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6ae34aaabe18c8b6d848">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance":["Jaccard index","Jensen-Shannon Divergence","Unweighted Unifrac","Weighted Unifrac"],"Year 0":[0.967,0.9481,0.975,0.953],"Year 1":[0.8511,0.8961,0.8901,0.7672],"Year 4":[0.3726,0.3916,0.3916,0.3586]},"columns":[{"accessor":"distance","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Jaccard index","Jensen-Shannon Divergence","Unweighted Unifrac","Weighted Unifrac"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Year 0","name":"Year 0","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.967","0.9481","0.975","0.953"],"header":"Year 0","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 1","name":"Year 1","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.8511","0.8961","0.8901","0.7672"],"header":"Year 1","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 4","name":"Year 4","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.3726","0.3916","0.3916","0.3586"],"header":"Year 4","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["Year 0","Year 1","Year 4"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"e294a4d8efe18587b2adf79d2fef711e"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
<section id="arbitrary-filtered" class="panel">
<h4>Arbitrary filtered</h4>
<p><small><strong>(16S rRNA) Table 3 | Summary of significant tests by Year for Arbitrary filtered data set. P-values in red indicate significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGlzdGFuY2UsWWVhciAwLFllYXIgMSxZZWFyIDQKSmFjY2FyZCBpbmRleCwwLjk1MiwwLjgwNTIsMC40NjY1CkplbnNlbi1TaGFubm9uIERpdmVyZ2VuY2UsMC45MDUxLDAuOTEwMSwwLjQwMTYKVW53ZWlnaHRlZCBVbmlmcmFjLDAuOTQ2MSwwLjc0NjMsMC41NjU0CldlaWdodGVkIFVuaWZyYWMsMC44ODgxLDAuODA2MiwwLjU0MjUK" download="ssu_tab_sig_test_filt.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-f9ebbf55e49feb2e5edc" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f9ebbf55e49feb2e5edc">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance":["Jaccard index","Jensen-Shannon Divergence","Unweighted Unifrac","Weighted Unifrac"],"Year 0":[0.952,0.9051,0.9461,0.8881],"Year 1":[0.8052,0.9101,0.7463,0.8062],"Year 4":[0.4665,0.4016,0.5654,0.5425]},"columns":[{"accessor":"distance","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Jaccard index","Jensen-Shannon Divergence","Unweighted Unifrac","Weighted Unifrac"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Year 0","name":"Year 0","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.952","0.9051","0.9461","0.8881"],"header":"Year 0","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 1","name":"Year 1","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.8052","0.9101","0.7463","0.8062"],"header":"Year 1","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 4","name":"Year 4","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.4665","0.4016","0.5654","0.5425"],"header":"Year 4","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["Year 0","Year 1","Year 4"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"7d2e9f0b11734b05d8d4be7bedf1eabf"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
<section id="perfect-filtered" class="panel">
<h4>PERFect filtered</h4>
<p><small><strong>(16S rRNA) Table 4 | Summary of significant tests by Year for PERFect filtered data set. P-values in red indicate significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGlzdGFuY2UsWWVhciAwLFllYXIgMSxZZWFyIDQKSmFjY2FyZCBpbmRleCwwLjk0OTEsMC44NDgyLDAuNjAwNApKZW5zZW4tU2hhbm5vbiBEaXZlcmdlbmNlLDAuODg5MSwwLjkwNjEsMC4zNzk2ClVud2VpZ2h0ZWQgVW5pZnJhYywwLjk2LDAuODMzMiwwLjU1MzQKV2VpZ2h0ZWQgVW5pZnJhYywwLjk0MzEsMC44NzYxLDAuNDY0NQo=" download="ssu_tab_sig_test_perfect.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-2b3a3be873d5658295db" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2b3a3be873d5658295db">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance":["Jaccard index","Jensen-Shannon Divergence","Unweighted Unifrac","Weighted Unifrac"],"Year 0":[0.9491,0.8891,0.96,0.9431],"Year 1":[0.8482,0.9061,0.8332,0.8761],"Year 4":[0.6004,0.3796,0.5534,0.4645]},"columns":[{"accessor":"distance","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Jaccard index","Jensen-Shannon Divergence","Unweighted Unifrac","Weighted Unifrac"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Year 0","name":"Year 0","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.9491","0.8891","0.960","0.9431"],"header":"Year 0","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 1","name":"Year 1","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.8482","0.9061","0.8332","0.8761"],"header":"Year 1","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 4","name":"Year 4","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.6004","0.3796","0.5534","0.4645"],"header":"Year 4","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["Year 0","Year 1","Year 4"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"e0b2aa90baaa6c5af82075473f9fbf70"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
<section id="pime-filtered" class="panel">
<h4>PIME filtered</h4>
<p><small><strong>(16S rRNA) Table 5 | Summary of significant tests by Year for PIME filtered data set. P-values in red indicate significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGlzdGFuY2UsWWVhciAwLFllYXIgMSxZZWFyIDQKSmFjY2FyZCBpbmRleCwwLjAwMSwwLjAwMSwwLjAwMQpKZW5zZW4tU2hhbm5vbiBEaXZlcmdlbmNlLDAuMDAxLDAuMDAxLDAuMDAxClVud2VpZ2h0ZWQgVW5pZnJhYywwLjAwMSwwLjAwMSwwLjAwMQpXZWlnaHRlZCBVbmlmcmFjLDAuMDAzLDAuMDAxLDAuMDAxCg==" download="ssu_tab_sig_test_pime.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-694007df96b2cb4e1d13" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-694007df96b2cb4e1d13">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance":["Jaccard index","Jensen-Shannon Divergence","Unweighted Unifrac","Weighted Unifrac"],"Year 0":[0.001,0.001,0.001,0.003],"Year 1":[0.001,0.001,0.001,0.001],"Year 4":[0.001,0.001,0.001,0.001]},"columns":[{"accessor":"distance","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Jaccard index","Jensen-Shannon Divergence","Unweighted Unifrac","Weighted Unifrac"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Year 0","name":"Year 0","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.003"],"header":"Year 0","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]},{"accessor":"Year 1","name":"Year 1","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"Year 1","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]},{"accessor":"Year 4","name":"Year 4","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"Year 4","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["Year 0","Year 1","Year 4"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"2d006092417cca6f3873c35165a09d54"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
</div>
</div>
<h1 id="its">ITS</h1>
<p>In order to test for significance between sample groups, we must perform the following steps:</p>
<ol type="1">
<li>Transform sample counts to relative abundance.</li>
<li>Create distance matrices based on some metrics, for example <code>wunifrac</code>, <code>unifrac</code>, and <code>jsd</code>. For this we use the function <code>phyloseq::distance</code>.</li>
<li>Next, calculate beta dispersion using the <code>betadisper</code> function from the <code>vegan</code> package <span class="citation" data-cites="oksanen2013community">(<a href="#ref-oksanen2013community" role="doc-biblioref">Oksanen et al. 2012</a>)</span>.</li>
<li>Then, use the function <code>permutest</code> to run a Permutation test for homogeneity of multivariate dispersions.</li>
<li>If the beta dispersion tests are not significant we will run a PERMANOVA using <code>adonis</code> (since PERMANOVA assumes equal dispersion), otherwise we will use Analysis of Similarity (ANOSIM), both available in the <code>vegan</code> package <span class="citation" data-cites="oksanen2013community">(<a href="#ref-oksanen2013community" role="doc-biblioref">Oksanen et al. 2012</a>)</span>.</li>
</ol>
<p>Steps 2–4 are combined in a <code>for</code> loop that tests all three distance metrics.</p>
<p>Again, transform sample counts to relative abundance and generate some new trees.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>its_samp_ps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;its_.*_Y[0-9]$&quot;</span>))</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>))</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(tmp_get, <span class="cf">function</span>(otu) </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>                                          <span class="fl">1e5</span> <span class="sc">*</span> otu<span class="sc">/</span><span class="fu">sum</span>(otu))</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>     tmp_ps<span class="sc">@</span>phy_tree <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_sums</span>(tmp_ps) <span class="sc">&gt;</span> <span class="dv">0</span>, tmp_ps)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>     tmp_ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tmp_ps, sample_data)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(tmp_name)</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_ps)</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<h2 id="beta-dispersion-1">Beta Dispersion</h2>
<p>Here we create distance matrices for each metric, calculate the beta dispersion, and run a permutation test for homogeneity of multivariate dispersions.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>its_dist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;jsd&quot;</span>, <span class="st">&quot;bray&quot;</span>, <span class="st">&quot;gower&quot;</span>, <span class="st">&quot;jaccard&quot;</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (d <span class="cf">in</span> its_dist){</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>       tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>       tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span>(d <span class="sc">==</span> <span class="st">&quot;jaccard&quot;</span>){</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>         tmp_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_get, <span class="at">method =</span> d, <span class="at">binary =</span> <span class="cn">TRUE</span>)</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>       } <span class="cf">else</span> {</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>         tmp_df <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_get, <span class="at">method =</span> d)</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>         }</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="co">#       tmp_df &lt;- phyloseq::distance(tmp_get, method = d)</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>       tmp_df_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dist_&quot;</span>, .))</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name, tmp_df)</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>       tmp_df2 <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(tmp_df, tmp_samp[[choose_group]], <span class="at">bias.adjust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>       tmp_df_name2 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_beta_dispersion_&quot;</span>, .))</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name2, tmp_df2)</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>       tmp_df3 <span class="ot">&lt;-</span> <span class="fu">permutest</span>(tmp_df2, <span class="at">pairwise =</span> <span class="cn">TRUE</span>,</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>                            <span class="at">permutations =</span> <span class="dv">1000</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>)</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>       tmp_df_name3 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(d, <span class="sc">~</span> <span class="fu">paste0</span>(i, <span class="st">&quot;_permutest_&quot;</span>, .))</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>       <span class="fu">assign</span>(tmp_df_name3, tmp_df3)</span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<summary>
Results of Beta Dispersion &amp; Permutation tests <strong>(FULL data set)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-8" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001654 0.00055145 0.6008   1000 0.6124
Residuals 55 0.050484 0.00091789                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.70430 0.42757 0.5115
Y0_N   0.69922         0.28771 0.3207
Y0_NP  0.42070 0.28079         0.8611
Y0_P   0.49947 0.33226 0.86631       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001797 0.00059892 0.6764   1000 0.5475
Residuals 55 0.048702 0.00088548                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.64635 0.41359 0.4406
Y0_N   0.66558         0.26573 0.2817
Y0_NP  0.39608 0.26070         0.8981
Y0_P   0.46061 0.30288 0.90021       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.0003246 0.00010819 0.6873   1000 0.5754
Residuals 55 0.0086586 0.00015743                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.58042 0.53447 0.6314
Y0_N   0.56597         0.25375 0.9590
Y0_NP  0.50300 0.23834         0.2927
Y0_P   0.60605 0.95922 0.28332       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.0006515 0.00021716 0.754   1000 0.5485
Residuals 55 0.0158418 0.00028803                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.85015 0.23776 0.8801
Y0_N   0.83679         0.10390 0.7153
Y0_NP  0.23163 0.11141         0.3347
Y0_P   0.87757 0.71314 0.33161       </code></pre>
</div>
</section>
<section id="year-1-8" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.000867 0.00028897 0.3946   1000 0.7942
Residuals 56 0.041007 0.00073226                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.85015 0.29970 0.9481
Y1_N   0.83528         0.45554 0.9061
Y1_NP  0.30371 0.45217         0.4146
Y1_P   0.95722 0.89564 0.40035       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.000874 0.00029133 0.463   1000 0.7193
Residuals 56 0.035237 0.00062923                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.87413 0.27473 0.9401
Y1_N   0.86904         0.40360 0.9411
Y1_NP  0.27187 0.39437         0.3746
Y1_P   0.93546 0.94294 0.37067       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.0005105 0.00017017 1.1519   1000 0.3207
Residuals 56 0.0082727 0.00014773                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.172827 0.989011 0.5015
Y1_N   0.178063          0.255744 0.0649
Y1_NP  0.993068 0.266141          0.5504
Y1_P   0.501716 0.079438 0.554559       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.0006062 0.00020206 0.8563   1000 0.4875
Residuals 56 0.0132146 0.00023598                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.73227 0.13287 0.8891
Y1_N   0.71783         0.36863 0.8152
Y1_NP  0.12345 0.34443         0.1708
Y1_P   0.88530 0.80996 0.16178       </code></pre>
</div>
</section>
<section id="year-4-8" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.003662 0.00122058 2.1132   1000 0.1219
Residuals 56 0.032345 0.00057759                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.323676 0.100899 0.5425
Y4_N   0.303419          0.032967 0.1469
Y4_NP  0.105648 0.030383          0.4006
Y4_P   0.532812 0.161683 0.429574       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     3 0.004469 0.0014898 2.4959   1000 0.06893 .
Residuals 56 0.033427 0.0005969                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.212787 0.090909 0.4436
Y4_N   0.234260          0.018981 0.1189
Y4_NP  0.091134 0.021316          0.4695
Y4_P   0.467005 0.112432 0.482470       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm   Pr(&gt;F)   
Groups     3 0.0010596 0.00035321 4.6998   1000 0.008991 **
Residuals 56 0.0042086 0.00007515                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          Y4_CTL      Y4_N     Y4_NP   Y4_P
Y4_CTL           0.2347652 0.0819181 0.0340
Y4_N   0.2309674           0.0059940 0.0050
Y4_NP  0.0870689 0.0041657           0.6074
Y4_P   0.0436390 0.0029297 0.5990723       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq     Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.0002889 0.000096311 0.3683   1000 0.7642
Residuals 56 0.0146443 0.000261506                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.85914 0.45754 0.8741
Y4_N   0.86158         0.32667 0.9880
Y4_NP  0.47347 0.31590         0.3906
Y4_P   0.88966 0.98614 0.39292       </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Beta Dispersion &amp; Permutation tests <strong>(Arbitrary filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-9" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm  Pr(&gt;F)  
Groups     3 0.016365 0.0054550 2.334   1000 0.08192 .
Residuals 55 0.128543 0.0023371                       
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y0_CTL     Y0_N    Y0_NP   Y0_P
Y0_CTL          0.036963 0.750250 0.9401
Y0_N   0.038225          0.033966 0.0579
Y0_NP  0.734143 0.031628          0.8042
Y0_P   0.942882 0.040995 0.790747       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)  
Groups     3 0.016574 0.0055247 2.0991   1000 0.0979 .
Residuals 55 0.144757 0.0026320                       
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y0_CTL     Y0_N    Y0_NP   Y0_P
Y0_CTL          0.052947 0.806194 0.9071
Y0_N   0.043966          0.034965 0.0320
Y0_NP  0.807456 0.047669          0.9291
Y0_P   0.882861 0.046035 0.920383       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     3 0.0046672 0.00155572 2.7542   1000 0.05594 .
Residuals 55 0.0310665 0.00056484                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          Y0_CTL      Y0_N     Y0_NP   Y0_P
Y0_CTL           0.0429570 0.1958042 0.9181
Y0_N   0.0407739           0.0019980 0.2028
Y0_NP  0.1995390 0.0014571           0.2527
Y0_P   0.9027490 0.1765820 0.2379865       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.007421 0.0024736 1.091   1000 0.3596
Residuals 55 0.124696 0.0022672                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y0_CTL     Y0_N    Y0_NP   Y0_P
Y0_CTL          0.562438 0.277722 0.7972
Y0_N   0.541930          0.065934 0.7582
Y0_NP  0.280553 0.075059          0.1538
Y0_P   0.777442 0.733934 0.153708       </code></pre>
</div>
</section>
<section id="year-1-9" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.004667 0.0015555 0.639   1000 0.5784
Residuals 56 0.136319 0.0024343                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.50350 0.44655 0.9181
Y1_N   0.48939         0.20080 0.5734
Y1_NP  0.46877 0.21248         0.4376
Y1_P   0.91122 0.56969 0.43470       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.0053 0.0017667 0.718   1000 0.5435
Residuals 56 0.1378 0.0024607                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.51648 0.38761 0.9451
Y1_N   0.50677         0.20979 0.5514
Y1_NP  0.39481 0.20113         0.3766
Y1_P   0.94712 0.55547 0.38758       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.003124 0.00104133 1.3243   1000 0.2967
Residuals 56 0.044034 0.00078633                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y1_CTL     Y1_N    Y1_NP   Y1_P
Y1_CTL          0.477522 0.220779 0.3776
Y1_N   0.483217          0.087912 0.1828
Y1_NP  0.214966 0.085741          0.8901
Y1_P   0.380437 0.175375 0.880386       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.004957 0.0016523 0.5988   1000 0.6034
Residuals 56 0.154537 0.0027596                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.79421 0.28771 0.8581
Y1_N   0.81727         0.29670 0.9530
Y1_NP  0.29620 0.28914         0.2727
Y1_P   0.85382 0.94863 0.26541       </code></pre>
</div>
</section>
<section id="year-4-9" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df  Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     3 0.01484 0.0049468 2.6666   1000 0.04695 *
Residuals 56 0.10389 0.0018551                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.030969 0.681319 0.6274
Y4_N   0.027735          0.041958 0.0390
Y4_NP  0.665321 0.031061          0.9261
Y4_P   0.641565 0.041024 0.938367       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     3 0.018794 0.0062646 3.1172   1000 0.03197 *
Residuals 56 0.112542 0.0020097                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.027972 0.667333 0.5554
Y4_N   0.022792          0.019980 0.0320
Y4_NP  0.676159 0.025202          0.8402
Y4_P   0.569667 0.028661 0.850818       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     3 0.011817 0.0039390 10.811   1000 0.000999 ***
Residuals 56 0.020404 0.0003644                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
            Y4_CTL        Y4_N       Y4_NP  Y4_P
Y4_CTL             0.235764236 0.094905095 0.003
Y4_N   0.216556582             0.000999001 0.001
Y4_NP  0.084029480 0.002056066             0.045
Y4_P   0.000865360 0.000013614 0.041288058      

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.004437 0.0014791 0.7877   1000 0.5005
Residuals 56 0.105148 0.0018776                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.39461 0.52947 0.8731
Y4_N   0.41109         0.12587 0.3057
Y4_NP  0.52631 0.12881         0.6044
Y4_P   0.88055 0.30493 0.60921       </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Beta Dispersion &amp; Permutation tests <strong>(PERFect filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-10" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.006412 0.0021374 0.9333   1000 0.4416
Residuals 55 0.125953 0.0022901                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.46553 0.43057 0.4535
Y0_N   0.46679         0.17982 0.1698
Y0_NP  0.42261 0.17922         0.8681
Y0_P   0.46103 0.17370 0.86244       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.006669 0.0022230 0.9192   1000 0.3986
Residuals 55 0.133016 0.0024185                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.46653 0.41558 0.4166
Y0_N   0.47553         0.19081 0.1588
Y0_NP  0.44521 0.19249         0.9191
Y0_P   0.43505 0.16397 0.92445       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq     F N.Perm  Pr(&gt;F)  
Groups     3 0.003018 0.00100601 2.506   1000 0.06793 .
Residuals 55 0.022080 0.00040145                       
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y0_CTL     Y0_N    Y0_NP   Y0_P
Y0_CTL          0.581419 0.052947 0.6114
Y0_N   0.540247          0.012987 0.3257
Y0_NP  0.051420 0.013321          0.2258
Y0_P   0.593373 0.323491 0.216879       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.007712 0.0025706 1.1035   1000 0.3586
Residuals 55 0.128121 0.0023295                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y0_CTL     Y0_N    Y0_NP   Y0_P
Y0_CTL          0.605395 0.237762 0.8032
Y0_N   0.605073          0.066933 0.8402
Y0_NP  0.235474 0.061909          0.1748
Y0_P   0.784801 0.847043 0.161290       </code></pre>
</div>
</section>
<section id="year-1-10" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001275 0.00042495 0.3261   1000 0.8122
Residuals 56 0.072964 0.00130293                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.91109 0.52248 0.7153
Y1_N   0.90191         0.60739 0.6653
Y1_NP  0.50394 0.61788         0.4036
Y1_P   0.71872 0.66322 0.37811       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001148 0.00038257 0.3498   1000 0.7792
Residuals 56 0.061254 0.00109382                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.88911 0.43357 0.7952
Y1_N   0.89888         0.57642 0.7483
Y1_NP  0.42950 0.54273         0.3756
Y1_P   0.80350 0.73449 0.36727       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.0008115 0.00027050 0.7978   1000 0.5135
Residuals 56 0.0189863 0.00033904                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.90709 0.28272 0.3207
Y1_N   0.90958         0.26873 0.3327
Y1_NP  0.25706 0.26157         0.8851
Y1_P   0.31290 0.32194 0.88007       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.002484 0.00082786 0.372   1000 0.7842
Residuals 56 0.124620 0.00222536                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.69231 0.55744 0.7682
Y1_N   0.70252         0.35365 0.9061
Y1_NP  0.54831 0.37320         0.3856
Y1_P   0.76826 0.88727 0.36022       </code></pre>
</div>
</section>
<section id="year-4-10" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     3 0.008489 0.0028298 2.4185   1000 0.07692 .
Residuals 56 0.065522 0.0011700                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.101898 0.278721 0.8082
Y4_N   0.103842          0.010989 0.1099
Y4_NP  0.261362 0.018590          0.5115
Y4_P   0.810756 0.114085 0.508795       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
Groups     3 0.009461 0.0031536 2.5675   1000 0.07193 .
Residuals 56 0.068784 0.0012283                        
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
         Y4_CTL     Y4_N    Y4_NP   Y4_P
Y4_CTL          0.086913 0.347652 0.8182
Y4_N   0.077402          0.012987 0.1109
Y4_NP  0.318852 0.017522          0.5884
Y4_P   0.820096 0.095298 0.588951       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq     F N.Perm   Pr(&gt;F)    
Groups     3 0.0018352 0.00061175 5.634   1000 0.000999 ***
Residuals 56 0.0060806 0.00010858                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
          Y4_CTL      Y4_N     Y4_NP   Y4_P
Y4_CTL           0.5944056 0.0199800 0.0090
Y4_N   0.6063545           0.0069930 0.0060
Y4_NP  0.0168272 0.0075225           0.8781
Y4_P   0.0072287 0.0031755 0.8770489       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq    Mean Sq    F N.Perm Pr(&gt;F)
Groups     3 0.002208 0.00073613 0.37   1000 0.7752
Residuals 56 0.111415 0.00198956                   

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.52348 0.72028 0.7592
Y4_N   0.51356         0.28971 0.7822
Y4_NP  0.74929 0.28580         0.5005
Y4_P   0.73226 0.77785 0.49115       </code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Beta Dispersion &amp; Permutation tests <strong>(PIME filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-11" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001088 0.0003627 0.1056   1000  0.963
Residuals 55 0.188892 0.0034344                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.74625 0.93407 0.8222
Y0_N   0.73007         0.83916 0.5974
Y0_NP  0.92131 0.83656         0.7393
Y0_P   0.81710 0.57087 0.75529       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.001571 0.0005237 0.1176   1000 0.9441
Residuals 55 0.244870 0.0044522                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.74725 0.75524 0.8691
Y0_N   0.75795         0.95005 0.6394
Y0_NP  0.74332 0.94619         0.6294
Y0_P   0.87126 0.64693 0.64153       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     3 0.0091388 0.00304626 13.648   1000 0.000999 ***
Residuals 55 0.0122765 0.00022321                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
             Y0_CTL         Y0_N        Y0_NP   Y0_P
Y0_CTL              0.0009990010 0.3226773227 0.0949
Y0_N   0.0001070407              0.0009990010 0.0020
Y0_NP  0.3411728262 0.0000022515              0.0050
Y0_P   0.0967061664 0.0031646756 0.0060367441       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df  Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.00610 0.0020336 0.2836   1000 0.8342
Residuals 55 0.39435 0.0071699                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y0_CTL    Y0_N   Y0_NP   Y0_P
Y0_CTL         0.68332 0.36963 0.7622
Y0_N   0.68206         0.59740 0.9381
Y0_NP  0.39155 0.60582         0.5724
Y0_P   0.74054 0.93081 0.54969       </code></pre>
</div>
</section>
<section id="year-1-11" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.000808 0.0002693 0.0556   1000  0.984
Residuals 56 0.271298 0.0048446                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.85714 0.64336 0.8821
Y1_N   0.85534         0.85015 0.9540
Y1_NP  0.65642 0.85956         0.7792
Y1_P   0.87678 0.95306 0.77455       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.000413 0.0001377 0.0259   1000  0.998
Residuals 56 0.298153 0.0053242                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.83317 0.77323 0.8881
Y1_N   0.84966         0.95904 0.9431
Y1_NP  0.77132 0.95686         0.8981
Y1_P   0.87991 0.94417 0.88326       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq    Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     3 0.0085902 0.00286342 9.0952   1000 0.000999 ***
Residuals 56 0.0176303 0.00031483                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
            Y1_CTL        Y1_N       Y1_NP   Y1_P
Y1_CTL             0.308691309 0.001998002 0.3237
Y1_N   0.313351121             0.000999001 0.0360
Y1_NP  0.000610784 0.000011737             0.0010
Y1_P   0.361095949 0.039215899 0.000911614       

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df  Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)
Groups     3 0.00510 0.0017001 0.157   1000 0.9291
Residuals 56 0.60656 0.0108314                    

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y1_CTL    Y1_N   Y1_NP   Y1_P
Y1_CTL         0.73626 0.78521 0.8142
Y1_N   0.72376         0.56044 0.8801
Y1_NP  0.76943 0.56368         0.6374
Y1_P   0.82248 0.87840 0.62896       </code></pre>
</div>
</section>
<section id="year-4-11" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
#################################################### 
BETA DISPERSION significance test jsd distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.009241 0.0030802 0.9655   1000 0.3866
Residuals 56 0.178660 0.0031904                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.41858 0.48152 0.5155
Y4_N   0.42466         0.15984 0.1548
Y4_NP  0.48571 0.17427         0.8551
Y4_P   0.52649 0.16478 0.87102       

#################################################### 
BETA DISPERSION significance test bray distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.013106 0.0043687 1.1958   1000 0.3187
Residuals 56 0.204586 0.0036533                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.32567 0.53746 0.5105
Y4_N   0.31096         0.17982 0.1169
Y4_NP  0.51245 0.14568         0.8991
Y4_P   0.53588 0.13269 0.90203       

#################################################### 
BETA DISPERSION significance test gower distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df    Sum Sq   Mean Sq      F N.Perm   Pr(&gt;F)    
Groups     3 0.0146799 0.0048933 29.218   1000 0.000999 ***
Residuals 56 0.0093786 0.0001675                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
                   Y4_CTL               Y4_N              Y4_NP
Y4_CTL                    0.0009990009990010 0.0179820179820180
Y4_N   0.0000158542360806                    0.0009990009990010
Y4_NP  0.0213642296673145 0.0000058465673024                   
Y4_P   0.0000005617246184 0.0000000000090329 0.0511003823700056
         Y4_P
Y4_CTL 0.0010
Y4_N   0.0010
Y4_NP  0.0559
Y4_P         

#################################################### 
BETA DISPERSION significance test jaccard distance 
####################################################

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 1000

Response: Distances
          Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
Groups     3 0.003549 0.0011831 0.2124   1000 0.8931
Residuals 56 0.311960 0.0055707                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above
diagonal)
        Y4_CTL    Y4_N   Y4_NP   Y4_P
Y4_CTL         0.99301 0.75824 0.6533
Y4_N   0.99261         0.73726 0.5974
Y4_NP  0.75804 0.73412         0.4066
Y4_P   0.66562 0.61635 0.46165       </code></pre>
</div>
</section>
</div>
</div>
</details>
<p>Remember, if the beta dispersion p-value is greater than <code>0.05</code> we use PERMANOVA, otherwise we use ANOSIM.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>its_perm_pvalues <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (d <span class="cf">in</span> its_dist){</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>          tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_permutest_&quot;</span>, d)))</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>          tmp_pv <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>tab<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>          tmp_test <span class="ot">&lt;-</span> <span class="cf">if</span>(tmp_pv <span class="sc">&lt;</span> <span class="fl">0.05</span>){</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>            tmp_test <span class="ot">&lt;-</span> <span class="st">&quot;ANOSIM&quot;</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>            tmp_test <span class="ot">&lt;-</span> <span class="st">&quot;ADONIS&quot;</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>          tmp_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(i, d, tmp_pv, tmp_test)</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>          its_perm_pvalues <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(its_perm_pvalues, tmp_df)</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>its_perm_pvalues <span class="ot">&lt;-</span> its_perm_pvalues <span class="sc">%&gt;%</span> </span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="fu">c</span>(<span class="st">&quot;data_set&quot;</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">&quot;dist&quot;</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">&quot;pval&quot;</span> <span class="ot">=</span> <span class="dv">3</span>, <span class="st">&quot;test&quot;</span> <span class="ot">=</span> <span class="dv">4</span>))</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>its_perm_pvalues <span class="ot">&lt;-</span> its_perm_pvalues <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>(ITS) Table 1 | Summary of Beta Dispersion Tests for unfiltered &amp; filtered data sets against various distance metrics. The <code>test</code> column shows the method chosen (based on the test results) to assess differences in beta diversity. If p-values are significant (red, p-value &lt; 0.05) we used ANOSIM, otherwise we used PERMANOVA (blue, p-value &gt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGF0YXNldCx5ZWFyLGRpc3QscHZhbCx0ZXN0CkZVTEwsWWVhciAwLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjYxMjQwMDAwMDAwMDAwMDA2LEFET05JUwpGVUxMLFllYXIgMCxCcmF5LUN1cnRpcyBkaXNzaW1pbGFyaXR5LDAuNTQ3NSxBRE9OSVMKRlVMTCxZZWFyIDAsR293ZXIgZGlzdGFuY2UsMC41NzU0LEFET05JUwpGVUxMLFllYXIgMCxKYWNjYXJkIGluZGV4LDAuNTQ4NSxBRE9OSVMKRlVMTCxZZWFyIDEsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuNzk0MixBRE9OSVMKRlVMTCxZZWFyIDEsQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjcxOTMsQURPTklTCkZVTEwsWWVhciAxLEdvd2VyIGRpc3RhbmNlLDAuMzIwNyxBRE9OSVMKRlVMTCxZZWFyIDEsSmFjY2FyZCBpbmRleCwwLjQ4NzUsQURPTklTCkZVTEwsWWVhciA0LEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjEyMTksQURPTklTCkZVTEwsWWVhciA0LEJyYXktQ3VydGlzIGRpc3NpbWlsYXJpdHksMC4wNjg5LEFET05JUwpGVUxMLFllYXIgNCxHb3dlciBkaXN0YW5jZSwwLjAwOSxBTk9TSU0KRlVMTCxZZWFyIDQsSmFjY2FyZCBpbmRleCwwLjc2NDIsQURPTklTCkZJTFQsWWVhciAwLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjA4MTksQURPTklTCkZJTFQsWWVhciAwLEJyYXktQ3VydGlzIGRpc3NpbWlsYXJpdHksMC4wOTc5LEFET05JUwpGSUxULFllYXIgMCxHb3dlciBkaXN0YW5jZSwwLjA1NTksQURPTklTCkZJTFQsWWVhciAwLEphY2NhcmQgaW5kZXgsMC4zNTk2LEFET05JUwpGSUxULFllYXIgMSxKZW5zZW7igJNTaGFubm9uIGRpdmVyZ2VuY2UsMC41Nzg0LEFET05JUwpGSUxULFllYXIgMSxCcmF5LUN1cnRpcyBkaXNzaW1pbGFyaXR5LDAuNTQzNSxBRE9OSVMKRklMVCxZZWFyIDEsR293ZXIgZGlzdGFuY2UsMC4yOTY3LEFET05JUwpGSUxULFllYXIgMSxKYWNjYXJkIGluZGV4LDAuNjAzNCxBRE9OSVMKRklMVCxZZWFyIDQsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuMDQ3LEFOT1NJTQpGSUxULFllYXIgNCxCcmF5LUN1cnRpcyBkaXNzaW1pbGFyaXR5LDAuMDMyLEFOT1NJTQpGSUxULFllYXIgNCxHb3dlciBkaXN0YW5jZSwwLjAwMSxBTk9TSU0KRklMVCxZZWFyIDQsSmFjY2FyZCBpbmRleCwwLjUwMDUsQURPTklTClBFUkZlY3QsWWVhciAwLEplbnNlbuKAk1NoYW5ub24gZGl2ZXJnZW5jZSwwLjQ0MTYsQURPTklTClBFUkZlY3QsWWVhciAwLEJyYXktQ3VydGlzIGRpc3NpbWlsYXJpdHksMC4zOTg2LEFET05JUwpQRVJGZWN0LFllYXIgMCxHb3dlciBkaXN0YW5jZSwwLjA2NzksQURPTklTClBFUkZlY3QsWWVhciAwLEphY2NhcmQgaW5kZXgsMC4zNTg2LEFET05JUwpQRVJGZWN0LFllYXIgMSxKZW5zZW7igJNTaGFubm9uIGRpdmVyZ2VuY2UsMC44MTIyLEFET05JUwpQRVJGZWN0LFllYXIgMSxCcmF5LUN1cnRpcyBkaXNzaW1pbGFyaXR5LDAuNzc5MixBRE9OSVMKUEVSRmVjdCxZZWFyIDEsR293ZXIgZGlzdGFuY2UsMC41MTM1LEFET05JUwpQRVJGZWN0LFllYXIgMSxKYWNjYXJkIGluZGV4LDAuNzg0MixBRE9OSVMKUEVSRmVjdCxZZWFyIDQsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuMDc2OSxBRE9OSVMKUEVSRmVjdCxZZWFyIDQsQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjA3MTksQURPTklTClBFUkZlY3QsWWVhciA0LEdvd2VyIGRpc3RhbmNlLDAuMDAxLEFOT1NJTQpQRVJGZWN0LFllYXIgNCxKYWNjYXJkIGluZGV4LDAuNzc1MixBRE9OSVMKUElNRSxZZWFyIDAsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuOTYzLEFET05JUwpQSU1FLFllYXIgMCxCcmF5LUN1cnRpcyBkaXNzaW1pbGFyaXR5LDAuOTQ0MSxBRE9OSVMKUElNRSxZZWFyIDAsR293ZXIgZGlzdGFuY2UsMC4wMDEsQU5PU0lNClBJTUUsWWVhciAwLEphY2NhcmQgaW5kZXgsMC44MzQyLEFET05JUwpQSU1FLFllYXIgMSxKZW5zZW7igJNTaGFubm9uIGRpdmVyZ2VuY2UsMC45ODQsQURPTklTClBJTUUsWWVhciAxLEJyYXktQ3VydGlzIGRpc3NpbWlsYXJpdHksMC45OTgsQURPTklTClBJTUUsWWVhciAxLEdvd2VyIGRpc3RhbmNlLDAuMDAxLEFOT1NJTQpQSU1FLFllYXIgMSxKYWNjYXJkIGluZGV4LDAuOTI5MSxBRE9OSVMKUElNRSxZZWFyIDQsSmVuc2Vu4oCTU2hhbm5vbiBkaXZlcmdlbmNlLDAuMzg2NixBRE9OSVMKUElNRSxZZWFyIDQsQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjMxODcsQURPTklTClBJTUUsWWVhciA0LEdvd2VyIGRpc3RhbmNlLDAuMDAxLEFOT1NJTQpQSU1FLFllYXIgNCxKYWNjYXJkIGluZGV4LDAuODkzMSxBRE9OSVMK" download="its_beta_disp_tests.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-60968004bdf2f60b4d5f" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-60968004bdf2f60b4d5f">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"dataset":["FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME"],"year":["Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4"],"dist":["Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index"],"pval":[0.6124,0.5475,0.5754,0.5485,0.7942,0.7193,0.3207,0.4875,0.1219,0.0689,0.009,0.7642,0.0819,0.0979,0.0559,0.3596,0.5784,0.5435,0.2967,0.6034,0.047,0.032,0.001,0.5005,0.4416,0.3986,0.0679,0.3586,0.8122,0.7792,0.5135,0.7842,0.0769,0.0719,0.001,0.7752,0.963,0.9441,0.001,0.8342,0.984,0.998,0.001,0.9291,0.3866,0.3187,0.001,0.8931],"test":["ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ANOSIM","ANOSIM","ANOSIM","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS"]},"columns":[{"accessor":"dataset","name":"Dataset","type":"factor","sortable":true,"resizable":true,"filterable":true,"cell":["FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FULL","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","FILT","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PERFect","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME","PIME"],"header":"Dataset","minWidth":75,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"year","name":"year","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4","Year 0","Year 0","Year 0","Year 0","Year 1","Year 1","Year 1","Year 1","Year 4","Year 4","Year 4","Year 4"],"header":"year","minWidth":75,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"dist","name":"distance","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen–Shannon divergence","Bray-Curtis dissimilarity","Gower distance","Jaccard index"],"header":"distance","minWidth":150,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}},{"accessor":"pval","name":"p-value","type":"numeric","sortable":true,"resizable":false,"filterable":false,"cell":["0.6124","0.5475","0.5754","0.5485","0.7942","0.7193","0.3207","0.4875","0.1219","0.0689","0.0090","0.7642","0.0819","0.0979","0.0559","0.3596","0.5784","0.5435","0.2967","0.6034","0.0470","0.0320","0.0010","0.5005","0.4416","0.3986","0.0679","0.3586","0.8122","0.7792","0.5135","0.7842","0.0769","0.0719","0.0010","0.7752","0.9630","0.9441","0.0010","0.8342","0.9840","0.9980","0.0010","0.9291","0.3866","0.3187","0.0010","0.8931"],"header":"p-value","minWidth":75,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"test","name":"test","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ANOSIM","ANOSIM","ANOSIM","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS","ADONIS","ADONIS","ANOSIM","ADONIS"],"header":"test","minWidth":75,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":12,"showPageSizeOptions":true,"pageSizeOptions":[4,12,24,48],"paginationType":"numbers","showPagination":true,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"f8b1fb7240e32ca12132f703c117f374"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<h2 id="significance-tests-1">Significance Tests</h2>
<p>Depending on the test, we need different data structures. For <code>adonis</code> we need the sample metadata and for <code>anosim</code> we start with a phyloseq object.</p>
<p>We begin by creating separate sample metadata data frames for each phyloseq object (for any <code>adonis</code> analyses).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>     tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>     tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(tmp_get))</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>     tmp_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sampledf&quot;</span>))</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>     <span class="fu">assign</span>(tmp_name, tmp_samp)</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>     <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern=</span><span class="st">&quot;_sampledf&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Next we can use a <code>for</code> loop with an <code>if/else</code> to perform significance tests. This loop does the following for every combination of phyloseq object and distance metric:</p>
<ol type="1">
<li>Get the results of each permutation test.</li>
<li>Get the p-value from the test.</li>
<li>If the p-value is &lt; 0.05, <code>anosim</code> is run, otherwise we use <code>adonis</code>.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (d <span class="cf">in</span> its_dist){</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>          tmp_get <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_permutest_&quot;</span>, d)))</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>          tmp_pv <span class="ot">&lt;-</span> tmp_get<span class="sc">$</span>tab<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(tmp_pv <span class="sc">&lt;</span> <span class="fl">0.05</span>){</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>            tmp_anosim_data <span class="ot">&lt;-</span> <span class="fu">get</span>(i)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>            tmp_groups <span class="ot">&lt;-</span> <span class="fu">get_variable</span>(tmp_anosim_data, choose_group)</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>            tmp_prop <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_prop&quot;</span>)))</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>            tmp_anosim_d <span class="ot">&lt;-</span> <span class="fu">anosim</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(tmp_prop, d), <span class="at">grouping =</span> tmp_groups)</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>            tmp_anosim_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sig_anosim_&quot;</span>, d))</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>            <span class="fu">assign</span>(tmp_anosim_name, tmp_anosim_d)</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>            tmp_adonis_data <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_beta_dist_&quot;</span>, d)))</span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>            tmp_adonis_sampledf <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sampledf&quot;</span>)))</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>            tmp_adonis <span class="ot">&lt;-</span> <span class="fu">adonis</span>(tmp_adonis_data <span class="sc">~</span> YR_TREAT,</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data =</span> tmp_adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>            tmp_adonis2 <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(tmp_adonis_data <span class="sc">~</span> YR_TREAT,</span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> tmp_adonis_sampledf, <span class="at">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>            tmp_adonis_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sig_adonis_&quot;</span>, d))</span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>            tmp_adonis2_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_sig_adonis2_&quot;</span>, d))</span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>            <span class="fu">assign</span>(tmp_adonis_name, tmp_adonis)</span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>            <span class="fu">assign</span>(tmp_adonis2_name, tmp_adonis2)</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<details>
<summary>
Results of Significance tests <strong>(FULL data set)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-12" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.0598 0.04374 0.8385  0.996
Residual 55  23.1721 0.95626              
Total    58  24.2319 1.00000              


----- PERMANOVA for Gower distance ( gower ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3 0.003129 0.04973 0.9594 0.8262
Residual 55 0.059787 0.95027              
Total    58 0.062916 1.00000              


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.1832 0.04712 0.9065      1
Residual 55  23.9283 0.95288              
Total    58  25.1115 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.4221 0.04016 0.7672  0.998
Residual 55  10.0868 0.95984              
Total    58  10.5088 1.00000              </code></pre>
</div>
</section>
<section id="year-1-12" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F Pr(&gt;F)
YR_TREAT  3   1.2654 0.04784 0.938 0.8501
Residual 56  25.1840 0.95216             
Total    59  26.4494 1.00000             


----- PERMANOVA for Gower distance ( gower ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3 0.002761 0.04911 0.9641 0.7852
Residual 56 0.053467 0.95089              
Total    59 0.056228 1.00000              


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.3022 0.04968 0.9757 0.7423
Residual 56  24.9128 0.95032              
Total    59  26.2150 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.5462 0.04602 0.9005 0.8482
Residual 56  11.3214 0.95398              
Total    59  11.8675 1.00000              </code></pre>
</div>
</section>
<section id="year-4-12" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   1.6826 0.06733 1.3476 0.000999 ***
Residual 56  23.3076 0.93267                    
Total    59  24.9902 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F   Pr(&gt;F)   
YR_TREAT  3    1.492 0.0588 1.1661 0.001998 **
Residual 56   23.884 0.9412                   
Total    59   25.375 1.0000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3    0.845 0.07747 1.5675 0.000999 ***
Residual 56   10.063 0.92253                    
Total    59   10.908 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- ANOSIM for Gower distance ( gower ) -----


Call:
anosim(x = phyloseq::distance(tmp_prop, d), grouping = tmp_groups) 
Dissimilarity: gower 

ANOSIM statistic R: -0.007704 
      Significance: 0.893 

Permutation: free
Number of permutations: 999</code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Significance tests <strong>(Arbitrary filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-13" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3    0.842 0.04172 0.7983  0.958
Residual 55   19.338 0.95828              
Total    58   20.180 1.00000              


----- PERMANOVA for Gower distance ( gower ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.0218 0.04319 0.8276  0.952
Residual 55   0.4830 0.95681              
Total    58   0.5048 1.00000              


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.6529 0.03853 0.7346  0.999
Residual 55  16.2938 0.96147              
Total    58  16.9467 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F Pr(&gt;F)
YR_TREAT  3   0.2784 0.0357 0.6787  0.971
Residual 55   7.5214 0.9643              
Total    58   7.7999 1.0000              </code></pre>
</div>
</section>
<section id="year-1-13" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.0938 0.04888 0.9593 0.5564
Residual 56  21.2846 0.95112              
Total    59  22.3784 1.00000              


----- PERMANOVA for Gower distance ( gower ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3  0.02040 0.04986 0.9795 0.5045
Residual 56  0.38888 0.95014              
Total    59  0.40928 1.00000              


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.8436 0.04844 0.9502 0.5964
Residual 56  16.5726 0.95156              
Total    59  17.4163 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.4111 0.04561 0.8922 0.6813
Residual 56   8.6019 0.95439              
Total    59   9.0131 1.00000              </code></pre>
</div>
</section>
<section id="year-4-13" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3    1.377 0.08209 1.6693 0.000999 ***
Residual 56   15.399 0.91791                    
Total    59   16.776 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- ANOSIM for Bray-Curtis dissimilarity ( bray ) -----


Call:
anosim(x = phyloseq::distance(tmp_prop, d), grouping = tmp_groups) 
Dissimilarity: bray 

ANOSIM statistic R: 0.1578 
      Significance: 0.001 

Permutation: free
Number of permutations: 999



----- ANOSIM for Gower distance ( gower ) -----


Call:
anosim(x = phyloseq::distance(tmp_prop, d), grouping = tmp_groups) 
Dissimilarity: gower 

ANOSIM statistic R: 0.06232 
      Significance: 0.001 

Permutation: free
Number of permutations: 999



----- ANOSIM for Jensen-Shannon Divergence ( jsd ) -----


Call:
anosim(x = phyloseq::distance(tmp_prop, d), grouping = tmp_groups) 
Dissimilarity: 

ANOSIM statistic R: 0.1686 
      Significance: 0.001 

Permutation: free
Number of permutations: 999</code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Significance tests <strong>(PERFect filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-14" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.0598 0.04374 0.8385  0.996
Residual 55  23.1721 0.95626              
Total    58  24.2319 1.00000              


----- PERMANOVA for Gower distance ( gower ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3 0.003129 0.04973 0.9594 0.8262
Residual 55 0.059787 0.95027              
Total    58 0.062916 1.00000              


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.1832 0.04712 0.9065      1
Residual 55  23.9283 0.95288              
Total    58  25.1115 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.4221 0.04016 0.7672  0.998
Residual 55  10.0868 0.95984              
Total    58  10.5088 1.00000              </code></pre>
</div>
</section>
<section id="year-1-14" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F Pr(&gt;F)
YR_TREAT  3   1.2654 0.04784 0.938 0.8501
Residual 56  25.1840 0.95216             
Total    59  26.4494 1.00000             


----- PERMANOVA for Gower distance ( gower ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3 0.002761 0.04911 0.9641 0.7852
Residual 56 0.053467 0.95089              
Total    59 0.056228 1.00000              


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   1.3022 0.04968 0.9757 0.7423
Residual 56  24.9128 0.95032              
Total    59  26.2150 1.00000              


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F Pr(&gt;F)
YR_TREAT  3   0.5462 0.04602 0.9005 0.8482
Residual 56  11.3214 0.95398              
Total    59  11.8675 1.00000              </code></pre>
</div>
</section>
<section id="year-4-14" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   1.6826 0.06733 1.3476 0.000999 ***
Residual 56  23.3076 0.93267                    
Total    59  24.9902 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F   Pr(&gt;F)   
YR_TREAT  3    1.492 0.0588 1.1661 0.001998 **
Residual 56   23.884 0.9412                   
Total    59   25.375 1.0000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3    0.845 0.07747 1.5675 0.000999 ***
Residual 56   10.063 0.92253                    
Total    59   10.908 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- ANOSIM for Gower distance ( gower ) -----


Call:
anosim(x = phyloseq::distance(tmp_prop, d), grouping = tmp_groups) 
Dissimilarity: gower 

ANOSIM statistic R: -0.007704 
      Significance: 0.893 

Permutation: free
Number of permutations: 999</code></pre>
</div>
</section>
</div>
</div>
</details>
<details>
<summary>
Results of Significance tests <strong>(PIME filtered)</strong>
</summary>
<div class="l-body">
<div class="panelset">
<section id="year-0-15" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   2.1091 0.12603 2.6439 0.000999 ***
Residual 55  14.6248 0.87397                    
Total    58  16.7339 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   4.8075 0.32904 8.9907 0.000999 ***
Residual 55   9.8032 0.67096                    
Total    58  14.6107 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   1.0571 0.18387 4.1304 0.000999 ***
Residual 55   4.6922 0.81613                    
Total    58   5.7494 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- ANOSIM for Gower distance ( gower ) -----


Call:
anosim(x = phyloseq::distance(tmp_prop, d), grouping = tmp_groups) 
Dissimilarity: gower 

ANOSIM statistic R: 0.4004 
      Significance: 0.001 

Permutation: free
Number of permutations: 999</code></pre>
</div>
</section>
<section id="year-1-15" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3    2.691 0.13947 3.0253 0.000999 ***
Residual 56   16.604 0.86053                    
Total    59   19.295 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   5.0778 0.35012 10.057 0.000999 ***
Residual 56   9.4251 0.64988                    
Total    59  14.5029 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   1.3362 0.18777 4.3153 0.000999 ***
Residual 56   5.7799 0.81223                    
Total    59   7.1161 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- ANOSIM for Gower distance ( gower ) -----


Call:
anosim(x = phyloseq::distance(tmp_prop, d), grouping = tmp_groups) 
Dissimilarity: gower 

ANOSIM statistic R: 0.3107 
      Significance: 0.001 

Permutation: free
Number of permutations: 999</code></pre>
</div>
</section>
<section id="year-4-15" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-body">
<pre><code>----- PERMANOVA for Bray-Curtis dissimilarity ( bray ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2     F   Pr(&gt;F)    
YR_TREAT  3   3.8036 0.19177 4.429 0.000999 ***
Residual 56  16.0308 0.80823                   
Total    59  19.8343 1.00000                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jaccard index ( jaccard ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs      R2      F   Pr(&gt;F)    
YR_TREAT  3   6.3251 0.39553 12.215 0.000999 ***
Residual 56   9.6661 0.60447                    
Total    59  15.9912 1.00000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- PERMANOVA for Jensen-Shannon Divergence ( jsd ) -----

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 1000

adonis2(formula = tmp_adonis_data ~ YR_TREAT, data = tmp_adonis_sampledf, permutations = 1000)
         Df SumOfSqs     R2      F   Pr(&gt;F)    
YR_TREAT  3   2.1589 0.2854 7.4553 0.000999 ***
Residual 56   5.4054 0.7146                    
Total    59   7.5643 1.0000                    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


----- ANOSIM for Gower distance ( gower ) -----


Call:
anosim(x = phyloseq::distance(tmp_prop, d), grouping = tmp_groups) 
Dissimilarity: gower 

ANOSIM statistic R: 0.4025 
      Significance: 0.001 

Permutation: free
Number of permutations: 999</code></pre>
</div>
</section>
</div>
</div>
</details>
<h2 id="summaries-1">Summaries</h2>
<p>Here is a quick summary of significance tests for the data sets against three distance matrices.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="l-body">
<div class="panelset">
<section id="full-data-sets-1" class="panel">
<h4>FULL data sets</h4>
<p><small><strong>(ITS) Table 2 | Summary of significant tests by Year for FULL, unfiltered data set. P-values in red indicate significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGlzdGFuY2UsWWVhciAwLFllYXIgMSxZZWFyIDQKQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjk5NiwwLjg1MDEsMC4wMDEKR293ZXIgZGlzdGFuY2UsMC44MjYyLDAuNzg1MiwwLjg5MwpKYWNjYXJkIGluZGV4LDEsMC43NDIzLDAuMDAyCkplbnNlbi1TaGFubm9uIERpdmVyZ2VuY2UsMC45OTgsMC44NDgyLDAuMDAxCg==" download="its_tab_sig_test_full.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-f210519e11807785dec1" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f210519e11807785dec1">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance":["Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen-Shannon Divergence"],"Year 0":[0.996,0.8262,1,0.998],"Year 1":[0.8501,0.7852,0.7423,0.8482],"Year 4":[0.001,0.893,0.002,0.001]},"columns":[{"accessor":"distance","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen-Shannon Divergence"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Year 0","name":"Year 0","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.996","0.8262","1.000","0.998"],"header":"Year 0","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 1","name":"Year 1","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.8501","0.7852","0.7423","0.8482"],"header":"Year 1","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 4","name":"Year 4","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.893","0.002","0.001"],"header":"Year 4","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["Year 0","Year 1","Year 4"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"1aba0cbd2c00acd6acbdc530596bc718"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
<section id="arbitrary-filtered-1" class="panel">
<h4>Arbitrary filtered</h4>
<p><small><strong>(ITS) Table 3 | Summary of significant tests by Year for Arbitrary filtered data set. P-values in red indicate significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGlzdGFuY2UsWWVhciAwLFllYXIgMSxZZWFyIDQKQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjk1OCwwLjU1NjQsMC4wMDEKR293ZXIgZGlzdGFuY2UsMC45NTIsMC41MDQ1LDAuMDAxCkphY2NhcmQgaW5kZXgsMC45OTksMC41OTY0LDAuMDAxCkplbnNlbi1TaGFubm9uIERpdmVyZ2VuY2UsMC45NzEsMC42ODEzLDAuMDAxCg==" download="its_tab_sig_test_filt.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-9e81a8e2313628607f53" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9e81a8e2313628607f53">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance":["Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen-Shannon Divergence"],"Year 0":[0.958,0.952,0.999,0.971],"Year 1":[0.5564,0.5045,0.5964,0.6813],"Year 4":[0.001,0.001,0.001,0.001]},"columns":[{"accessor":"distance","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen-Shannon Divergence"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Year 0","name":"Year 0","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.958","0.952","0.999","0.971"],"header":"Year 0","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 1","name":"Year 1","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.5564","0.5045","0.5964","0.6813"],"header":"Year 1","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 4","name":"Year 4","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"Year 4","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["Year 0","Year 1","Year 4"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"864b491095ff24c7616af01b116fa265"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
<section id="perfect-filtered-1" class="panel">
<h4>PERFect filtered</h4>
<p><small><strong>(ITS) Table 4 | Summary of significant tests by Year for PERFect filtered data set. P-values in red indicate significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGlzdGFuY2UsWWVhciAwLFllYXIgMSxZZWFyIDQKQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjk4OSwwLjg4MzEsMC4wMDIKR293ZXIgZGlzdGFuY2UsMC45NjIsMC40NzE1LDAuMDAxCkphY2NhcmQgaW5kZXgsMC45OTMsMC42OTIzLDAuMDAzCkplbnNlbi1TaGFubm9uIERpdmVyZ2VuY2UsMC45ODksMC45MTcxLDAuMDAxCg==" download="its_tab_sig_test_perfect.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-72fb4cfdc117e69e1888" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-72fb4cfdc117e69e1888">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance":["Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen-Shannon Divergence"],"Year 0":[0.989,0.962,0.993,0.989],"Year 1":[0.8831,0.4715,0.6923,0.9171],"Year 4":[0.002,0.001,0.003,0.001]},"columns":[{"accessor":"distance","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen-Shannon Divergence"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Year 0","name":"Year 0","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.989","0.962","0.993","0.989"],"header":"Year 0","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 1","name":"Year 1","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.8831","0.4715","0.6923","0.9171"],"header":"Year 1","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"},{"color":"#2271B2","fontWeight":"bold"}]},{"accessor":"Year 4","name":"Year 4","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.002","0.001","0.003","0.001"],"header":"Year 4","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["Year 0","Year 1","Year 4"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"ff1c1db7da6486737d1937f710f96e84"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
<section id="pime-filtered-1" class="panel">
<h4>PIME filtered</h4>
<p><small><strong>(ITS) Table 5 | Summary of significant tests by Year for PIME filtered data set. P-values in red indicate significant differences (p-value &lt; 0.05).</strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<a href="data:text/csv;base64,ZGlzdGFuY2UsWWVhciAwLFllYXIgMSxZZWFyIDQKQnJheS1DdXJ0aXMgZGlzc2ltaWxhcml0eSwwLjAwMSwwLjAwMSwwLjAwMQpHb3dlciBkaXN0YW5jZSwwLjAwMSwwLjAwMSwwLjAwMQpKYWNjYXJkIGluZGV4LDAuMDAxLDAuMDAxLDAuMDAxCkplbnNlbi1TaGFubm9uIERpdmVyZ2VuY2UsMC4wMDEsMC4wMDEsMC4wMDEK" download="its_tab_sig_test_pime.csv">
<button class="btn btn-default" has_icon="TRUE"><i class="fa fa-save"></i> Download data as csv file</button>
</a>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-d411b5e91e65001f3e0f" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d411b5e91e65001f3e0f">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"distance":["Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen-Shannon Divergence"],"Year 0":[0.001,0.001,0.001,0.001],"Year 1":[0.001,0.001,0.001,0.001],"Year 4":[0.001,0.001,0.001,0.001]},"columns":[{"accessor":"distance","name":"Distance metric","type":"character","sortable":false,"resizable":true,"filterable":false,"cell":["Bray-Curtis dissimilarity","Gower distance","Jaccard index","Jensen-Shannon Divergence"],"header":"Distance metric","minWidth":200,"align":"left","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"Year 0","name":"Year 0","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"Year 0","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]},{"accessor":"Year 1","name":"Year 1","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"Year 1","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]},{"accessor":"Year 4","name":"Year 4","type":"numeric","sortable":false,"resizable":true,"filterable":false,"cell":["0.001","0.001","0.001","0.001"],"header":"Year 4","minWidth":100,"align":"center","headerVAlign":"center","footerStyle":{"fontWeight":"bold"},"style":[{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"},{"color":"#B22271","fontWeight":"bold"}]}],"columnGroups":[{"name":"p-values","headerStyle":{"fontSize":"1.3em"},"columns":["Year 0","Year 1","Year 4"]}],"defaultPageSize":12,"paginationType":"numbers","showPagination":false,"showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"compact":true,"theme":{"style":{"fontSize":"0.9em"}},"dataKey":"d1d327de2731feb95a512bdbf01a6426"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="ordinations">Ordinations</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Now we can visualize the results of several distance matrices to access dissimilarity among sample. Here we ordinate using Principal Coordinate Analysis (PCoA) but we could also test other methods like NMDS, CCA, etc.</p>
<h2 id="plot-code">Plot Code</h2>
<p>Here I made a custom “function” to run the analysis, plot the graphs, save graph objects, and save plots (as <code>.png</code> and <code>.pdf</code> files). I am sure an actual programmer would be shocked, but it works. This is used for both the 16S rRNA and ITS data sets.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>agua_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#B26322&quot;</span>, <span class="st">&quot;#29B222&quot;</span>, <span class="st">&quot;#2271B2&quot;</span>, <span class="st">&quot;#AB22B2&quot;</span>)</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="co">#agua_col2 &lt;- c(&quot;#2271B2&quot;, &quot;#3DB7E9&quot;, &quot;#F748A5&quot;, &quot;#359B73&quot;, &quot;#D55E00&quot;, &quot;#E69F00&quot;)</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>agua_col2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#191919&quot;</span>, <span class="st">&quot;#191919&quot;</span>, <span class="st">&quot;#191919&quot;</span>, <span class="st">&quot;#191919&quot;</span>, <span class="st">&quot;#191919&quot;</span>, <span class="st">&quot;#191919&quot;</span>)</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co">#source(&quot;hack_code/trans_beta_jjs.R&quot;)</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>treat_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;CTL&quot;</span>, <span class="st">&quot;N&quot;</span>, <span class="st">&quot;P&quot;</span>, <span class="st">&quot;NP&quot;</span>) </span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>age_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NEW&quot;</span>, <span class="st">&quot;YNG&quot;</span>, <span class="st">&quot;OLD&quot;</span>)</span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>microeco_beta_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(choose_input, choose_metric, choose_ord) {  </span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>  tmp_dataset <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(choose_input, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)))</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a><span class="do">###### ORDER factors</span></span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>sample_table<span class="sc">$</span>TREAT <span class="sc">%&lt;&gt;%</span> <span class="fu">factor</span>(., <span class="at">levels =</span>  treat_order)</span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>sample_table<span class="sc">$</span>AGE <span class="sc">%&lt;&gt;%</span> <span class="fu">factor</span>(., <span class="at">levels =</span> age_order)</span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a><span class="do">###### GENERATE plots</span></span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a>  tmp_t1 <span class="ot">&lt;-</span> trans_beta<span class="sc">$</span><span class="fu">new</span>(<span class="at">dataset =</span> tmp_dataset, <span class="at">group =</span> <span class="st">&quot;TREAT&quot;</span>, <span class="at">measure =</span> choose_metric)</span>
<span id="cb62-17"><a href="#cb62-17" aria-hidden="true" tabindex="-1"></a>  tmp_t1<span class="sc">$</span><span class="fu">cal_ordination</span>(<span class="at">ordination =</span> choose_ord)</span>
<span id="cb62-18"><a href="#cb62-18" aria-hidden="true" tabindex="-1"></a>  tmp_t1_ord_plot <span class="ot">&lt;-</span> tmp_t1<span class="sc">$</span><span class="fu">plot_ordination</span>(<span class="at">plot_color =</span> <span class="st">&quot;TREAT&quot;</span>, </span>
<span id="cb62-19"><a href="#cb62-19" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">plot_shape =</span> <span class="st">&quot;AGE&quot;</span>,</span>
<span id="cb62-20"><a href="#cb62-20" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">plot_group_ellipse =</span> <span class="cn">FALSE</span>,</span>
<span id="cb62-21"><a href="#cb62-21" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">color_values =</span> agua_col, </span>
<span id="cb62-22"><a href="#cb62-22" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">shape_values =</span> <span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">17</span>, <span class="dv">15</span>)</span>
<span id="cb62-23"><a href="#cb62-23" aria-hidden="true" tabindex="-1"></a>                                            ) <span class="sc">+</span> </span>
<span id="cb62-24"><a href="#cb62-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span>  </span>
<span id="cb62-25"><a href="#cb62-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb62-26"><a href="#cb62-26" aria-hidden="true" tabindex="-1"></a>  tmp_t1<span class="sc">$</span><span class="fu">cal_group_distance</span>()</span>
<span id="cb62-27"><a href="#cb62-27" aria-hidden="true" tabindex="-1"></a>  tmp_t1_within_group_plot <span class="ot">&lt;-</span> tmp_t1<span class="sc">$</span><span class="fu">plot_group_distance</span>(<span class="at">plot_group_order =</span> treat_order, </span>
<span id="cb62-28"><a href="#cb62-28" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">distance_pair_stat =</span> <span class="cn">TRUE</span>, </span>
<span id="cb62-29"><a href="#cb62-29" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">hide_ns =</span> <span class="cn">TRUE</span>, <span class="at">hide_ns_more =</span> <span class="cn">NULL</span>, </span>
<span id="cb62-30"><a href="#cb62-30" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">color_values =</span> agua_col)</span>
<span id="cb62-31"><a href="#cb62-31" aria-hidden="true" tabindex="-1"></a>  tmp_t1<span class="sc">$</span>res_group_distance</span>
<span id="cb62-32"><a href="#cb62-32" aria-hidden="true" tabindex="-1"></a>  tmp_t1<span class="sc">$</span><span class="fu">cal_group_distance</span>(<span class="at">within_group =</span> <span class="cn">FALSE</span>)</span>
<span id="cb62-33"><a href="#cb62-33" aria-hidden="true" tabindex="-1"></a>  tmp_t1_btwn_group_plot <span class="ot">&lt;-</span> tmp_t1<span class="sc">$</span><span class="fu">plot_group_distance</span>(<span class="at">distance_pair_stat =</span> <span class="cn">TRUE</span>, </span>
<span id="cb62-34"><a href="#cb62-34" aria-hidden="true" tabindex="-1"></a>                                                       <span class="at">hide_ns =</span> <span class="cn">TRUE</span>, <span class="at">hide_ns_more =</span> <span class="cn">NULL</span>,</span>
<span id="cb62-35"><a href="#cb62-35" aria-hidden="true" tabindex="-1"></a>                                                       <span class="at">color_values =</span> agua_col2)</span>
<span id="cb62-36"><a href="#cb62-36" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (choose_metric <span class="sc">==</span> <span class="st">&quot;jsd&quot;</span>) {</span>
<span id="cb62-37"><a href="#cb62-37" aria-hidden="true" tabindex="-1"></a>    tmp_t1_within_group_plot <span class="ot">&lt;-</span> tmp_t1_within_group_plot <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Jensen-Shannon Index&quot;</span>)</span>
<span id="cb62-38"><a href="#cb62-38" aria-hidden="true" tabindex="-1"></a>    tmp_t1_btwn_group_plot <span class="ot">&lt;-</span> tmp_t1_btwn_group_plot <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Jensen-Shannon Index&quot;</span>)</span>
<span id="cb62-39"><a href="#cb62-39" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (choose_metric <span class="sc">==</span> <span class="st">&quot;manhattan&quot;</span>) {</span>
<span id="cb62-40"><a href="#cb62-40" aria-hidden="true" tabindex="-1"></a>    tmp_t1_within_group_plot <span class="ot">&lt;-</span> tmp_t1_within_group_plot <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Manhattan distance&quot;</span>)</span>
<span id="cb62-41"><a href="#cb62-41" aria-hidden="true" tabindex="-1"></a>    tmp_t1_btwn_group_plot <span class="ot">&lt;-</span> tmp_t1_btwn_group_plot <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Manhattan distance&quot;</span>)</span>
<span id="cb62-42"><a href="#cb62-42" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (choose_metric <span class="sc">==</span> <span class="st">&quot;gower&quot;</span>) {</span>
<span id="cb62-43"><a href="#cb62-43" aria-hidden="true" tabindex="-1"></a>    tmp_t1_within_group_plot <span class="ot">&lt;-</span> tmp_t1_within_group_plot <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Gower distance&quot;</span>)</span>
<span id="cb62-44"><a href="#cb62-44" aria-hidden="true" tabindex="-1"></a>    tmp_t1_btwn_group_plot <span class="ot">&lt;-</span> tmp_t1_btwn_group_plot <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Gower distance&quot;</span>)</span>
<span id="cb62-45"><a href="#cb62-45" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (choose_metric <span class="sc">==</span> <span class="st">&quot;canberra&quot;</span>) {</span>
<span id="cb62-46"><a href="#cb62-46" aria-hidden="true" tabindex="-1"></a>    tmp_t1_within_group_plot <span class="ot">&lt;-</span> tmp_t1_within_group_plot <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Canberra distance&quot;</span>)</span>
<span id="cb62-47"><a href="#cb62-47" aria-hidden="true" tabindex="-1"></a>    tmp_t1_btwn_group_plot <span class="ot">&lt;-</span> tmp_t1_btwn_group_plot <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Canberra distance&quot;</span>)</span>
<span id="cb62-48"><a href="#cb62-48" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span>  {</span>
<span id="cb62-49"><a href="#cb62-49" aria-hidden="true" tabindex="-1"></a>    tmp_t1_within_group_plot <span class="ot">&lt;-</span> tmp_t1_within_group_plot</span>
<span id="cb62-50"><a href="#cb62-50" aria-hidden="true" tabindex="-1"></a>    tmp_t1_btwn_group_plot <span class="ot">&lt;-</span> tmp_t1_btwn_group_plot</span>
<span id="cb62-51"><a href="#cb62-51" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb62-52"><a href="#cb62-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-53"><a href="#cb62-53" aria-hidden="true" tabindex="-1"></a><span class="do">###### SET names</span></span>
<span id="cb62-54"><a href="#cb62-54" aria-hidden="true" tabindex="-1"></a>  tmp_name_ord <span class="ot">&lt;-</span> <span class="fu">paste</span>(choose_input, <span class="st">&quot;_me_&quot;</span>, choose_metric, <span class="st">&quot;_&quot;</span>, choose_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb62-55"><a href="#cb62-55" aria-hidden="true" tabindex="-1"></a>  tmp_name_wg <span class="ot">&lt;-</span> <span class="fu">paste</span>(choose_input, <span class="st">&quot;_me_wg_&quot;</span>, choose_metric, <span class="st">&quot;_&quot;</span>, choose_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb62-56"><a href="#cb62-56" aria-hidden="true" tabindex="-1"></a>  tmp_name_bg <span class="ot">&lt;-</span> <span class="fu">paste</span>(choose_input, <span class="st">&quot;_me_bg_&quot;</span>, choose_metric, <span class="st">&quot;_&quot;</span>, choose_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb62-57"><a href="#cb62-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-58"><a href="#cb62-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_ord, tmp_t1_ord_plot, <span class="at">envir =</span> <span class="fu">parent.frame</span>(<span class="at">n =</span> <span class="dv">2</span>))</span>
<span id="cb62-59"><a href="#cb62-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_wg, tmp_t1_within_group_plot, <span class="at">envir =</span> <span class="fu">parent.frame</span>(<span class="at">n =</span> <span class="dv">2</span>))</span>
<span id="cb62-60"><a href="#cb62-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(tmp_name_bg, tmp_t1_btwn_group_plot, <span class="at">envir =</span> <span class="fu">parent.frame</span>(<span class="at">n =</span> <span class="dv">2</span>))</span>
<span id="cb62-61"><a href="#cb62-61" aria-hidden="true" tabindex="-1"></a><span class="do">###### SET paths</span></span>
<span id="cb62-62"><a href="#cb62-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste</span>(microeco_path, choose_input, <span class="st">&quot;/ordination/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb62-63"><a href="#cb62-63" aria-hidden="true" tabindex="-1"></a>  tmp_path <span class="ot">&lt;-</span> <span class="fu">paste</span>(microeco_path, choose_input, <span class="st">&quot;/ordination/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb62-64"><a href="#cb62-64" aria-hidden="true" tabindex="-1"></a>  tmp_path_ord <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, <span class="st">&quot;ord_plots/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb62-65"><a href="#cb62-65" aria-hidden="true" tabindex="-1"></a>  tmp_path_wg <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, <span class="st">&quot;within_group/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb62-66"><a href="#cb62-66" aria-hidden="true" tabindex="-1"></a>  tmp_path_bg <span class="ot">&lt;-</span> <span class="fu">paste</span>(tmp_path, <span class="st">&quot;between_group/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb62-67"><a href="#cb62-67" aria-hidden="true" tabindex="-1"></a><span class="do">###### SAVE plots  </span></span>
<span id="cb62-68"><a href="#cb62-68" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_ord_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_ord, <span class="st">&quot;.png&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb62-69"><a href="#cb62-69" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_ord_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_ord, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_ord, <span class="st">&quot;.pdf&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))                    </span>
<span id="cb62-70"><a href="#cb62-70" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_within_group_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_wg, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_wg, <span class="st">&quot;.png&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb62-71"><a href="#cb62-71" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_within_group_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_wg, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_wg, <span class="st">&quot;.pdf&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))                  </span>
<span id="cb62-72"><a href="#cb62-72" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_btwn_group_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_bg, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_bg, <span class="st">&quot;.png&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb62-73"><a href="#cb62-73" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(tmp_t1_btwn_group_plot, <span class="at">path =</span> <span class="fu">paste</span>(tmp_path_bg, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">filename =</span> <span class="fu">paste0</span>(tmp_name_bg, <span class="st">&quot;.pdf&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))                  </span>
<span id="cb62-74"><a href="#cb62-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-75"><a href="#cb62-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb62-76"><a href="#cb62-76" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<p>And then run it like so:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> <span class="st">&quot;ssu_ps_pime_Y0&quot;</span>, </span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">choose_metric =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">choose_ord =</span> <span class="st">&quot;PCoA&quot;</span>)</span></code></pre></div>
</div>
<p>However, since we have a lot of data sets and distance metrics to test, I wrote a little function that runs the function described above on a bunch of data sets, distance metrics, and ordination methods.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>run_microeco_beta_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(dataset, year, ord) {</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">get</span>(<span class="fu">paste</span>(dataset, <span class="st">&quot;_&quot;</span>,  year, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity)) {</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>  tmp_metric <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">get</span>(<span class="fu">paste</span>(dataset, <span class="st">&quot;_&quot;</span>,  year, <span class="st">&quot;_me&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))<span class="sc">$</span>beta_diversity[j])</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>  tmp_choose_input <span class="ot">&lt;-</span> <span class="fu">paste</span>(dataset, <span class="st">&quot;_&quot;</span>,  year, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">microeco_beta_plot</span>(<span class="at">choose_input =</span> tmp_choose_input, <span class="at">choose_metric =</span> tmp_metric, <span class="at">choose_ord =</span> ord)</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>} </span></code></pre></div>
</details>
</div>
<p>So if we wanted to plot the 16S rRNA, PERFect filtered data set, from Year 4 using PCoA, we would run the command like so.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_perfect&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span></code></pre></div>
</div>
<p>The two functions could likely be combined but I could not figure out how to make that work :/</p>
<h2 id="s-rrna-plots">16S rRNA plots</h2>
<p>To begin, we calculate distance matrices using several methods. By default, the <code>microeco</code> package will calculate Bray-Curtis, Unweighted UniFrac, and Weighted UniFrac. If we want to include others (which we do) we simply add code to calculate distance using the desire metric. The code below illustrates how this is accomplished.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>ssu_ps_pime_Y4</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>microeco_path <span class="ot">&lt;-</span> <span class="st">&quot;files/beta/microeco/&quot;</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> ssu_samp_ps) {</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>  tmp_dataset <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)))</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span><span class="fu">cal_betadiv</span>(<span class="at">unifrac =</span> <span class="cn">TRUE</span>)</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD JSD DISTANCE ####  </span></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;jsd&quot;</span>) </span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_jsd), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_jsd)</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>jsd <span class="ot">&lt;-</span> tmp_jsd</span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD jaccard DISTANCE ####  </span></span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a>  tmp_jac <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;jaccard&quot;</span>, <span class="at">binary =</span> <span class="cn">TRUE</span>) </span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>  tmp_jac <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_jac), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>  tmp_jac <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_jac)</span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>jaccard <span class="ot">&lt;-</span> tmp_jac</span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a><span class="do">#### #### #### #### #### #### #### ####  </span></span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste</span>(microeco_path, i, <span class="st">&quot;/metrics/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span><span class="fu">save_betadiv</span>(<span class="at">dirpath =</span> <span class="fu">paste</span>(microeco_path, i, <span class="st">&quot;/metrics/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(ssu_ps_perfect_Y0_me<span class="sc">$</span>beta_diversity)</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(ssu_ps_perfect_Y0_me<span class="sc">$</span>beta_diversity)</span></code></pre></div>
<pre><code>[1] &quot;list&quot;
              Length Class  Mode   
bray          3481   -none- numeric
jaccard       3481   -none- numeric
wei_unifrac   3481   -none- numeric
unwei_unifrac 3481   -none- numeric
jsd           3481   -none- numeric</code></pre>
</div>
<p>Now we run the custom function <code>run_microeco_beta_plot</code>, which incorporates the other custom function <code>microeco_beta_plot</code>, for each of the filtered and unfiltered data sets, split by Year. Like so…</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_work&quot;</span>, <span class="st">&quot;Y0&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_work&quot;</span>, <span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_work&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_filt&quot;</span>, <span class="st">&quot;Y0&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_filt&quot;</span>, <span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_filt&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_perfect&quot;</span>, <span class="st">&quot;Y0&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_perfect&quot;</span>, <span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_perfect&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_pime&quot;</span>, <span class="st">&quot;Y0&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_pime&quot;</span>, <span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;ssu_ps_pime&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>The remaining code blocks in this section are used to generate combined figures. These code blocks are hidden by default but can be accessed using the repo link at the bottom of the page.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="l-page">
<div class="panelset">
<section id="full-data-set" class="panel">
<h4>FULL data set</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_work_combo_plot.png" width="4724" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 1 | PCoA ordination plots of the FULL data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right = UniFrac (wieghted).</strong></small></p>
</section>
<section id="arbitrary-filtered-2" class="panel">
<h4>Arbitrary filtered</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_filt_combo_plot.png" width="4724" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 2 | PCoA ordination plots of the Arbitrary filtered data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right = UniFrac (wieghted).</strong></small></p>
</section>
<section id="perfect-filtered-2" class="panel">
<h4>PERFect filtered</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_perfect_combo_plot.png" width="4724" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 3 | PCoA ordination plots of the PERfect filtered data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right = UniFrac (wieghted).</strong></small></p>
</section>
<section id="pime-filtered-2" class="panel">
<h4>PIME filtered</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_pime_combo_plot.png" width="4724" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 4 | PCoA ordination plots of the PIME filtered data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon Divergence, middle = UniFrac (unwieghted), right = UniFrac (wieghted).</strong></small></p>
</section>
</div>
</div>
<h2 id="its-plots">ITS Plots</h2>
<p>The ITS code is nearly identical. The only differences are related to selected distance metrics.</p>
<details>
<summary>
Code to reproduce <strong>ITS</strong> plots
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>microeco_path <span class="ot">&lt;-</span> <span class="st">&quot;files/beta/microeco/&quot;</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> its_samp_ps) {</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>  tmp_dataset <span class="ot">&lt;-</span> <span class="fu">get</span>(purrr<span class="sc">::</span><span class="fu">map_chr</span>(i, <span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_me&quot;</span>)))</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span><span class="fu">cal_betadiv</span>(<span class="at">unifrac =</span> <span class="cn">FALSE</span>)</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD JSD DISTANCE ####  </span></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;jsd&quot;</span>) </span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_jsd), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>  tmp_jsd <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_jsd)</span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>jsd <span class="ot">&lt;-</span> tmp_jsd</span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD jaccard DISTANCE ####  </span></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>  tmp_jac <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;jaccard&quot;</span>, <span class="at">binary =</span> <span class="cn">TRUE</span>) </span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>  tmp_jac <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_jac), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a>  tmp_jac <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_jac)</span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>jaccard <span class="ot">&lt;-</span> tmp_jac</span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD GOWER DISTANCE ####  </span></span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a>  tmp_gower <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;gower&quot;</span>) </span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a>  tmp_gower <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_gower), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>  tmp_gower <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_gower)</span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>gower <span class="ot">&lt;-</span> tmp_gower</span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD canberra DISTANCE ####  </span></span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a>  tmp_canberra <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;canberra&quot;</span>) </span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a>  tmp_canberra <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_canberra), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb70-23"><a href="#cb70-23" aria-hidden="true" tabindex="-1"></a>  tmp_canberra <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_canberra)</span>
<span id="cb70-24"><a href="#cb70-24" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>canberra <span class="ot">&lt;-</span> tmp_canberra</span>
<span id="cb70-25"><a href="#cb70-25" aria-hidden="true" tabindex="-1"></a><span class="do">#### CODE TO ADD manhattan DISTANCE ####  </span></span>
<span id="cb70-26"><a href="#cb70-26" aria-hidden="true" tabindex="-1"></a>  tmp_manhattan <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(<span class="fu">get</span>(i), <span class="at">method =</span> <span class="st">&quot;manhattan&quot;</span>) </span>
<span id="cb70-27"><a href="#cb70-27" aria-hidden="true" tabindex="-1"></a>  tmp_manhattan <span class="ot">&lt;-</span> <span class="fu">forceSymmetric</span>(<span class="fu">as.matrix</span>(tmp_manhattan), <span class="at">uplo =</span> <span class="st">&quot;L&quot;</span>)</span>
<span id="cb70-28"><a href="#cb70-28" aria-hidden="true" tabindex="-1"></a>  tmp_manhattan <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tmp_manhattan)</span>
<span id="cb70-29"><a href="#cb70-29" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span>beta_diversity<span class="sc">$</span>manhattan <span class="ot">&lt;-</span> tmp_manhattan</span>
<span id="cb70-30"><a href="#cb70-30" aria-hidden="true" tabindex="-1"></a><span class="do">#### #### #### #### #### #### #### ####  </span></span>
<span id="cb70-31"><a href="#cb70-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste</span>(microeco_path, i, <span class="st">&quot;/metrics/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb70-32"><a href="#cb70-32" aria-hidden="true" tabindex="-1"></a>  tmp_dataset<span class="sc">$</span><span class="fu">save_betadiv</span>(<span class="at">dirpath =</span> <span class="fu">paste</span>(microeco_path, i, <span class="st">&quot;/metrics/&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb70-33"><a href="#cb70-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb70-34"><a href="#cb70-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb70-35"><a href="#cb70-35" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;its_&quot;</span>)</span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(its_ps_perfect_Y0_me<span class="sc">$</span>beta_diversity)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(its_ps_perfect_Y0_me<span class="sc">$</span>beta_diversity)</span></code></pre></div>
<pre><code>[1] &quot;list&quot;
          Length Class  Mode   
bray      3481   -none- numeric
jaccard   3481   -none- numeric
jsd       3481   -none- numeric
gower     3481   -none- numeric
canberra  3481   -none- numeric
manhattan 3481   -none- numeric</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_work&quot;</span>, <span class="st">&quot;Y0&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_work&quot;</span>, <span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_work&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_filt&quot;</span>, <span class="st">&quot;Y0&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_filt&quot;</span>, <span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_filt&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_perfect&quot;</span>, <span class="st">&quot;Y0&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_perfect&quot;</span>, <span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_perfect&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_pime&quot;</span>, <span class="st">&quot;Y0&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_pime&quot;</span>, <span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a><span class="fu">run_microeco_beta_plot</span>(<span class="st">&quot;its_ps_pime&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>(<span class="at">pattern =</span> <span class="st">&quot;PCoA$&quot;</span>)</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
</details>
<p></br></p>
<div class="l-page">
<div class="panelset">
<section id="full-data-set-1" class="panel">
<h4>FULL data set</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_work_combo_plot.png" width="4724" /></p>
</div>
<p><small><strong>(ITS) Figure 1 | PCoA ordination plots of the FULL data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right = Gower distance.</strong></small></p>
</section>
<section id="arbitrary-filtered-3" class="panel">
<h4>Arbitrary filtered</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_filt_combo_plot.png" width="4724" /></p>
</div>
<p><small><strong>(ITS) Figure 2 | PCoA ordination plots of the Arbitrary filtered data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right = Gower distance.</strong></small></p>
</section>
<section id="perfect-filtered-3" class="panel">
<h4>PERFect filtered</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_perfect_combo_plot.png" width="4724" /></p>
</div>
<p><small><strong>(ITS) Figure 3 | PCoA ordination plots of the PERfect filtered data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right = Gower distance.</strong></small></p>
</section>
<section id="pime-filtered-3" class="panel">
<h4>PIME filtered</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_pime_combo_plot.png" width="4724" /></p>
</div>
<p><small><strong>(ITS) Figure 4 | PCoA ordination plots of the PIME filtered data set. Top = phyloseq, bottom = microeco; Left = Jensen-Shannon Divergence, middle = Bray-Curtis dissimilarity, right = Gower distance.</strong></small></p>
</section>
</div>
</div>
<h1 id="within-between-group-distance-plots">Within &amp; Between Group Distance Plots</h1>
<p>Part of the custom function described above—<code>microeco_beta_plot</code>—calculated the sample distances within and between treatment groups. Let’s see what those plots look like. These code blocks are hidden by default but can be accessed using the repo link at the bottom of the page.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="s-rrna-plots-1">16S rRNA Plots</h2>
<div class="l-page">
<details>
<summary>
<strong>FULL data set:</strong> within &amp; between group variation plots.
</summary>
<div class="panelset">
<section id="year-0-16" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_work_Y0_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 5 | Year 0 within (top) and between (bottom) group variation plots of the FULL data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-1-16" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_work_Y1_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 6 | Year 1 within (top) and between (bottom) group variation plots of the FULL data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-4-16" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_work_Y4_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 7 | Year 4 within (top) and between (bottom) group variation plots of the FULL data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
</div>
</details>
<details>
<summary>
<strong>Arbitrary filtered:</strong> within &amp; between group variation plots.
</summary>
<div class="panelset">
<section id="year-0-17" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_filt_Y0_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 8 | Year 0 within (top) and between (bottom) group variation plots of the Arbitrary filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-1-17" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_filt_Y1_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 9 | Year 1 within (top) and between (bottom) group variation plots of the Arbitrary filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-4-17" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_filt_Y4_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 10 | Year 4 within (top) and between (bottom) group variation plots of the Arbitrary filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
</div>
</details>
<details>
<summary>
<strong>PERFect filtered:</strong> within &amp; between group variation plots.
</summary>
<div class="panelset">
<section id="year-0-18" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_perfect_Y0_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 11 | Year 0 within (top) and between (bottom) group variation plots of the PERFect filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-1-18" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_perfect_Y1_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 12 | Year 1 within (top) and between (bottom) group variation plots of the PERFect filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-4-18" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_perfect_Y4_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 13 | Year 4 within (top) and between (bottom) group variation plots of the PERFect filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
</div>
</details>
</div>
<div class="l-page">
<details>
<summary>
<strong>PIME filtered:</strong> within &amp; between group variation plots.
</summary>
<div class="panelset">
<section id="year-0-19" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_pime_Y0_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 14 | Year 0 within (top) and between (bottom) group variation plots of the PIME filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-1-19" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_pime_Y1_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 15 | Year 1 within (top) and between (bottom) group variation plots of the PIME filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-4-19" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/ssu_ps_pime_Y4_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(16S rRNA) Figure 16 | Year 4 within (top) and between (bottom) group variation plots of the PIME filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
</div>
</details>
</div>
<h2 id="its-plots-1">ITS Plots</h2>
<div class="l-page">
<details>
<summary>
<strong>FULL data set:</strong> within &amp; between group variation plots.
</summary>
<div class="panelset">
<section id="year-0-20" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_work_Y0_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 5 | Year 0 within (top) and between (bottom) group variation plots of the FULL data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-1-20" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_work_Y1_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 6 | Year 1 within (top) and between (bottom) group variation plots of the FULL data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-4-20" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_work_Y4_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 7 | Year 4 within (top) and between (bottom) group variation plots of the FULL data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
</div>
</details>
<details>
<summary>
<strong>Arbitrary filtered:</strong> within &amp; between group variation plots.
</summary>
<div class="panelset">
<section id="year-0-21" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_filt_Y0_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 8 | Year 0 within (top) and between (bottom) group variation plots of the Arbitrary filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-1-21" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_filt_Y1_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 9 | Year 1 within (top) and between (bottom) group variation plots of the Arbitrary filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-4-21" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_filt_Y4_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 10 | Year 4 within (top) and between (bottom) group variation plots of the Arbitrary filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
</div>
</details>
<details>
<summary>
<strong>PERFect filtered:</strong> within &amp; between group variation plots.
</summary>
<div class="panelset">
<section id="year-0-22" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_perfect_Y0_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 11 | Year 0 within (top) and between (bottom) group variation plots of the PERFect filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-1-22" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_perfect_Y1_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 12 | Year 1 within (top) and between (bottom) group variation plots of the PERFect filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-4-22" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_perfect_Y4_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 13 | Year 4 within (top) and between (bottom) group variation plots of the PERFect filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
</div>
</details>
</div>
<div class="l-page">
<details>
<summary>
<strong>PIME filtered:</strong> within &amp; between group variation plots.
</summary>
<div class="panelset">
<section id="year-0-23" class="panel">
<h4>Year 0</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_pime_Y0_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 14 | Year 0 within (top) and between (bottom) group variation plots of the PIME filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-1-23" class="panel">
<h4>Year 1</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_pime_Y1_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 15 | Year 1 within (top) and between (bottom) group variation plots of the PIME filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
<section id="year-4-23" class="panel">
<h4>Year 4</h4>
<div class="layout-chunk" data-layout="l-page">
<p><img src="include/beta/its_ps_pime_Y4_me_variation_plots.png" width="4252" /></p>
</div>
<p><small><strong>(ITS) Figure 16 | Year 4 within (top) and between (bottom) group variation plots of the PIME filtered data set. Left to right: Jaccard, Jensen-Shannon, UniFrac (unweighted), UniFrac (weighted). Significant differences denoted by asterisks (* p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001).</strong></small></p>
</section>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 class="appendix" id="source-code">Source Code</h2>
<p>The source code for this page can be accessed on GitHub by <a href="https://github.com/agua-salud/nutrients/blob/main/beta.Rmd">clicking this link</a>.</p>
<h2 class="appendix" id="data-availability">Data Availability</h2>
<p>Data generated in this workflow and the Rdata need to run the workflow can be accessed on figshare at <a href="https://doi.org/10.25573/data.16828063">10.25573/data.16828063</a>.</p>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-oksanen2013community" class="csl-entry" role="doc-biblioentry">
Oksanen, Jari, F. Guillaume Blanchet, Roeland Kindt, Pierre Legendre, Peter R. Minchin, R. B. O’Hara, Gavin L. Simpson, Peter Sólymos, M. Henry H. Stevens, and Helene Wagner. 2012. <span>“Vegan: Community Ecology Package.”</span> <em>R Package Version</em> 2 (0). <a href="https://cran.r-project.org/web/packages/vegan/index.html">https://cran.r-project.org/web/packages/vegan/index.html</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/agua-salud/nutrients/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/agua-salud/nutrients/">https://github.com/agua-salud/nutrients/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<div class="distill-site-nav distill-site-footer">
<p>© Copyright 2021 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; The Agua Salud Project.</p>
<p>Site constructed using <a href="https://rstudio.github.io/distill/">Distill for R Markdown</a>.</p>
</div>
<!--/radix_placeholder_navigation_after_body-->


</body>

</html>
